// @openrouter/sdk/esm/sdk/files.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: bb9172f41ecb
 */
import { filesDelete } from "../funcs/filesDelete.js";
import { filesDownload } from "../funcs/filesDownload.js";
import { filesList } from "../funcs/filesList.js";
import { filesRetrieve } from "../funcs/filesRetrieve.js";
import { filesUpload } from "../funcs/filesUpload.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
import { unwrapResultIterator } from "../types/operations.js";
export class Files extends ClientSDK {
    /**
     * List files
     *
     * @remarks
     * Lists files belonging to the workspace of the authenticating API key.
     */
    async list(request, options) {
        return unwrapResultIterator(filesList(this, request, options));
    }
    /**
     * Upload a file
     *
     * @remarks
     * Uploads a file to be referenced in future API calls. The file is stored under the workspace of the authenticating API key. Maximum file size: 100 MB.
     */
    async upload(request, options) {
        return unwrapAsync(filesUpload(this, request, options));
    }
    /**
     * Delete a file
     *
     * @remarks
     * Deletes a file owned by the requesting workspace. Deletion is irreversible.
     */
    async delete(request, options) {
        return unwrapAsync(filesDelete(this, request, options));
    }
    /**
     * Get file metadata
     *
     * @remarks
     * Retrieves metadata for a single file owned by the requesting workspace.
     */
    async retrieve(request, options) {
        return unwrapAsync(filesRetrieve(this, request, options));
    }
    /**
     * Download file content
     *
     * @remarks
     * Downloads the raw bytes of a file. Only files created server-side are downloadable; uploaded files return 400.
     */
    async download(request, options) {
        return unwrapAsync(filesDownload(this, request, options));
    }
}
