// @openrouter/sdk/esm/sdk/presets.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c13a03e2e16b
 */
import { presetsCreatePresetsChatCompletions } from "../funcs/presetsCreatePresetsChatCompletions.js";
import { presetsCreatePresetsMessages } from "../funcs/presetsCreatePresetsMessages.js";
import { presetsCreatePresetsResponses } from "../funcs/presetsCreatePresetsResponses.js";
import { presetsGet } from "../funcs/presetsGet.js";
import { presetsGetVersion } from "../funcs/presetsGetVersion.js";
import { presetsList } from "../funcs/presetsList.js";
import { presetsListVersions } from "../funcs/presetsListVersions.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
import { unwrapResultIterator } from "../types/operations.js";
export class Presets extends ClientSDK {
    /**
     * List presets
     *
     * @remarks
     * Lists all presets for the authenticated user, ordered by most recently updated first.
     */
    async list(request, options) {
        return unwrapResultIterator(presetsList(this, request, options));
    }
    /**
     * Get a preset
     *
     * @remarks
     * Retrieves a preset by its slug with its currently designated version inline.
     */
    async get(request, options) {
        return unwrapAsync(presetsGet(this, request, options));
    }
    /**
     * Create a preset from a chat-completions request body
     *
     * @remarks
     * Creates a preset (or a new version of an existing one) from an inference request body. Only fields that overlap with the preset config are persisted; other fields (e.g. `messages`, `stream`, `prompt`) are silently ignored.
     */
    async createPresetsChatCompletions(request, options) {
        return unwrapAsync(presetsCreatePresetsChatCompletions(this, request, options));
    }
    /**
     * Create a preset from a messages request body
     *
     * @remarks
     * Creates a preset (or a new version of an existing one) from an inference request body. Only fields that overlap with the preset config are persisted; other fields (e.g. `messages`, `stream`, `prompt`) are silently ignored.
     */
    async createPresetsMessages(request, options) {
        return unwrapAsync(presetsCreatePresetsMessages(this, request, options));
    }
    /**
     * Create a preset from a responses request body
     *
     * @remarks
     * Creates a preset (or a new version of an existing one) from an inference request body. Only fields that overlap with the preset config are persisted; other fields (e.g. `messages`, `stream`, `prompt`) are silently ignored.
     */
    async createPresetsResponses(request, options) {
        return unwrapAsync(presetsCreatePresetsResponses(this, request, options));
    }
    /**
     * List versions of a preset
     *
     * @remarks
     * Lists all versions of a preset, ordered by version number ascending (oldest first).
     */
    async listVersions(request, options) {
        return unwrapResultIterator(presetsListVersions(this, request, options));
    }
    /**
     * Get a specific version of a preset
     *
     * @remarks
     * Retrieves a specific version of a preset by its slug and version number.
     */
    async getVersion(request, options) {
        return unwrapAsync(presetsGetVersion(this, request, options));
    }
}
