// @openrouter/sdk/esm/sdk/sdk.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 784571af2f69
 */
import { ClientSDK } from "../lib/sdks.js";
import { Analytics } from "./analytics.js";
import { APIKeys } from "./apikeys.js";
import { Benchmarks } from "./benchmarks.js";
import { Beta } from "./beta.js";
import { Byok } from "./byok.js";
import { Chat } from "./chat.js";
import { Classifications } from "./classifications.js";
import { Credits } from "./credits.js";
import { Datasets } from "./datasets.js";
import { Embeddings } from "./embeddings.js";
import { Endpoints } from "./endpoints.js";
import { Files } from "./files.js";
import { Generations } from "./generations.js";
import { Guardrails } from "./guardrails.js";
import { Images } from "./images.js";
import { Models } from "./models.js";
import { OAuth } from "./oauth.js";
import { Observability } from "./observability.js";
import { Organization } from "./organization.js";
import { Presets } from "./presets.js";
import { Providers } from "./providers.js";
import { Rerank } from "./rerank.js";
import { Stt } from "./stt.js";
import { Tts } from "./tts.js";
import { VideoGeneration } from "./videogeneration.js";
import { Workspaces } from "./workspaces.js";
import { callModel as callModelFunc, } from "../funcs/call-model.js";
import { ToolType } from "../lib/tool-types.js";
export { ToolType };
// #endregion imports
export class OpenRouter extends ClientSDK {
    get analytics() {
        return (this._analytics ?? (this._analytics = new Analytics(this._options)));
    }
    get beta() {
        return (this._beta ?? (this._beta = new Beta(this._options)));
    }
    get tts() {
        return (this._tts ?? (this._tts = new Tts(this._options)));
    }
    get stt() {
        return (this._stt ?? (this._stt = new Stt(this._options)));
    }
    get oAuth() {
        return (this._oAuth ?? (this._oAuth = new OAuth(this._options)));
    }
    get benchmarks() {
        return (this._benchmarks ?? (this._benchmarks = new Benchmarks(this._options)));
    }
    get byok() {
        return (this._byok ?? (this._byok = new Byok(this._options)));
    }
    get chat() {
        return (this._chat ?? (this._chat = new Chat(this._options)));
    }
    get classifications() {
        return (this._classifications ?? (this._classifications = new Classifications(this._options)));
    }
    get credits() {
        return (this._credits ?? (this._credits = new Credits(this._options)));
    }
    get datasets() {
        return (this._datasets ?? (this._datasets = new Datasets(this._options)));
    }
    get embeddings() {
        return (this._embeddings ?? (this._embeddings = new Embeddings(this._options)));
    }
    get endpoints() {
        return (this._endpoints ?? (this._endpoints = new Endpoints(this._options)));
    }
    get files() {
        return (this._files ?? (this._files = new Files(this._options)));
    }
    get generations() {
        return (this._generations ?? (this._generations = new Generations(this._options)));
    }
    get guardrails() {
        return (this._guardrails ?? (this._guardrails = new Guardrails(this._options)));
    }
    get images() {
        return (this._images ?? (this._images = new Images(this._options)));
    }
    get apiKeys() {
        return (this._apiKeys ?? (this._apiKeys = new APIKeys(this._options)));
    }
    get models() {
        return (this._models ?? (this._models = new Models(this._options)));
    }
    get observability() {
        return (this._observability ?? (this._observability = new Observability(this._options)));
    }
    get organization() {
        return (this._organization ?? (this._organization = new Organization(this._options)));
    }
    get presets() {
        return (this._presets ?? (this._presets = new Presets(this._options)));
    }
    get providers() {
        return (this._providers ?? (this._providers = new Providers(this._options)));
    }
    get rerank() {
        return (this._rerank ?? (this._rerank = new Rerank(this._options)));
    }
    get videoGeneration() {
        return (this._videoGeneration ?? (this._videoGeneration = new VideoGeneration(this._options)));
    }
    get workspaces() {
        return (this._workspaces ?? (this._workspaces = new Workspaces(this._options)));
    }
    // #region sdk-class-body
    callModel(request, options) {
        return callModelFunc(this, request, options);
    }
}
