// @openrouter/sdk/esm/sdk/workspaces.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 72b45379c8d9
 */
import { workspacesBulkAddMembers } from "../funcs/workspacesBulkAddMembers.js";
import { workspacesBulkRemoveMembers } from "../funcs/workspacesBulkRemoveMembers.js";
import { workspacesCreate } from "../funcs/workspacesCreate.js";
import { workspacesDelete } from "../funcs/workspacesDelete.js";
import { workspacesDeleteBudget } from "../funcs/workspacesDeleteBudget.js";
import { workspacesGet } from "../funcs/workspacesGet.js";
import { workspacesList } from "../funcs/workspacesList.js";
import { workspacesListBudgets } from "../funcs/workspacesListBudgets.js";
import { workspacesListMembers } from "../funcs/workspacesListMembers.js";
import { workspacesSetBudget } from "../funcs/workspacesSetBudget.js";
import { workspacesUpdate } from "../funcs/workspacesUpdate.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
import { unwrapResultIterator } from "../types/operations.js";
export class Workspaces extends ClientSDK {
    /**
     * List workspaces
     *
     * @remarks
     * List all workspaces for the authenticated user. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async list(request, options) {
        return unwrapResultIterator(workspacesList(this, request, options));
    }
    /**
     * Create a workspace
     *
     * @remarks
     * Create a new workspace for the authenticated user. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async create(request, options) {
        return unwrapAsync(workspacesCreate(this, request, options));
    }
    /**
     * Delete a workspace
     *
     * @remarks
     * Delete an existing workspace. The default workspace cannot be deleted. Workspaces with active API keys cannot be deleted; remove the keys first. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async delete(request, options) {
        return unwrapAsync(workspacesDelete(this, request, options));
    }
    /**
     * Get a workspace
     *
     * @remarks
     * Get a single workspace by ID or slug. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async get(request, options) {
        return unwrapAsync(workspacesGet(this, request, options));
    }
    /**
     * Update a workspace
     *
     * @remarks
     * Update an existing workspace by ID or slug. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async update(request, options) {
        return unwrapAsync(workspacesUpdate(this, request, options));
    }
    /**
     * List workspace budgets
     *
     * @remarks
     * List all budgets configured for a workspace. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async listBudgets(request, options) {
        return unwrapAsync(workspacesListBudgets(this, request, options));
    }
    /**
     * Delete a workspace budget
     *
     * @remarks
     * Remove the budget for a given interval. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async deleteBudget(request, options) {
        return unwrapAsync(workspacesDeleteBudget(this, request, options));
    }
    /**
     * Create or update a workspace budget
     *
     * @remarks
     * Create or update the budget for a given interval. Budget limits must strictly decrease as the interval narrows (lifetime > monthly > weekly > daily). [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async setBudget(request, options) {
        return unwrapAsync(workspacesSetBudget(this, request, options));
    }
    /**
     * List workspace members
     *
     * @remarks
     * List all members of a workspace. Returns paginated results. For the default workspace, returns all organization members (implicit membership). [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async listMembers(request, options) {
        return unwrapResultIterator(workspacesListMembers(this, request, options));
    }
    /**
     * Bulk add members to a workspace
     *
     * @remarks
     * Add multiple organization members to a workspace. Members are assigned the same role they hold in the organization. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async bulkAddMembers(request, options) {
        return unwrapAsync(workspacesBulkAddMembers(this, request, options));
    }
    /**
     * Bulk remove members from a workspace
     *
     * @remarks
     * Remove multiple members from a workspace. Members with active API keys in the workspace cannot be removed. SCIM-managed members cannot be removed; changes must be made in your identity provider. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async bulkRemoveMembers(request, options) {
        return unwrapAsync(workspacesBulkRemoveMembers(this, request, options));
    }
}
