// @openrouter/sdk/esm/models/additionaltoolsitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d33059f81c21
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { AdvisorServerToolOpenRouter$outboundSchema, } from "./advisorservertoolopenrouter.js";
import { ApplyPatchServerTool$outboundSchema, } from "./applypatchservertool.js";
import { ApplyPatchServerToolOpenRouter$outboundSchema, } from "./applypatchservertoolopenrouter.js";
import { BashServerTool$outboundSchema, } from "./bashservertool.js";
import { CodeInterpreterServerTool$outboundSchema, } from "./codeinterpreterservertool.js";
import { CodexLocalShellTool$outboundSchema, } from "./codexlocalshelltool.js";
import { ComputerUseServerTool$outboundSchema, } from "./computeruseservertool.js";
import { CustomTool$outboundSchema, } from "./customtool.js";
import { DatetimeServerTool$outboundSchema, } from "./datetimeservertool.js";
import { FileSearchServerTool$outboundSchema, } from "./filesearchservertool.js";
import { FilesServerTool$outboundSchema, } from "./filesservertool.js";
import { FusionServerToolOpenRouter$outboundSchema, } from "./fusionservertoolopenrouter.js";
import { ImageGenerationServerTool$outboundSchema, } from "./imagegenerationservertool.js";
import { ImageGenerationServerToolOpenRouter$outboundSchema, } from "./imagegenerationservertoolopenrouter.js";
import { LegacyWebSearchServerTool$outboundSchema, } from "./legacywebsearchservertool.js";
import { McpServerTool$outboundSchema, } from "./mcpservertool.js";
import { NamespaceTool$outboundSchema, } from "./namespacetool.js";
import { Preview20250311WebSearchServerTool$outboundSchema, } from "./preview20250311websearchservertool.js";
import { PreviewWebSearchServerTool$outboundSchema, } from "./previewwebsearchservertool.js";
import { SearchModelsServerToolOpenRouter$outboundSchema, } from "./searchmodelsservertoolopenrouter.js";
import { ShellServerTool$outboundSchema, } from "./shellservertool.js";
import { ShellServerToolOpenRouter$outboundSchema, } from "./shellservertoolopenrouter.js";
import { SubagentServerToolOpenRouter$outboundSchema, } from "./subagentservertoolopenrouter.js";
import { WebFetchServerTool$outboundSchema, } from "./webfetchservertool.js";
import { WebSearchServerTool$outboundSchema, } from "./websearchservertool.js";
import { WebSearchServerToolOpenRouter$outboundSchema, } from "./websearchservertoolopenrouter.js";
export const AdditionalToolsItemRole = {
    Unknown: "unknown",
    User: "user",
    Assistant: "assistant",
    System: "system",
    Critic: "critic",
    Discriminator: "discriminator",
    Developer: "developer",
    Tool: "tool",
};
export const AdditionalToolsItemToolType = {
    Function: "function",
};
export const TypeAdditionalTools = {
    AdditionalTools: "additional_tools",
};
/** @internal */
export const AdditionalToolsItemRole$outboundSchema = openEnums.outboundSchema(AdditionalToolsItemRole);
/** @internal */
export const AdditionalToolsItemTool$outboundSchema = z.object({
    type: z.string(),
    additionalProperties: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function additionalToolsItemToolToJSON(additionalToolsItemTool) {
    return JSON.stringify(AdditionalToolsItemTool$outboundSchema.parse(additionalToolsItemTool));
}
/** @internal */
export const AdditionalToolsItemToolType$outboundSchema = z.enum(AdditionalToolsItemToolType);
/** @internal */
export const AdditionalToolsItemToolFunction$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    name: z.string(),
    parameters: z.nullable(z.record(z.string(), z.any())),
    strict: z.nullable(z.boolean()).optional(),
    type: AdditionalToolsItemToolType$outboundSchema,
});
export function additionalToolsItemToolFunctionToJSON(additionalToolsItemToolFunction) {
    return JSON.stringify(AdditionalToolsItemToolFunction$outboundSchema.parse(additionalToolsItemToolFunction));
}
/** @internal */
export const AdditionalToolsItemToolUnion$outboundSchema = z.union([
    ComputerUseServerTool$outboundSchema,
    NamespaceTool$outboundSchema,
    z.lazy(() => AdditionalToolsItemToolFunction$outboundSchema),
    FileSearchServerTool$outboundSchema,
    CodeInterpreterServerTool$outboundSchema,
    McpServerTool$outboundSchema,
    CustomTool$outboundSchema,
    PreviewWebSearchServerTool$outboundSchema,
    Preview20250311WebSearchServerTool$outboundSchema,
    LegacyWebSearchServerTool$outboundSchema,
    WebSearchServerTool$outboundSchema,
    ImageGenerationServerTool$outboundSchema,
    CodexLocalShellTool$outboundSchema,
    ShellServerTool$outboundSchema,
    ApplyPatchServerTool$outboundSchema,
    AdvisorServerToolOpenRouter$outboundSchema,
    SubagentServerToolOpenRouter$outboundSchema,
    DatetimeServerTool$outboundSchema,
    FilesServerTool$outboundSchema,
    FusionServerToolOpenRouter$outboundSchema,
    ImageGenerationServerToolOpenRouter$outboundSchema,
    SearchModelsServerToolOpenRouter$outboundSchema,
    WebFetchServerTool$outboundSchema,
    WebSearchServerToolOpenRouter$outboundSchema,
    ApplyPatchServerToolOpenRouter$outboundSchema,
    BashServerTool$outboundSchema,
    ShellServerToolOpenRouter$outboundSchema,
    z.lazy(() => AdditionalToolsItemTool$outboundSchema),
]);
export function additionalToolsItemToolUnionToJSON(additionalToolsItemToolUnion) {
    return JSON.stringify(AdditionalToolsItemToolUnion$outboundSchema.parse(additionalToolsItemToolUnion));
}
/** @internal */
export const TypeAdditionalTools$outboundSchema = z.enum(TypeAdditionalTools);
/** @internal */
export const AdditionalToolsItem$outboundSchema = z.object({
    id: z.nullable(z.string()).optional(),
    role: AdditionalToolsItemRole$outboundSchema,
    tools: z.array(z.union([
        ComputerUseServerTool$outboundSchema,
        NamespaceTool$outboundSchema,
        z.lazy(() => AdditionalToolsItemToolFunction$outboundSchema),
        FileSearchServerTool$outboundSchema,
        CodeInterpreterServerTool$outboundSchema,
        McpServerTool$outboundSchema,
        CustomTool$outboundSchema,
        PreviewWebSearchServerTool$outboundSchema,
        Preview20250311WebSearchServerTool$outboundSchema,
        LegacyWebSearchServerTool$outboundSchema,
        WebSearchServerTool$outboundSchema,
        ImageGenerationServerTool$outboundSchema,
        CodexLocalShellTool$outboundSchema,
        ShellServerTool$outboundSchema,
        ApplyPatchServerTool$outboundSchema,
        AdvisorServerToolOpenRouter$outboundSchema,
        SubagentServerToolOpenRouter$outboundSchema,
        DatetimeServerTool$outboundSchema,
        FilesServerTool$outboundSchema,
        FusionServerToolOpenRouter$outboundSchema,
        ImageGenerationServerToolOpenRouter$outboundSchema,
        SearchModelsServerToolOpenRouter$outboundSchema,
        WebFetchServerTool$outboundSchema,
        WebSearchServerToolOpenRouter$outboundSchema,
        ApplyPatchServerToolOpenRouter$outboundSchema,
        BashServerTool$outboundSchema,
        ShellServerToolOpenRouter$outboundSchema,
        z.lazy(() => AdditionalToolsItemTool$outboundSchema),
    ])),
    type: TypeAdditionalTools$outboundSchema,
});
export function additionalToolsItemToJSON(additionalToolsItem) {
    return JSON.stringify(AdditionalToolsItem$outboundSchema.parse(additionalToolsItem));
}
