// @openrouter/sdk/esm/models/advisorreasoning.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4c86be6688b0
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
/**
 * Reasoning effort level for the advisor call.
 */
export const AdvisorReasoningEffort = {
    Max: "max",
    Xhigh: "xhigh",
    High: "high",
    Medium: "medium",
    Low: "low",
    Minimal: "minimal",
    None: "none",
};
/** @internal */
export const AdvisorReasoningEffort$outboundSchema = openEnums.outboundSchema(AdvisorReasoningEffort);
/** @internal */
export const AdvisorReasoning$outboundSchema = z.object({
    effort: AdvisorReasoningEffort$outboundSchema.optional(),
    maxTokens: z.int().optional(),
}).transform((v) => {
    return remap$(v, {
        maxTokens: "max_tokens",
    });
});
export function advisorReasoningToJSON(advisorReasoning) {
    return JSON.stringify(AdvisorReasoning$outboundSchema.parse(advisorReasoning));
}
