// @openrouter/sdk/esm/models/advisorservertoolconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4c2b85fa1b08
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { AdvisorNestedTool$outboundSchema, } from "./advisornestedtool.js";
import { AdvisorReasoning$outboundSchema, } from "./advisorreasoning.js";
/** @internal */
export const AdvisorServerToolConfig$outboundSchema = z.object({
    forwardTranscript: z.boolean().optional(),
    instructions: z.string().optional(),
    maxCompletionTokens: z.int().optional(),
    maxToolCalls: z.int().optional(),
    model: z.string().optional(),
    name: z.string().optional(),
    reasoning: AdvisorReasoning$outboundSchema.optional(),
    stream: z.boolean().optional(),
    temperature: z.number().optional(),
    tools: z.array(AdvisorNestedTool$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        forwardTranscript: "forward_transcript",
        maxCompletionTokens: "max_completion_tokens",
        maxToolCalls: "max_tool_calls",
    });
});
export function advisorServerToolConfigToJSON(advisorServerToolConfig) {
    return JSON.stringify(AdvisorServerToolConfig$outboundSchema.parse(advisorServerToolConfig));
}
