// @openrouter/sdk/esm/models/agentmessageitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: fc9151085f62
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { InputText$outboundSchema, } from "./inputtext.js";
export const AgentMessageItemDetail = {
    Auto: "auto",
    High: "high",
    Low: "low",
    Original: "original",
};
export const TypeAgentMessage = {
    AgentMessage: "agent_message",
};
/** @internal */
export const Agent$outboundSchema = z.object({
    agentName: z.string(),
    additionalProperties: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            agentName: "agent_name",
            additionalProperties: null,
        }),
    };
});
export function agentToJSON(agent) {
    return JSON.stringify(Agent$outboundSchema.parse(agent));
}
/** @internal */
export const ContentEncryptedContent$outboundSchema = z.object({
    encryptedContent: z.string(),
    type: z.literal("encrypted_content"),
    additionalProperties: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            encryptedContent: "encrypted_content",
            additionalProperties: null,
        }),
    };
});
export function contentEncryptedContentToJSON(contentEncryptedContent) {
    return JSON.stringify(ContentEncryptedContent$outboundSchema.parse(contentEncryptedContent));
}
/** @internal */
export const AgentMessageItemDetail$outboundSchema = openEnums.outboundSchema(AgentMessageItemDetail);
/** @internal */
export const AgentMessageItemContentInputImage$outboundSchema = z.object({
    detail: AgentMessageItemDetail$outboundSchema,
    imageUrl: z.nullable(z.string()).optional(),
    type: z.literal("input_image"),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function agentMessageItemContentInputImageToJSON(agentMessageItemContentInputImage) {
    return JSON.stringify(AgentMessageItemContentInputImage$outboundSchema.parse(agentMessageItemContentInputImage));
}
/** @internal */
export const AgentMessageItemContentUnion$outboundSchema = z.union([
    InputText$outboundSchema,
    z.lazy(() => AgentMessageItemContentInputImage$outboundSchema),
    z.lazy(() => ContentEncryptedContent$outboundSchema),
]);
export function agentMessageItemContentUnionToJSON(agentMessageItemContentUnion) {
    return JSON.stringify(AgentMessageItemContentUnion$outboundSchema.parse(agentMessageItemContentUnion));
}
/** @internal */
export const TypeAgentMessage$outboundSchema = z.enum(TypeAgentMessage);
/** @internal */
export const AgentMessageItem$outboundSchema = z.object({
    agent: z.nullable(z.lazy(() => Agent$outboundSchema)).optional(),
    author: z.string(),
    content: z.array(z.union([
        InputText$outboundSchema,
        z.lazy(() => AgentMessageItemContentInputImage$outboundSchema),
        z.lazy(() => ContentEncryptedContent$outboundSchema),
    ])),
    id: z.nullable(z.string()).optional(),
    recipient: z.string(),
    type: TypeAgentMessage$outboundSchema,
    additionalProperties: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function agentMessageItemToJSON(agentMessageItem) {
    return JSON.stringify(AgentMessageItem$outboundSchema.parse(agentMessageItem));
}
