// @openrouter/sdk/esm/models/anthropictoolsearchtoolregex.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 70c2d1fd66ec
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { AnthropicAllowedCallers$outboundSchema, } from "./anthropicallowedcallers.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
export const AnthropicToolSearchToolRegexName = {
    ToolSearchToolRegex: "tool_search_tool_regex",
};
export const AnthropicToolSearchToolRegexType = {
    ToolSearchToolRegex20251119: "tool_search_tool_regex_20251119",
    ToolSearchToolRegex: "tool_search_tool_regex",
};
/** @internal */
export const AnthropicToolSearchToolRegexName$outboundSchema = z.enum(AnthropicToolSearchToolRegexName);
/** @internal */
export const AnthropicToolSearchToolRegexType$outboundSchema = openEnums.outboundSchema(AnthropicToolSearchToolRegexType);
/** @internal */
export const AnthropicToolSearchToolRegex$outboundSchema = z.object({
    allowedCallers: z.array(AnthropicAllowedCallers$outboundSchema).optional(),
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    deferLoading: z.boolean().optional(),
    name: AnthropicToolSearchToolRegexName$outboundSchema,
    strict: z.boolean().optional(),
    type: AnthropicToolSearchToolRegexType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        allowedCallers: "allowed_callers",
        cacheControl: "cache_control",
        deferLoading: "defer_loading",
    });
});
export function anthropicToolSearchToolRegexToJSON(anthropicToolSearchToolRegex) {
    return JSON.stringify(AnthropicToolSearchToolRegex$outboundSchema.parse(anthropicToolSearchToolRegex));
}
