// @openrouter/sdk/esm/models/anthropiccompactionusageiteration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5982ebca6c31
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { AnthropicIterationCacheCreation$inboundSchema, } from "./anthropiciterationcachecreation.js";
export const AnthropicCompactionUsageIterationType = {
    Compaction: "compaction",
};
/** @internal */
export const AnthropicCompactionUsageIterationType$inboundSchema = z.enum(AnthropicCompactionUsageIterationType);
/** @internal */
export const AnthropicCompactionUsageIteration$inboundSchema = z.object({
    cache_creation: z.nullable(AnthropicIterationCacheCreation$inboundSchema)
        .optional(),
    cache_creation_input_tokens: z.int().optional(),
    cache_read_input_tokens: z.int().optional(),
    input_tokens: z.int().optional(),
    output_tokens: z.int().optional(),
    type: AnthropicCompactionUsageIterationType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "cache_creation": "cacheCreation",
        "cache_creation_input_tokens": "cacheCreationInputTokens",
        "cache_read_input_tokens": "cacheReadInputTokens",
        "input_tokens": "inputTokens",
        "output_tokens": "outputTokens",
    });
});
export function anthropicCompactionUsageIterationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AnthropicCompactionUsageIteration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AnthropicCompactionUsageIteration' from JSON`);
}
