// @openrouter/sdk/esm/models/anthropicwebsearchresultblockparam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b8fe75186751
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
export const AnthropicWebSearchResultBlockParamType = {
    WebSearchResult: "web_search_result",
};
/** @internal */
export const AnthropicWebSearchResultBlockParamType$outboundSchema = z.enum(AnthropicWebSearchResultBlockParamType);
/** @internal */
export const AnthropicWebSearchResultBlockParam$outboundSchema = z.object({
    encryptedContent: z.string(),
    pageAge: z.nullable(z.string()).optional(),
    title: z.string(),
    type: AnthropicWebSearchResultBlockParamType$outboundSchema,
    url: z.string(),
}).transform((v) => {
    return remap$(v, {
        encryptedContent: "encrypted_content",
        pageAge: "page_age",
    });
});
export function anthropicWebSearchResultBlockParamToJSON(anthropicWebSearchResultBlockParam) {
    return JSON.stringify(AnthropicWebSearchResultBlockParam$outboundSchema.parse(anthropicWebSearchResultBlockParam));
}
