// @openrouter/sdk/esm/models/anthropicdocumentblockparam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: da62cae77bdf
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { AnthropicBase64PdfSource$outboundSchema, } from "./anthropicbase64pdfsource.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
import { AnthropicFileDocumentSource$outboundSchema, } from "./anthropicfiledocumentsource.js";
import { AnthropicImageBlockParam$outboundSchema, } from "./anthropicimageblockparam.js";
import { AnthropicPlainTextSource$outboundSchema, } from "./anthropicplaintextsource.js";
import { AnthropicTextBlockParam$outboundSchema, } from "./anthropictextblockparam.js";
import { AnthropicUrlPdfSource$outboundSchema, } from "./anthropicurlpdfsource.js";
/** @internal */
export const AnthropicDocumentBlockParamCitations$outboundSchema = z.object({
    enabled: z.boolean().optional(),
});
export function anthropicDocumentBlockParamCitationsToJSON(anthropicDocumentBlockParamCitations) {
    return JSON.stringify(AnthropicDocumentBlockParamCitations$outboundSchema.parse(anthropicDocumentBlockParamCitations));
}
/** @internal */
export const AnthropicDocumentBlockParamContent1$outboundSchema = z.union([
    AnthropicImageBlockParam$outboundSchema,
    AnthropicTextBlockParam$outboundSchema,
]);
export function anthropicDocumentBlockParamContent1ToJSON(anthropicDocumentBlockParamContent1) {
    return JSON.stringify(AnthropicDocumentBlockParamContent1$outboundSchema.parse(anthropicDocumentBlockParamContent1));
}
/** @internal */
export const AnthropicDocumentBlockParamContent2$outboundSchema = z.union([
    z.string(),
    z.array(z.union([
        AnthropicImageBlockParam$outboundSchema,
        AnthropicTextBlockParam$outboundSchema,
    ])),
]);
export function anthropicDocumentBlockParamContent2ToJSON(anthropicDocumentBlockParamContent2) {
    return JSON.stringify(AnthropicDocumentBlockParamContent2$outboundSchema.parse(anthropicDocumentBlockParamContent2));
}
/** @internal */
export const SourceContent$outboundSchema = z.object({
    content: z.union([
        z.string(),
        z.array(z.union([
            AnthropicImageBlockParam$outboundSchema,
            AnthropicTextBlockParam$outboundSchema,
        ])),
    ]),
    type: z.literal("content"),
});
export function sourceContentToJSON(sourceContent) {
    return JSON.stringify(SourceContent$outboundSchema.parse(sourceContent));
}
/** @internal */
export const AnthropicDocumentBlockParamSourceUnion$outboundSchema = z.union([
    AnthropicBase64PdfSource$outboundSchema,
    AnthropicPlainTextSource$outboundSchema,
    z.lazy(() => SourceContent$outboundSchema),
    AnthropicUrlPdfSource$outboundSchema,
    AnthropicFileDocumentSource$outboundSchema,
]);
export function anthropicDocumentBlockParamSourceUnionToJSON(anthropicDocumentBlockParamSourceUnion) {
    return JSON.stringify(AnthropicDocumentBlockParamSourceUnion$outboundSchema.parse(anthropicDocumentBlockParamSourceUnion));
}
/** @internal */
export const AnthropicDocumentBlockParam$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    citations: z.nullable(z.lazy(() => AnthropicDocumentBlockParamCitations$outboundSchema)).optional(),
    context: z.nullable(z.string()).optional(),
    source: z.union([
        AnthropicBase64PdfSource$outboundSchema,
        AnthropicPlainTextSource$outboundSchema,
        z.lazy(() => SourceContent$outboundSchema),
        AnthropicUrlPdfSource$outboundSchema,
        AnthropicFileDocumentSource$outboundSchema,
    ]),
    title: z.nullable(z.string()).optional(),
    type: z.literal("document"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function anthropicDocumentBlockParamToJSON(anthropicDocumentBlockParam) {
    return JSON.stringify(AnthropicDocumentBlockParam$outboundSchema.parse(anthropicDocumentBlockParam));
}
