// @openrouter/sdk/esm/models/anthropictoolsearchtoolbm25.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d06d81d905be
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { AnthropicAllowedCallers$outboundSchema, } from "./anthropicallowedcallers.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
export const AnthropicToolSearchToolBm25Name = {
    ToolSearchToolBm25: "tool_search_tool_bm25",
};
export const AnthropicToolSearchToolBm25Type = {
    ToolSearchToolBm2520251119: "tool_search_tool_bm25_20251119",
    ToolSearchToolBm25: "tool_search_tool_bm25",
};
/** @internal */
export const AnthropicToolSearchToolBm25Name$outboundSchema = z.enum(AnthropicToolSearchToolBm25Name);
/** @internal */
export const AnthropicToolSearchToolBm25Type$outboundSchema = openEnums.outboundSchema(AnthropicToolSearchToolBm25Type);
/** @internal */
export const AnthropicToolSearchToolBm25$outboundSchema = z.object({
    allowedCallers: z.array(AnthropicAllowedCallers$outboundSchema).optional(),
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    deferLoading: z.boolean().optional(),
    name: AnthropicToolSearchToolBm25Name$outboundSchema,
    strict: z.boolean().optional(),
    type: AnthropicToolSearchToolBm25Type$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        allowedCallers: "allowed_callers",
        cacheControl: "cache_control",
        deferLoading: "defer_loading",
    });
});
export function anthropicToolSearchToolBm25ToJSON(anthropicToolSearchToolBm25) {
    return JSON.stringify(AnthropicToolSearchToolBm25$outboundSchema.parse(anthropicToolSearchToolBm25));
}
