// @openrouter/sdk/esm/models/anthropicimageblockparam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 463612aae623
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { AnthropicBase64ImageSource$outboundSchema, } from "./anthropicbase64imagesource.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
import { AnthropicUrlImageSource$outboundSchema, } from "./anthropicurlimagesource.js";
/** @internal */
export const AnthropicImageBlockParamSource$outboundSchema = z.union([
    AnthropicBase64ImageSource$outboundSchema,
    AnthropicUrlImageSource$outboundSchema,
]);
export function anthropicImageBlockParamSourceToJSON(anthropicImageBlockParamSource) {
    return JSON.stringify(AnthropicImageBlockParamSource$outboundSchema.parse(anthropicImageBlockParamSource));
}
/** @internal */
export const AnthropicImageBlockParam$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    source: z.union([
        AnthropicBase64ImageSource$outboundSchema,
        AnthropicUrlImageSource$outboundSchema,
    ]),
    type: z.literal("image"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function anthropicImageBlockParamToJSON(anthropicImageBlockParam) {
    return JSON.stringify(AnthropicImageBlockParam$outboundSchema.parse(anthropicImageBlockParam));
}
