// @openrouter/sdk/esm/models/anthropiccitationcontentblocklocationparam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1a2db0e73f5d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
/** @internal */
export const AnthropicCitationContentBlockLocationParam$outboundSchema = z.object({
    citedText: z.string(),
    documentIndex: z.int(),
    documentTitle: z.nullable(z.string()),
    endBlockIndex: z.int(),
    startBlockIndex: z.int(),
    type: z.literal("content_block_location"),
}).transform((v) => {
    return remap$(v, {
        citedText: "cited_text",
        documentIndex: "document_index",
        documentTitle: "document_title",
        endBlockIndex: "end_block_index",
        startBlockIndex: "start_block_index",
    });
});
export function anthropicCitationContentBlockLocationParamToJSON(anthropicCitationContentBlockLocationParam) {
    return JSON.stringify(AnthropicCitationContentBlockLocationParam$outboundSchema.parse(anthropicCitationContentBlockLocationParam));
}
