// @openrouter/sdk/esm/models/anthropicmessageusageiteration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: fabb23153447
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { AnthropicIterationCacheCreation$inboundSchema, } from "./anthropiciterationcachecreation.js";
export const AnthropicMessageUsageIterationType = {
    Message: "message",
};
/** @internal */
export const AnthropicMessageUsageIterationType$inboundSchema = z.enum(AnthropicMessageUsageIterationType);
/** @internal */
export const AnthropicMessageUsageIteration$inboundSchema = z.object({
    cache_creation: z.nullable(AnthropicIterationCacheCreation$inboundSchema)
        .optional(),
    cache_creation_input_tokens: z.int().optional(),
    cache_read_input_tokens: z.int().optional(),
    input_tokens: z.int().optional(),
    output_tokens: z.int().optional(),
    model: z.string().optional(),
    type: AnthropicMessageUsageIterationType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "cache_creation": "cacheCreation",
        "cache_creation_input_tokens": "cacheCreationInputTokens",
        "cache_read_input_tokens": "cacheReadInputTokens",
        "input_tokens": "inputTokens",
        "output_tokens": "outputTokens",
    });
});
export function anthropicMessageUsageIterationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AnthropicMessageUsageIteration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AnthropicMessageUsageIteration' from JSON`);
}
