// @openrouter/sdk/esm/models/anthropicsearchresultblockparam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: db204b792de1
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
import { AnthropicTextBlockParam$outboundSchema, } from "./anthropictextblockparam.js";
/** @internal */
export const AnthropicSearchResultBlockParamCitations$outboundSchema = z.object({
    enabled: z.boolean().optional(),
});
export function anthropicSearchResultBlockParamCitationsToJSON(anthropicSearchResultBlockParamCitations) {
    return JSON.stringify(AnthropicSearchResultBlockParamCitations$outboundSchema.parse(anthropicSearchResultBlockParamCitations));
}
/** @internal */
export const AnthropicSearchResultBlockParam$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    citations: z.lazy(() => AnthropicSearchResultBlockParamCitations$outboundSchema).optional(),
    content: z.array(AnthropicTextBlockParam$outboundSchema),
    source: z.string(),
    title: z.string(),
    type: z.literal("search_result"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function anthropicSearchResultBlockParamToJSON(anthropicSearchResultBlockParam) {
    return JSON.stringify(AnthropicSearchResultBlockParam$outboundSchema.parse(anthropicSearchResultBlockParam));
}
