// @openrouter/sdk/esm/models/anthropicadvisormessageusageiteration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ff2ce164571e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { AnthropicIterationCacheCreation$inboundSchema, } from "./anthropiciterationcachecreation.js";
export const AnthropicAdvisorMessageUsageIterationType = {
    AdvisorMessage: "advisor_message",
};
/** @internal */
export const AnthropicAdvisorMessageUsageIterationType$inboundSchema = z.enum(AnthropicAdvisorMessageUsageIterationType);
/** @internal */
export const AnthropicAdvisorMessageUsageIteration$inboundSchema = z.object({
    cache_creation: z.nullable(AnthropicIterationCacheCreation$inboundSchema)
        .optional(),
    cache_creation_input_tokens: z.int().optional(),
    cache_read_input_tokens: z.int().optional(),
    input_tokens: z.int().optional(),
    output_tokens: z.int().optional(),
    model: z.string(),
    type: AnthropicAdvisorMessageUsageIterationType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "cache_creation": "cacheCreation",
        "cache_creation_input_tokens": "cacheCreationInputTokens",
        "cache_read_input_tokens": "cacheReadInputTokens",
        "input_tokens": "inputTokens",
        "output_tokens": "outputTokens",
    });
});
export function anthropicAdvisorMessageUsageIterationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AnthropicAdvisorMessageUsageIteration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AnthropicAdvisorMessageUsageIteration' from JSON`);
}
