// @openrouter/sdk/esm/models/apierrortype.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 240e785ddd59
 */
import * as openEnums from "../types/enums.js";
/**
 * Canonical OpenRouter error type, stable across all API formats
 */
export const ApiErrorType = {
    ContextLengthExceeded: "context_length_exceeded",
    MaxTokensExceeded: "max_tokens_exceeded",
    TokenLimitExceeded: "token_limit_exceeded",
    StringTooLong: "string_too_long",
    Authentication: "authentication",
    PermissionDenied: "permission_denied",
    PaymentRequired: "payment_required",
    RateLimitExceeded: "rate_limit_exceeded",
    ProviderOverloaded: "provider_overloaded",
    ProviderUnavailable: "provider_unavailable",
    InvalidRequest: "invalid_request",
    InvalidPrompt: "invalid_prompt",
    NotFound: "not_found",
    PreconditionFailed: "precondition_failed",
    PayloadTooLarge: "payload_too_large",
    Unprocessable: "unprocessable",
    ContentPolicyViolation: "content_policy_violation",
    Refusal: "refusal",
    InvalidImage: "invalid_image",
    ImageTooLarge: "image_too_large",
    ImageTooSmall: "image_too_small",
    UnsupportedImageFormat: "unsupported_image_format",
    ImageNotFound: "image_not_found",
    ImageDownloadFailed: "image_download_failed",
    Server: "server",
    Timeout: "timeout",
    Unmapped: "unmapped",
};
/** @internal */
export const ApiErrorType$inboundSchema = openEnums.inboundSchema(ApiErrorType);
