// @openrouter/sdk/esm/models/applypatchcalloperation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 11811921d55f
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { ApplyPatchCreateFileOperation$inboundSchema, ApplyPatchCreateFileOperation$outboundSchema, } from "./applypatchcreatefileoperation.js";
import { ApplyPatchDeleteFileOperation$inboundSchema, ApplyPatchDeleteFileOperation$outboundSchema, } from "./applypatchdeletefileoperation.js";
import { ApplyPatchUpdateFileOperation$inboundSchema, ApplyPatchUpdateFileOperation$outboundSchema, } from "./applypatchupdatefileoperation.js";
/** @internal */
export const ApplyPatchCallOperation$inboundSchema = discriminatedUnion("type", {
    create_file: ApplyPatchCreateFileOperation$inboundSchema,
    delete_file: ApplyPatchDeleteFileOperation$inboundSchema,
    update_file: ApplyPatchUpdateFileOperation$inboundSchema,
});
/** @internal */
export const ApplyPatchCallOperation$outboundSchema = z.union([
    ApplyPatchCreateFileOperation$outboundSchema,
    ApplyPatchDeleteFileOperation$outboundSchema,
    ApplyPatchUpdateFileOperation$outboundSchema,
]);
export function applyPatchCallOperationToJSON(applyPatchCallOperation) {
    return JSON.stringify(ApplyPatchCallOperation$outboundSchema.parse(applyPatchCallOperation));
}
export function applyPatchCallOperationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApplyPatchCallOperation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApplyPatchCallOperation' from JSON`);
}
