// @openrouter/sdk/esm/models/applypatchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ee989280d886
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const ApplyPatchServerToolType = {
    ApplyPatch: "apply_patch",
};
/** @internal */
export const ApplyPatchServerToolType$inboundSchema = z.enum(ApplyPatchServerToolType);
/** @internal */
export const ApplyPatchServerToolType$outboundSchema = ApplyPatchServerToolType$inboundSchema;
/** @internal */
export const ApplyPatchServerTool$inboundSchema = z.object({
    type: ApplyPatchServerToolType$inboundSchema,
});
/** @internal */
export const ApplyPatchServerTool$outboundSchema = z.object({
    type: ApplyPatchServerToolType$outboundSchema,
});
export function applyPatchServerToolToJSON(applyPatchServerTool) {
    return JSON.stringify(ApplyPatchServerTool$outboundSchema.parse(applyPatchServerTool));
}
export function applyPatchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApplyPatchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApplyPatchServerTool' from JSON`);
}
