// @openrouter/sdk/esm/models/applypatchcalloutputitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: df3c379b4d4e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ApplyPatchCallOutputItemStatus = {
    Completed: "completed",
    Failed: "failed",
};
export const ApplyPatchCallOutputItemType = {
    ApplyPatchCallOutput: "apply_patch_call_output",
};
/** @internal */
export const ApplyPatchCallOutputItemStatus$inboundSchema = openEnums.inboundSchema(ApplyPatchCallOutputItemStatus);
/** @internal */
export const ApplyPatchCallOutputItemStatus$outboundSchema = openEnums.outboundSchema(ApplyPatchCallOutputItemStatus);
/** @internal */
export const ApplyPatchCallOutputItemType$inboundSchema = z.enum(ApplyPatchCallOutputItemType);
/** @internal */
export const ApplyPatchCallOutputItemType$outboundSchema = ApplyPatchCallOutputItemType$inboundSchema;
/** @internal */
export const ApplyPatchCallOutputItem$inboundSchema = z.object({
    call_id: z.string(),
    id: z.nullable(z.string()).optional(),
    output: z.nullable(z.string()).optional(),
    status: ApplyPatchCallOutputItemStatus$inboundSchema,
    type: ApplyPatchCallOutputItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
/** @internal */
export const ApplyPatchCallOutputItem$outboundSchema = z.object({
    callId: z.string(),
    id: z.nullable(z.string()).optional(),
    output: z.nullable(z.string()).optional(),
    status: ApplyPatchCallOutputItemStatus$outboundSchema,
    type: ApplyPatchCallOutputItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function applyPatchCallOutputItemToJSON(applyPatchCallOutputItem) {
    return JSON.stringify(ApplyPatchCallOutputItem$outboundSchema.parse(applyPatchCallOutputItem));
}
export function applyPatchCallOutputItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApplyPatchCallOutputItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApplyPatchCallOutputItem' from JSON`);
}
