// @openrouter/sdk/esm/models/applypatchcallitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: bb5c3b2b01b9
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ApplyPatchCallOperation$inboundSchema, ApplyPatchCallOperation$outboundSchema, } from "./applypatchcalloperation.js";
import { ApplyPatchCallStatus$inboundSchema, ApplyPatchCallStatus$outboundSchema, } from "./applypatchcallstatus.js";
export const ApplyPatchCallItemType = {
    ApplyPatchCall: "apply_patch_call",
};
/** @internal */
export const ApplyPatchCallItemType$inboundSchema = z.enum(ApplyPatchCallItemType);
/** @internal */
export const ApplyPatchCallItemType$outboundSchema = ApplyPatchCallItemType$inboundSchema;
/** @internal */
export const ApplyPatchCallItem$inboundSchema = z.object({
    call_id: z.string(),
    id: z.nullable(z.string()).optional(),
    operation: ApplyPatchCallOperation$inboundSchema,
    status: ApplyPatchCallStatus$inboundSchema,
    type: ApplyPatchCallItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
/** @internal */
export const ApplyPatchCallItem$outboundSchema = z.object({
    callId: z.string(),
    id: z.nullable(z.string()).optional(),
    operation: ApplyPatchCallOperation$outboundSchema,
    status: ApplyPatchCallStatus$outboundSchema,
    type: ApplyPatchCallItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function applyPatchCallItemToJSON(applyPatchCallItem) {
    return JSON.stringify(ApplyPatchCallItem$outboundSchema.parse(applyPatchCallItem));
}
export function applyPatchCallItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApplyPatchCallItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApplyPatchCallItem' from JSON`);
}
