// @openrouter/sdk/esm/models/byokkey.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7f8b3abf1e77
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { BYOKProviderSlug$inboundSchema, } from "./byokproviderslug.js";
/** @internal */
export const BYOKKey$inboundSchema = z.object({
    allowed_api_key_hashes: z.nullable(z.array(z.string())),
    allowed_models: z.nullable(z.array(z.string())),
    allowed_user_ids: z.nullable(z.array(z.string())),
    created_at: z.string(),
    disabled: z.boolean(),
    id: z.string(),
    is_fallback: z.boolean(),
    label: z.string(),
    name: z.nullable(z.string()).optional(),
    provider: BYOKProviderSlug$inboundSchema,
    sort_order: z.int(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "allowed_api_key_hashes": "allowedApiKeyHashes",
        "allowed_models": "allowedModels",
        "allowed_user_ids": "allowedUserIds",
        "created_at": "createdAt",
        "is_fallback": "isFallback",
        "sort_order": "sortOrder",
        "workspace_id": "workspaceId",
    });
});
export function byokKeyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BYOKKey$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BYOKKey' from JSON`);
}
