// @openrouter/sdk/esm/models/chatrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5d2646ed8dcf
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
import { AutoBetaRouterPlugin$outboundSchema, } from "./autobetarouterplugin.js";
import { AutoRouterPlugin$outboundSchema, } from "./autorouterplugin.js";
import { ChatDebugOptions$outboundSchema, } from "./chatdebugoptions.js";
import { ChatFormatGrammarConfig$outboundSchema, } from "./chatformatgrammarconfig.js";
import { ChatFormatJsonObjectConfig$outboundSchema, } from "./chatformatjsonobjectconfig.js";
import { ChatFormatJsonSchemaConfig$outboundSchema, } from "./chatformatjsonschemaconfig.js";
import { ChatFormatPythonConfig$outboundSchema, } from "./chatformatpythonconfig.js";
import { ChatFormatTextConfig$outboundSchema, } from "./chatformattextconfig.js";
import { ChatFunctionTool$outboundSchema, } from "./chatfunctiontool.js";
import { ChatMessages$outboundSchema, } from "./chatmessages.js";
import { ChatReasoningSummaryVerbosityEnum$outboundSchema, } from "./chatreasoningsummaryverbosityenum.js";
import { ChatStreamOptions$outboundSchema, } from "./chatstreamoptions.js";
import { ChatToolChoice$outboundSchema, } from "./chattoolchoice.js";
import { ContextCompressionPlugin$outboundSchema, } from "./contextcompressionplugin.js";
import { FileParserPlugin$outboundSchema, } from "./fileparserplugin.js";
import { FusionPlugin$outboundSchema, } from "./fusionplugin.js";
import { ImageConfig$outboundSchema, } from "./imageconfig.js";
import { ModerationPlugin$outboundSchema, } from "./moderationplugin.js";
import { ParetoRouterPlugin$outboundSchema, } from "./paretorouterplugin.js";
import { Prediction$outboundSchema, } from "./prediction.js";
import { PromptCacheOptions$outboundSchema, } from "./promptcacheoptions.js";
import { ProviderPreferences$outboundSchema, } from "./providerpreferences.js";
import { ResponseHealingPlugin$outboundSchema, } from "./responsehealingplugin.js";
import { StopServerToolsWhenCondition$outboundSchema, } from "./stopservertoolswhencondition.js";
import { TraceConfig$outboundSchema, } from "./traceconfig.js";
import { WebFetchPlugin$outboundSchema, } from "./webfetchplugin.js";
import { WebSearchPlugin$outboundSchema, } from "./websearchplugin.js";
export const Modality = {
    Text: "text",
    Image: "image",
    Audio: "audio",
};
/**
 * Constrains effort on reasoning for reasoning models
 */
export const ChatRequestEffort = {
    Max: "max",
    Xhigh: "xhigh",
    High: "high",
    Medium: "medium",
    Low: "low",
    Minimal: "minimal",
    None: "none",
};
/**
 * Shorthand for setting reasoning effort. Equivalent to setting reasoning.effort. Cannot be used simultaneously with reasoning.effort if they differ.
 */
export const ChatRequestReasoningEffort = {
    Max: "max",
    Xhigh: "xhigh",
    High: "high",
    Medium: "medium",
    Low: "low",
    Minimal: "minimal",
    None: "none",
};
/**
 * The service tier to use for processing this request.
 */
export const ChatRequestServiceTier = {
    Auto: "auto",
    Default: "default",
    Flex: "flex",
    Priority: "priority",
    Scale: "scale",
};
/** @internal */
export const Modality$outboundSchema = openEnums
    .outboundSchema(Modality);
/** @internal */
export const ChatRequestPlugin$outboundSchema = z.union([
    AutoBetaRouterPlugin$outboundSchema,
    AutoRouterPlugin$outboundSchema,
    ContextCompressionPlugin$outboundSchema,
    FileParserPlugin$outboundSchema,
    FusionPlugin$outboundSchema,
    ModerationPlugin$outboundSchema,
    ParetoRouterPlugin$outboundSchema,
    ResponseHealingPlugin$outboundSchema,
    WebSearchPlugin$outboundSchema,
    WebFetchPlugin$outboundSchema,
]);
export function chatRequestPluginToJSON(chatRequestPlugin) {
    return JSON.stringify(ChatRequestPlugin$outboundSchema.parse(chatRequestPlugin));
}
/** @internal */
export const ChatRequestEffort$outboundSchema = openEnums.outboundSchema(ChatRequestEffort);
/** @internal */
export const ChatRequestReasoning$outboundSchema = z.object({
    effort: z.nullable(ChatRequestEffort$outboundSchema).optional(),
    summary: z.nullable(ChatReasoningSummaryVerbosityEnum$outboundSchema)
        .optional(),
});
export function chatRequestReasoningToJSON(chatRequestReasoning) {
    return JSON.stringify(ChatRequestReasoning$outboundSchema.parse(chatRequestReasoning));
}
/** @internal */
export const ChatRequestReasoningEffort$outboundSchema = openEnums.outboundSchema(ChatRequestReasoningEffort);
/** @internal */
export const ResponseFormat$outboundSchema = z.union([
    ChatFormatGrammarConfig$outboundSchema,
    ChatFormatJsonObjectConfig$outboundSchema,
    ChatFormatJsonSchemaConfig$outboundSchema,
    ChatFormatPythonConfig$outboundSchema,
    ChatFormatTextConfig$outboundSchema,
]);
export function responseFormatToJSON(responseFormat) {
    return JSON.stringify(ResponseFormat$outboundSchema.parse(responseFormat));
}
/** @internal */
export const ChatRequestServiceTier$outboundSchema = openEnums.outboundSchema(ChatRequestServiceTier);
/** @internal */
export const Stop$outboundSchema = z.union([
    z.string(),
    z.array(z.string()),
]);
export function stopToJSON(stop) {
    return JSON.stringify(Stop$outboundSchema.parse(stop));
}
/** @internal */
export const ChatRequest$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    debug: ChatDebugOptions$outboundSchema.optional(),
    frequencyPenalty: z.nullable(z.number()).optional(),
    imageConfig: z.record(z.string(), ImageConfig$outboundSchema).optional(),
    logitBias: z.nullable(z.record(z.string(), z.number())).optional(),
    logprobs: z.nullable(z.boolean()).optional(),
    maxCompletionTokens: z.nullable(z.int()).optional(),
    maxTokens: z.nullable(z.int()).optional(),
    messages: z.array(ChatMessages$outboundSchema),
    metadata: z.record(z.string(), z.string()).optional(),
    minP: z.nullable(z.number()).optional(),
    modalities: z.array(Modality$outboundSchema).optional(),
    model: z.string().optional(),
    models: z.array(z.string()).optional(),
    parallelToolCalls: z.nullable(z.boolean()).optional(),
    plugins: z.array(z.union([
        AutoBetaRouterPlugin$outboundSchema,
        AutoRouterPlugin$outboundSchema,
        ContextCompressionPlugin$outboundSchema,
        FileParserPlugin$outboundSchema,
        FusionPlugin$outboundSchema,
        ModerationPlugin$outboundSchema,
        ParetoRouterPlugin$outboundSchema,
        ResponseHealingPlugin$outboundSchema,
        WebSearchPlugin$outboundSchema,
        WebFetchPlugin$outboundSchema,
    ])).optional(),
    prediction: z.nullable(Prediction$outboundSchema).optional(),
    presencePenalty: z.nullable(z.number()).optional(),
    promptCacheKey: z.nullable(z.string()).optional(),
    promptCacheOptions: z.nullable(PromptCacheOptions$outboundSchema).optional(),
    provider: z.nullable(ProviderPreferences$outboundSchema).optional(),
    reasoning: z.lazy(() => ChatRequestReasoning$outboundSchema).optional(),
    reasoningEffort: z.nullable(ChatRequestReasoningEffort$outboundSchema)
        .optional(),
    repetitionPenalty: z.nullable(z.number()).optional(),
    responseFormat: z.union([
        ChatFormatGrammarConfig$outboundSchema,
        ChatFormatJsonObjectConfig$outboundSchema,
        ChatFormatJsonSchemaConfig$outboundSchema,
        ChatFormatPythonConfig$outboundSchema,
        ChatFormatTextConfig$outboundSchema,
    ]).optional(),
    seed: z.nullable(z.int()).optional(),
    serviceTier: z.nullable(ChatRequestServiceTier$outboundSchema).optional(),
    sessionId: z.string().optional(),
    stop: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    stopServerToolsWhen: z.array(StopServerToolsWhenCondition$outboundSchema)
        .optional(),
    stream: z.boolean().default(false),
    streamOptions: z.nullable(ChatStreamOptions$outboundSchema).optional(),
    temperature: z.nullable(z.number()).optional(),
    toolChoice: ChatToolChoice$outboundSchema.optional(),
    tools: z.array(ChatFunctionTool$outboundSchema).optional(),
    topA: z.nullable(z.number()).optional(),
    topK: z.nullable(z.int()).optional(),
    topLogprobs: z.nullable(z.int()).optional(),
    topP: z.nullable(z.number()).optional(),
    trace: TraceConfig$outboundSchema.optional(),
    user: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
        frequencyPenalty: "frequency_penalty",
        imageConfig: "image_config",
        logitBias: "logit_bias",
        maxCompletionTokens: "max_completion_tokens",
        maxTokens: "max_tokens",
        minP: "min_p",
        parallelToolCalls: "parallel_tool_calls",
        presencePenalty: "presence_penalty",
        promptCacheKey: "prompt_cache_key",
        promptCacheOptions: "prompt_cache_options",
        reasoningEffort: "reasoning_effort",
        repetitionPenalty: "repetition_penalty",
        responseFormat: "response_format",
        serviceTier: "service_tier",
        sessionId: "session_id",
        stopServerToolsWhen: "stop_server_tools_when",
        streamOptions: "stream_options",
        toolChoice: "tool_choice",
        topA: "top_a",
        topK: "top_k",
        topLogprobs: "top_logprobs",
        topP: "top_p",
    });
});
export function chatRequestToJSON(chatRequest) {
    return JSON.stringify(ChatRequest$outboundSchema.parse(chatRequest));
}
