// @openrouter/sdk/esm/models/chatusage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1e677d1ed4fc
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { CostDetails$inboundSchema } from "./costdetails.js";
import { ServerToolUseDetails$inboundSchema, } from "./servertoolusedetails.js";
/** @internal */
export const ChatUsageCompletionTokensDetails$inboundSchema = z.object({
    accepted_prediction_tokens: z.nullable(z.int()).optional(),
    audio_tokens: z.nullable(z.int()).optional(),
    reasoning_tokens: z.nullable(z.int()).optional(),
    rejected_prediction_tokens: z.nullable(z.int()).optional(),
}).transform((v) => {
    return remap$(v, {
        "accepted_prediction_tokens": "acceptedPredictionTokens",
        "audio_tokens": "audioTokens",
        "reasoning_tokens": "reasoningTokens",
        "rejected_prediction_tokens": "rejectedPredictionTokens",
    });
});
export function chatUsageCompletionTokensDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatUsageCompletionTokensDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatUsageCompletionTokensDetails' from JSON`);
}
/** @internal */
export const ChatUsagePromptTokensDetails$inboundSchema = z.object({
    audio_tokens: z.int().optional(),
    cache_write_tokens: z.int().optional(),
    cached_tokens: z.int().optional(),
    video_tokens: z.int().optional(),
}).transform((v) => {
    return remap$(v, {
        "audio_tokens": "audioTokens",
        "cache_write_tokens": "cacheWriteTokens",
        "cached_tokens": "cachedTokens",
        "video_tokens": "videoTokens",
    });
});
export function chatUsagePromptTokensDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatUsagePromptTokensDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatUsagePromptTokensDetails' from JSON`);
}
/** @internal */
export const ChatUsage$inboundSchema = z.object({
    completion_tokens: z.int(),
    completion_tokens_details: z.nullable(z.lazy(() => ChatUsageCompletionTokensDetails$inboundSchema)).optional(),
    cost: z.nullable(z.number()).optional(),
    cost_details: z.nullable(CostDetails$inboundSchema).optional(),
    is_byok: z.boolean().optional(),
    prompt_tokens: z.int(),
    prompt_tokens_details: z.nullable(z.lazy(() => ChatUsagePromptTokensDetails$inboundSchema)).optional(),
    server_tool_use_details: z.nullable(ServerToolUseDetails$inboundSchema)
        .optional(),
    total_tokens: z.int(),
}).transform((v) => {
    return remap$(v, {
        "completion_tokens": "completionTokens",
        "completion_tokens_details": "completionTokensDetails",
        "cost_details": "costDetails",
        "is_byok": "isByok",
        "prompt_tokens": "promptTokens",
        "prompt_tokens_details": "promptTokensDetails",
        "server_tool_use_details": "serverToolUseDetails",
        "total_tokens": "totalTokens",
    });
});
export function chatUsageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatUsage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatUsage' from JSON`);
}
