// @openrouter/sdk/esm/models/chatcontenttext.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 09b5a48d2bab
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatContentCacheControl$inboundSchema, ChatContentCacheControl$outboundSchema, } from "./chatcontentcachecontrol.js";
import { PromptCacheBreakpoint$inboundSchema, PromptCacheBreakpoint$outboundSchema, } from "./promptcachebreakpoint.js";
export const ChatContentTextType = {
    Text: "text",
};
/** @internal */
export const ChatContentTextType$inboundSchema = z.enum(ChatContentTextType);
/** @internal */
export const ChatContentTextType$outboundSchema = ChatContentTextType$inboundSchema;
/** @internal */
export const ChatContentText$inboundSchema = z.object({
    cache_control: ChatContentCacheControl$inboundSchema.optional(),
    prompt_cache_breakpoint: z.nullable(PromptCacheBreakpoint$inboundSchema)
        .optional(),
    text: z.string(),
    type: ChatContentTextType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "cache_control": "cacheControl",
        "prompt_cache_breakpoint": "promptCacheBreakpoint",
    });
});
/** @internal */
export const ChatContentText$outboundSchema = z.object({
    cacheControl: ChatContentCacheControl$outboundSchema.optional(),
    promptCacheBreakpoint: z.nullable(PromptCacheBreakpoint$outboundSchema)
        .optional(),
    text: z.string(),
    type: ChatContentTextType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
        promptCacheBreakpoint: "prompt_cache_breakpoint",
    });
});
export function chatContentTextToJSON(chatContentText) {
    return JSON.stringify(ChatContentText$outboundSchema.parse(chatContentText));
}
export function chatContentTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatContentText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatContentText' from JSON`);
}
