// @openrouter/sdk/esm/models/chatstreamchunk.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 361f940e91e4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ApiErrorType$inboundSchema } from "./apierrortype.js";
import { ChatStreamChoice$inboundSchema, } from "./chatstreamchoice.js";
import { ChatUsage$inboundSchema } from "./chatusage.js";
import { OpenRouterMetadata$inboundSchema, } from "./openroutermetadata.js";
export const ChatStreamChunkObject = {
    ChatCompletionChunk: "chat.completion.chunk",
};
/** @internal */
export const ChatStreamChunkMetadata$inboundSchema = z.object({
    error_type: ApiErrorType$inboundSchema,
    provider_code: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "error_type": "errorType",
        "provider_code": "providerCode",
    });
});
export function chatStreamChunkMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamChunkMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamChunkMetadata' from JSON`);
}
/** @internal */
export const ChatStreamChunkError$inboundSchema = z.object({
    code: z.int(),
    message: z.string(),
    metadata: z.lazy(() => ChatStreamChunkMetadata$inboundSchema).optional(),
});
export function chatStreamChunkErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamChunkError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamChunkError' from JSON`);
}
/** @internal */
export const ChatStreamChunkObject$inboundSchema = z.enum(ChatStreamChunkObject);
/** @internal */
export const ChatStreamChunk$inboundSchema = z.object({
    choices: z.array(ChatStreamChoice$inboundSchema),
    created: z.int(),
    error: z.lazy(() => ChatStreamChunkError$inboundSchema).optional(),
    id: z.string(),
    model: z.string(),
    object: ChatStreamChunkObject$inboundSchema,
    openrouter_metadata: OpenRouterMetadata$inboundSchema.optional(),
    service_tier: z.nullable(z.string()).optional(),
    system_fingerprint: z.string().optional(),
    usage: ChatUsage$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "openrouter_metadata": "openrouterMetadata",
        "service_tier": "serviceTier",
        "system_fingerprint": "systemFingerprint",
    });
});
export function chatStreamChunkFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamChunk$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamChunk' from JSON`);
}
