// @openrouter/sdk/esm/models/codexlocalshelltool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5cd3cc003b0b
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const CodexLocalShellToolType = {
    LocalShell: "local_shell",
};
/** @internal */
export const CodexLocalShellToolType$inboundSchema = z.enum(CodexLocalShellToolType);
/** @internal */
export const CodexLocalShellToolType$outboundSchema = CodexLocalShellToolType$inboundSchema;
/** @internal */
export const CodexLocalShellTool$inboundSchema = z.object({
    type: CodexLocalShellToolType$inboundSchema,
});
/** @internal */
export const CodexLocalShellTool$outboundSchema = z.object({
    type: CodexLocalShellToolType$outboundSchema,
});
export function codexLocalShellToolToJSON(codexLocalShellTool) {
    return JSON.stringify(CodexLocalShellTool$outboundSchema.parse(codexLocalShellTool));
}
export function codexLocalShellToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CodexLocalShellTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CodexLocalShellTool' from JSON`);
}
