// @openrouter/sdk/esm/models/contextcompactionitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8f97e5138d46
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
export const ContextCompactionItemType = {
    ContextCompaction: "context_compaction",
};
/** @internal */
export const ContextCompactionItemType$outboundSchema = z.enum(ContextCompactionItemType);
/** @internal */
export const ContextCompactionItem$outboundSchema = z.object({
    encryptedContent: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    type: ContextCompactionItemType$outboundSchema,
    additionalProperties: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            encryptedContent: "encrypted_content",
            additionalProperties: null,
        }),
    };
});
export function contextCompactionItemToJSON(contextCompactionItem) {
    return JSON.stringify(ContextCompactionItem$outboundSchema.parse(contextCompactionItem));
}
