// @openrouter/sdk/esm/models/contentfilterentry.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 309efd3ad069
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ContentFilterAction$inboundSchema, ContentFilterAction$outboundSchema, } from "./contentfilteraction.js";
/** @internal */
export const ContentFilterEntry$inboundSchema = z.object({
    action: ContentFilterAction$inboundSchema,
    label: z.nullable(z.string()).optional(),
    pattern: z.string(),
});
/** @internal */
export const ContentFilterEntry$outboundSchema = z.object({
    action: ContentFilterAction$outboundSchema,
    label: z.nullable(z.string()).optional(),
    pattern: z.string(),
});
export function contentFilterEntryToJSON(contentFilterEntry) {
    return JSON.stringify(ContentFilterEntry$outboundSchema.parse(contentFilterEntry));
}
export function contentFilterEntryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ContentFilterEntry$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ContentFilterEntry' from JSON`);
}
