// @openrouter/sdk/esm/models/createbyokkeyrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5adb6ef47627
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { BYOKProviderSlug$outboundSchema, } from "./byokproviderslug.js";
/** @internal */
export const CreateBYOKKeyRequest$outboundSchema = z.object({
    allowedModels: z.nullable(z.array(z.string())).optional(),
    allowedUserIds: z.nullable(z.array(z.string())).optional(),
    disabled: z.boolean().optional(),
    isFallback: z.boolean().optional(),
    key: z.string(),
    name: z.nullable(z.string()).optional(),
    provider: BYOKProviderSlug$outboundSchema,
    workspaceId: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        allowedModels: "allowed_models",
        allowedUserIds: "allowed_user_ids",
        isFallback: "is_fallback",
        workspaceId: "workspace_id",
    });
});
export function createBYOKKeyRequestToJSON(createBYOKKeyRequest) {
    return JSON.stringify(CreateBYOKKeyRequest$outboundSchema.parse(createBYOKKeyRequest));
}
