// @openrouter/sdk/esm/models/createguardrailrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: fbf07f1435fd
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ContentFilterBuiltinEntryInput$outboundSchema, } from "./contentfilterbuiltinentryinput.js";
import { ContentFilterEntry$outboundSchema, } from "./contentfilterentry.js";
import { GuardrailInterval$outboundSchema, } from "./guardrailinterval.js";
/** @internal */
export const CreateGuardrailRequest$outboundSchema = z.object({
    allowedModels: z.nullable(z.array(z.string())).optional(),
    allowedProviders: z.nullable(z.array(z.string())).optional(),
    contentFilterBuiltins: z.nullable(z.array(ContentFilterBuiltinEntryInput$outboundSchema)).optional(),
    contentFilters: z.nullable(z.array(ContentFilterEntry$outboundSchema))
        .optional(),
    description: z.nullable(z.string()).optional(),
    enforceZdr: z.nullable(z.boolean()).optional(),
    enforceZdrAnthropic: z.nullable(z.boolean()).optional(),
    enforceZdrGoogle: z.nullable(z.boolean()).optional(),
    enforceZdrOpenai: z.nullable(z.boolean()).optional(),
    enforceZdrOther: z.nullable(z.boolean()).optional(),
    enforceZdrXai: z.nullable(z.boolean()).optional(),
    ignoredModels: z.nullable(z.array(z.string())).optional(),
    ignoredProviders: z.nullable(z.array(z.string())).optional(),
    includeByokInBudgets: z.boolean().optional(),
    limitUsd: z.nullable(z.number()).optional(),
    name: z.string(),
    resetInterval: z.nullable(GuardrailInterval$outboundSchema).optional(),
    workspaceId: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        allowedModels: "allowed_models",
        allowedProviders: "allowed_providers",
        contentFilterBuiltins: "content_filter_builtins",
        contentFilters: "content_filters",
        enforceZdr: "enforce_zdr",
        enforceZdrAnthropic: "enforce_zdr_anthropic",
        enforceZdrGoogle: "enforce_zdr_google",
        enforceZdrOpenai: "enforce_zdr_openai",
        enforceZdrOther: "enforce_zdr_other",
        enforceZdrXai: "enforce_zdr_xai",
        ignoredModels: "ignored_models",
        ignoredProviders: "ignored_providers",
        includeByokInBudgets: "include_byok_in_budgets",
        limitUsd: "limit_usd",
        resetInterval: "reset_interval",
        workspaceId: "workspace_id",
    });
});
export function createGuardrailRequestToJSON(createGuardrailRequest) {
    return JSON.stringify(CreateGuardrailRequest$outboundSchema.parse(createGuardrailRequest));
}
