// @openrouter/sdk/esm/models/createobservabilitydestinationrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6e5b38b96555
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ObservabilityFilterRulesConfigNullable$outboundSchema, } from "./observabilityfilterrulesconfignullable.js";
/**
 * The destination type. Only stable destination types are accepted.
 */
export const CreateObservabilityDestinationRequestType = {
    Arize: "arize",
    Braintrust: "braintrust",
    Clickhouse: "clickhouse",
    Datadog: "datadog",
    Grafana: "grafana",
    Langfuse: "langfuse",
    Langsmith: "langsmith",
    Newrelic: "newrelic",
    Opik: "opik",
    OtelCollector: "otel-collector",
    Posthog: "posthog",
    Ramp: "ramp",
    S3: "s3",
    Sentry: "sentry",
    Snowflake: "snowflake",
    Weave: "weave",
    Webhook: "webhook",
};
/** @internal */
export const CreateObservabilityDestinationRequestType$outboundSchema = openEnums
    .outboundSchema(CreateObservabilityDestinationRequestType);
/** @internal */
export const CreateObservabilityDestinationRequest$outboundSchema = z.object({
    apiKeyHashes: z.nullable(z.array(z.string())).optional(),
    config: z.record(z.string(), z.any()),
    enabled: z.boolean().default(true),
    filterRules: z.nullable(ObservabilityFilterRulesConfigNullable$outboundSchema)
        .optional(),
    name: z.string(),
    privacyMode: z.boolean().default(false),
    samplingRate: z.number().optional(),
    type: CreateObservabilityDestinationRequestType$outboundSchema,
    workspaceId: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        apiKeyHashes: "api_key_hashes",
        filterRules: "filter_rules",
        privacyMode: "privacy_mode",
        samplingRate: "sampling_rate",
        workspaceId: "workspace_id",
    });
});
export function createObservabilityDestinationRequestToJSON(createObservabilityDestinationRequest) {
    return JSON.stringify(CreateObservabilityDestinationRequest$outboundSchema.parse(createObservabilityDestinationRequest));
}
