// @openrouter/sdk/esm/models/createworkspacerequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 729716a42b74
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
/** @internal */
export const CreateWorkspaceRequest$outboundSchema = z.object({
    defaultImageModel: z.nullable(z.string()).optional(),
    defaultProviderSort: z.nullable(z.string()).optional(),
    defaultTextModel: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    ioLoggingApiKeyIds: z.nullable(z.array(z.int())).optional(),
    ioLoggingSamplingRate: z.number().optional(),
    isDataDiscountLoggingEnabled: z.boolean().optional(),
    isObservabilityBroadcastEnabled: z.boolean().optional(),
    isObservabilityIoLoggingEnabled: z.boolean().optional(),
    name: z.string(),
    slug: z.string(),
}).transform((v) => {
    return remap$(v, {
        defaultImageModel: "default_image_model",
        defaultProviderSort: "default_provider_sort",
        defaultTextModel: "default_text_model",
        ioLoggingApiKeyIds: "io_logging_api_key_ids",
        ioLoggingSamplingRate: "io_logging_sampling_rate",
        isDataDiscountLoggingEnabled: "is_data_discount_logging_enabled",
        isObservabilityBroadcastEnabled: "is_observability_broadcast_enabled",
        isObservabilityIoLoggingEnabled: "is_observability_io_logging_enabled",
    });
});
export function createWorkspaceRequestToJSON(createWorkspaceRequest) {
    return JSON.stringify(CreateWorkspaceRequest$outboundSchema.parse(createWorkspaceRequest));
}
