// @openrouter/sdk/esm/models/customtoolcalloutputitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 02aed5cefca5
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { InputFile$outboundSchema, } from "./inputfile.js";
import { InputText$outboundSchema, } from "./inputtext.js";
export const CustomToolCallOutputItemDetail = {
    Auto: "auto",
    High: "high",
    Low: "low",
    Original: "original",
};
export const CustomToolCallOutputItemTypeCustomToolCallOutput = {
    CustomToolCallOutput: "custom_tool_call_output",
};
/** @internal */
export const CustomToolCallOutputItemDetail$outboundSchema = openEnums.outboundSchema(CustomToolCallOutputItemDetail);
/** @internal */
export const CustomToolCallOutputItemOutputInputImage$outboundSchema = z.object({
    detail: CustomToolCallOutputItemDetail$outboundSchema,
    imageUrl: z.nullable(z.string()).optional(),
    type: z.literal("input_image"),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function customToolCallOutputItemOutputInputImageToJSON(customToolCallOutputItemOutputInputImage) {
    return JSON.stringify(CustomToolCallOutputItemOutputInputImage$outboundSchema.parse(customToolCallOutputItemOutputInputImage));
}
/** @internal */
export const CustomToolCallOutputItemOutputUnion1$outboundSchema = z.union([
    InputText$outboundSchema,
    z.lazy(() => CustomToolCallOutputItemOutputInputImage$outboundSchema),
    InputFile$outboundSchema,
]);
export function customToolCallOutputItemOutputUnion1ToJSON(customToolCallOutputItemOutputUnion1) {
    return JSON.stringify(CustomToolCallOutputItemOutputUnion1$outboundSchema.parse(customToolCallOutputItemOutputUnion1));
}
/** @internal */
export const CustomToolCallOutputItemOutputUnion2$outboundSchema = z.union([
    z.string(),
    z.array(z.union([
        InputText$outboundSchema,
        z.lazy(() => CustomToolCallOutputItemOutputInputImage$outboundSchema),
        InputFile$outboundSchema,
    ])),
]);
export function customToolCallOutputItemOutputUnion2ToJSON(customToolCallOutputItemOutputUnion2) {
    return JSON.stringify(CustomToolCallOutputItemOutputUnion2$outboundSchema.parse(customToolCallOutputItemOutputUnion2));
}
/** @internal */
export const CustomToolCallOutputItemTypeCustomToolCallOutput$outboundSchema = z.enum(CustomToolCallOutputItemTypeCustomToolCallOutput);
/** @internal */
export const CustomToolCallOutputItem$outboundSchema = z.object({
    callId: z.string(),
    id: z.string().optional(),
    output: z.union([
        z.string(),
        z.array(z.union([
            InputText$outboundSchema,
            z.lazy(() => CustomToolCallOutputItemOutputInputImage$outboundSchema),
            InputFile$outboundSchema,
        ])),
    ]),
    type: CustomToolCallOutputItemTypeCustomToolCallOutput$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function customToolCallOutputItemToJSON(customToolCallOutputItem) {
    return JSON.stringify(CustomToolCallOutputItem$outboundSchema.parse(customToolCallOutputItem));
}
