// @openrouter/sdk/esm/models/debugevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8afa2111f54c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Event = {
    AdapterRequest: "adapter_request",
    UpstreamHeadersReceived: "upstream_headers_received",
    FirstTokenReceived: "first_token_received",
    UpstreamBodyEnded: "upstream_body_ended",
};
/** @internal */
export const Event$inboundSchema = openEnums
    .inboundSchema(Event);
/** @internal */
export const Timings$inboundSchema = z.object({
    epoch_ms: z.int(),
    event: Event$inboundSchema,
    start_ms: z.int(),
}).transform((v) => {
    return remap$(v, {
        "epoch_ms": "epochMs",
        "start_ms": "startMs",
    });
});
export function timingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Timings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Timings' from JSON`);
}
/** @internal */
export const Debug$inboundSchema = z.object({
    echo_upstream_body: z.record(z.string(), z.any()).optional(),
    timings: z.lazy(() => Timings$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "echo_upstream_body": "echoUpstreamBody",
    });
});
export function debugFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Debug$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Debug' from JSON`);
}
/** @internal */
export const DebugEvent$inboundSchema = z
    .object({
    debug: z.lazy(() => Debug$inboundSchema),
    sequence_number: z.int(),
    type: z.literal("response.debug"),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function debugEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DebugEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DebugEvent' from JSON`);
}
