// @openrouter/sdk/esm/models/filesearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e9b1dc2a101e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { CompoundFilter$inboundSchema, CompoundFilter$outboundSchema, } from "./compoundfilter.js";
export const FiltersType = {
    Eq: "eq",
    Ne: "ne",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
};
export const Ranker = {
    Auto: "auto",
    Default20241115: "default-2024-11-15",
};
export const TypeFileSearch = {
    FileSearch: "file_search",
};
/** @internal */
export const FiltersType$inboundSchema = openEnums.inboundSchema(FiltersType);
/** @internal */
export const FiltersType$outboundSchema = openEnums.outboundSchema(FiltersType);
/** @internal */
export const FileSearchServerToolValue1$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const FileSearchServerToolValue1$outboundSchema = z.union([z.string(), z.number()]);
export function fileSearchServerToolValue1ToJSON(fileSearchServerToolValue1) {
    return JSON.stringify(FileSearchServerToolValue1$outboundSchema.parse(fileSearchServerToolValue1));
}
export function fileSearchServerToolValue1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => FileSearchServerToolValue1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileSearchServerToolValue1' from JSON`);
}
/** @internal */
export const FileSearchServerToolValue2$inboundSchema = z.union([
    z.string(),
    z.number(),
    z.boolean(),
    z.array(z.union([z.string(), z.number()])),
]);
/** @internal */
export const FileSearchServerToolValue2$outboundSchema = z.union([
    z.string(),
    z.number(),
    z.boolean(),
    z.array(z.union([z.string(), z.number()])),
]);
export function fileSearchServerToolValue2ToJSON(fileSearchServerToolValue2) {
    return JSON.stringify(FileSearchServerToolValue2$outboundSchema.parse(fileSearchServerToolValue2));
}
export function fileSearchServerToolValue2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => FileSearchServerToolValue2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileSearchServerToolValue2' from JSON`);
}
/** @internal */
export const Filters$inboundSchema = z.object({
    key: z.string(),
    type: FiltersType$inboundSchema,
    value: z.union([
        z.string(),
        z.number(),
        z.boolean(),
        z.array(z.union([z.string(), z.number()])),
    ]),
});
/** @internal */
export const Filters$outboundSchema = z
    .object({
    key: z.string(),
    type: FiltersType$outboundSchema,
    value: z.union([
        z.string(),
        z.number(),
        z.boolean(),
        z.array(z.union([z.string(), z.number()])),
    ]),
});
export function filtersToJSON(filters) {
    return JSON.stringify(Filters$outboundSchema.parse(filters));
}
export function filtersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Filters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Filters' from JSON`);
}
/** @internal */
export const FiltersUnion$inboundSchema = z
    .union([z.lazy(() => Filters$inboundSchema), CompoundFilter$inboundSchema]);
/** @internal */
export const FiltersUnion$outboundSchema = z.union([
    z.lazy(() => Filters$outboundSchema),
    CompoundFilter$outboundSchema,
]);
export function filtersUnionToJSON(filtersUnion) {
    return JSON.stringify(FiltersUnion$outboundSchema.parse(filtersUnion));
}
export function filtersUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FiltersUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FiltersUnion' from JSON`);
}
/** @internal */
export const Ranker$inboundSchema = openEnums
    .inboundSchema(Ranker);
/** @internal */
export const Ranker$outboundSchema = openEnums
    .outboundSchema(Ranker);
/** @internal */
export const RankingOptions$inboundSchema = z.object({
    ranker: Ranker$inboundSchema.optional(),
    score_threshold: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        "score_threshold": "scoreThreshold",
    });
});
/** @internal */
export const RankingOptions$outboundSchema = z.object({
    ranker: Ranker$outboundSchema.optional(),
    scoreThreshold: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        scoreThreshold: "score_threshold",
    });
});
export function rankingOptionsToJSON(rankingOptions) {
    return JSON.stringify(RankingOptions$outboundSchema.parse(rankingOptions));
}
export function rankingOptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RankingOptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RankingOptions' from JSON`);
}
/** @internal */
export const TypeFileSearch$inboundSchema = z
    .enum(TypeFileSearch);
/** @internal */
export const TypeFileSearch$outboundSchema = TypeFileSearch$inboundSchema;
/** @internal */
export const FileSearchServerTool$inboundSchema = z.object({
    filters: z.nullable(z.union([
        z.lazy(() => Filters$inboundSchema),
        CompoundFilter$inboundSchema,
    ])).optional(),
    max_num_results: z.int().optional(),
    ranking_options: z.lazy(() => RankingOptions$inboundSchema).optional(),
    type: TypeFileSearch$inboundSchema,
    vector_store_ids: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        "max_num_results": "maxNumResults",
        "ranking_options": "rankingOptions",
        "vector_store_ids": "vectorStoreIds",
    });
});
/** @internal */
export const FileSearchServerTool$outboundSchema = z.object({
    filters: z.nullable(z.union([
        z.lazy(() => Filters$outboundSchema),
        CompoundFilter$outboundSchema,
    ])).optional(),
    maxNumResults: z.int().optional(),
    rankingOptions: z.lazy(() => RankingOptions$outboundSchema).optional(),
    type: TypeFileSearch$outboundSchema,
    vectorStoreIds: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        maxNumResults: "max_num_results",
        rankingOptions: "ranking_options",
        vectorStoreIds: "vector_store_ids",
    });
});
export function fileSearchServerToolToJSON(fileSearchServerTool) {
    return JSON.stringify(FileSearchServerTool$outboundSchema.parse(fileSearchServerTool));
}
export function fileSearchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FileSearchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileSearchServerTool' from JSON`);
}
