// @openrouter/sdk/esm/models/filemetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7a5c3a86d7a2
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const FileMetadataType = {
    File: "file",
};
/** @internal */
export const FileMetadataType$inboundSchema = z.enum(FileMetadataType);
/** @internal */
export const FileMetadata$inboundSchema = z
    .object({
    created_at: z.string(),
    downloadable: z.boolean(),
    filename: z.string(),
    id: z.string(),
    mime_type: z.string(),
    size_bytes: z.int(),
    type: FileMetadataType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "mime_type": "mimeType",
        "size_bytes": "sizeBytes",
    });
});
export function fileMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FileMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileMetadata' from JSON`);
}
