// @openrouter/sdk/esm/models/fusionplugin.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f2f6926f21f9
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
/**
 * A curated OpenRouter fusion preset (slugs follow `<task>-<tier>`, e.g. `general-high`). Expands server-side into the preset's analysis_models panel and judge model, so callers never name individual models. Explicitly provided `analysis_models` / `model` take precedence.
 */
export const PresetEnum = {
    GeneralHigh: "general-high",
    GeneralBudget: "general-budget",
    GeneralFast: "general-fast",
};
/** @internal */
export const PresetEnum$outboundSchema = openEnums.outboundSchema(PresetEnum);
/** @internal */
export const Parameters2$outboundSchema = z.union([z.string(), z.number(), z.boolean()]);
export function parameters2ToJSON(parameters2) {
    return JSON.stringify(Parameters2$outboundSchema.parse(parameters2));
}
/** @internal */
export const Parameters1$outboundSchema = z.union([z.string(), z.number(), z.boolean()]);
export function parameters1ToJSON(parameters1) {
    return JSON.stringify(Parameters1$outboundSchema.parse(parameters1));
}
/** @internal */
export const Parameters3$outboundSchema = z.union([
    z.string(),
    z.number(),
    z.boolean(),
    z.array(z.nullable(z.union([z.string(), z.number(), z.boolean()]))),
    z.record(z.string(), z.nullable(z.union([z.string(), z.number(), z.boolean()]))),
]);
export function parameters3ToJSON(parameters3) {
    return JSON.stringify(Parameters3$outboundSchema.parse(parameters3));
}
/** @internal */
export const FusionPluginTool$outboundSchema = z.object({
    parameters: z.record(z.string(), z.nullable(z.union([
        z.string(),
        z.number(),
        z.boolean(),
        z.array(z.nullable(z.union([z.string(), z.number(), z.boolean()]))),
        z.record(z.string(), z.nullable(z.union([z.string(), z.number(), z.boolean()]))),
    ]))).optional(),
    type: z.string(),
});
export function fusionPluginToolToJSON(fusionPluginTool) {
    return JSON.stringify(FusionPluginTool$outboundSchema.parse(fusionPluginTool));
}
/** @internal */
export const FusionPlugin$outboundSchema = z.object({
    analysisModels: z.array(z.string()).optional(),
    enabled: z.boolean().optional(),
    id: z.literal("fusion"),
    maxToolCalls: z.int().optional(),
    model: z.string().optional(),
    preset: PresetEnum$outboundSchema.optional(),
    tools: z.array(z.lazy(() => FusionPluginTool$outboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        analysisModels: "analysis_models",
        maxToolCalls: "max_tool_calls",
    });
});
export function fusionPluginToJSON(fusionPlugin) {
    return JSON.stringify(FusionPlugin$outboundSchema.parse(fusionPlugin));
}
