// @openrouter/sdk/esm/models/fusioncallanalysiscompletedevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0df2084e1a56
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { FusionAnalysisResult$inboundSchema, } from "./fusionanalysisresult.js";
/** @internal */
export const FusionCallAnalysisCompletedEvent$inboundSchema = z.object({
    analysis: FusionAnalysisResult$inboundSchema,
    item_id: z.string(),
    output_index: z.int(),
    sequence_number: z.int(),
    type: z.literal("response.fusion_call.analysis.completed"),
}).transform((v) => {
    return remap$(v, {
        "item_id": "itemId",
        "output_index": "outputIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function fusionCallAnalysisCompletedEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FusionCallAnalysisCompletedEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FusionCallAnalysisCompletedEvent' from JSON`);
}
