// @openrouter/sdk/esm/models/fusionanalysisresult.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ffc6f5c74b6c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const Stance$inboundSchema = z.object({
    model: z.string(),
    stance: z.string(),
});
/** @internal */
export const Stance$outboundSchema = z
    .object({
    model: z.string(),
    stance: z.string(),
});
export function stanceToJSON(stance) {
    return JSON.stringify(Stance$outboundSchema.parse(stance));
}
export function stanceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Stance$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Stance' from JSON`);
}
/** @internal */
export const Contradiction$inboundSchema = z
    .object({
    stances: z.array(z.lazy(() => Stance$inboundSchema)),
    topic: z.string(),
});
/** @internal */
export const Contradiction$outboundSchema = z.object({
    stances: z.array(z.lazy(() => Stance$outboundSchema)),
    topic: z.string(),
});
export function contradictionToJSON(contradiction) {
    return JSON.stringify(Contradiction$outboundSchema.parse(contradiction));
}
export function contradictionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Contradiction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Contradiction' from JSON`);
}
/** @internal */
export const PartialCoverage$inboundSchema = z.object({
    models: z.array(z.string()),
    point: z.string(),
});
/** @internal */
export const PartialCoverage$outboundSchema = z.object({
    models: z.array(z.string()),
    point: z.string(),
});
export function partialCoverageToJSON(partialCoverage) {
    return JSON.stringify(PartialCoverage$outboundSchema.parse(partialCoverage));
}
export function partialCoverageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PartialCoverage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PartialCoverage' from JSON`);
}
/** @internal */
export const UniqueInsight$inboundSchema = z
    .object({
    insight: z.string(),
    model: z.string(),
});
/** @internal */
export const UniqueInsight$outboundSchema = z.object({
    insight: z.string(),
    model: z.string(),
});
export function uniqueInsightToJSON(uniqueInsight) {
    return JSON.stringify(UniqueInsight$outboundSchema.parse(uniqueInsight));
}
export function uniqueInsightFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UniqueInsight$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UniqueInsight' from JSON`);
}
/** @internal */
export const FusionAnalysisResult$inboundSchema = z.object({
    blind_spots: z.array(z.string()),
    consensus: z.array(z.string()),
    contradictions: z.array(z.lazy(() => Contradiction$inboundSchema)),
    partial_coverage: z.array(z.lazy(() => PartialCoverage$inboundSchema)),
    unique_insights: z.array(z.lazy(() => UniqueInsight$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "blind_spots": "blindSpots",
        "partial_coverage": "partialCoverage",
        "unique_insights": "uniqueInsights",
    });
});
/** @internal */
export const FusionAnalysisResult$outboundSchema = z.object({
    blindSpots: z.array(z.string()),
    consensus: z.array(z.string()),
    contradictions: z.array(z.lazy(() => Contradiction$outboundSchema)),
    partialCoverage: z.array(z.lazy(() => PartialCoverage$outboundSchema)),
    uniqueInsights: z.array(z.lazy(() => UniqueInsight$outboundSchema)),
}).transform((v) => {
    return remap$(v, {
        blindSpots: "blind_spots",
        partialCoverage: "partial_coverage",
        uniqueInsights: "unique_insights",
    });
});
export function fusionAnalysisResultToJSON(fusionAnalysisResult) {
    return JSON.stringify(FusionAnalysisResult$outboundSchema.parse(fusionAnalysisResult));
}
export function fusionAnalysisResultFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FusionAnalysisResult$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FusionAnalysisResult' from JSON`);
}
