// @openrouter/sdk/esm/models/generationcontentdata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f55aeaffdbd5
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const Input2$inboundSchema = z.object({
    messages: z.array(z.any()),
});
export function input2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Input2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Input2' from JSON`);
}
/** @internal */
export const Input1$inboundSchema = z.object({
    prompt: z.string(),
});
export function input1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Input1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Input1' from JSON`);
}
/** @internal */
export const InputUnion$inboundSchema = z.union([z.lazy(() => Input1$inboundSchema), z.lazy(() => Input2$inboundSchema)]);
export function inputUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InputUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InputUnion' from JSON`);
}
/** @internal */
export const GenerationContentDataOutput$inboundSchema = z.object({
    completion: z.nullable(z.string()),
    reasoning: z.nullable(z.string()),
});
export function generationContentDataOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GenerationContentDataOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenerationContentDataOutput' from JSON`);
}
/** @internal */
export const GenerationContentData$inboundSchema = z.object({
    input: z.union([
        z.lazy(() => Input1$inboundSchema),
        z.lazy(() => Input2$inboundSchema),
    ]),
    output: z.lazy(() => GenerationContentDataOutput$inboundSchema),
});
export function generationContentDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GenerationContentData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenerationContentData' from JSON`);
}
