// @openrouter/sdk/esm/models/generationresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f9109ae06502
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ProviderResponse$inboundSchema, } from "./providerresponse.js";
/**
 * Type of API used for the generation
 */
export const ApiType = {
    Completions: "completions",
    Embeddings: "embeddings",
    Rerank: "rerank",
    Tts: "tts",
    Stt: "stt",
    Video: "video",
    Image: "image",
};
/**
 * The data region this generation was routed through. 'europe' for EU-routed requests, 'global' otherwise.
 */
export const DataRegion = {
    Global: "global",
    Europe: "europe",
};
/** @internal */
export const ApiType$inboundSchema = openEnums
    .inboundSchema(ApiType);
/** @internal */
export const DataRegion$inboundSchema = openEnums.inboundSchema(DataRegion);
/** @internal */
export const GenerationResponseData$inboundSchema = z.object({
    api_type: z.nullable(ApiType$inboundSchema),
    app_id: z.nullable(z.int()),
    cache_discount: z.nullable(z.number()),
    cancelled: z.nullable(z.boolean()),
    created_at: z.string(),
    data_region: DataRegion$inboundSchema,
    external_user: z.nullable(z.string()),
    finish_reason: z.nullable(z.string()),
    generation_time: z.nullable(z.number()),
    http_referer: z.nullable(z.string()),
    id: z.string(),
    is_byok: z.boolean(),
    latency: z.nullable(z.number()),
    model: z.string(),
    moderation_latency: z.nullable(z.number()),
    native_finish_reason: z.nullable(z.string()),
    native_tokens_cached: z.nullable(z.int()),
    native_tokens_completion: z.nullable(z.int()),
    native_tokens_completion_images: z.nullable(z.int()),
    native_tokens_prompt: z.nullable(z.int()),
    native_tokens_reasoning: z.nullable(z.int()),
    num_fetches: z.nullable(z.int()),
    num_input_audio_prompt: z.nullable(z.int()),
    num_media_completion: z.nullable(z.int()),
    num_media_prompt: z.nullable(z.int()),
    num_search_results: z.nullable(z.int()),
    origin: z.string(),
    preset_id: z.nullable(z.string()),
    provider_name: z.nullable(z.string()),
    provider_responses: z.nullable(z.array(ProviderResponse$inboundSchema)),
    request_id: z.nullable(z.string()).optional(),
    response_cache_source_id: z.nullable(z.string()).optional(),
    router: z.nullable(z.string()),
    service_tier: z.nullable(z.string()),
    session_id: z.nullable(z.string()).optional(),
    streamed: z.nullable(z.boolean()),
    tokens_completion: z.nullable(z.int()),
    tokens_prompt: z.nullable(z.int()),
    total_cost: z.number(),
    upstream_id: z.nullable(z.string()),
    upstream_inference_cost: z.nullable(z.number()),
    usage: z.number(),
    user_agent: z.nullable(z.string()),
    web_search_engine: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "api_type": "apiType",
        "app_id": "appId",
        "cache_discount": "cacheDiscount",
        "created_at": "createdAt",
        "data_region": "dataRegion",
        "external_user": "externalUser",
        "finish_reason": "finishReason",
        "generation_time": "generationTime",
        "http_referer": "httpReferer",
        "is_byok": "isByok",
        "moderation_latency": "moderationLatency",
        "native_finish_reason": "nativeFinishReason",
        "native_tokens_cached": "nativeTokensCached",
        "native_tokens_completion": "nativeTokensCompletion",
        "native_tokens_completion_images": "nativeTokensCompletionImages",
        "native_tokens_prompt": "nativeTokensPrompt",
        "native_tokens_reasoning": "nativeTokensReasoning",
        "num_fetches": "numFetches",
        "num_input_audio_prompt": "numInputAudioPrompt",
        "num_media_completion": "numMediaCompletion",
        "num_media_prompt": "numMediaPrompt",
        "num_search_results": "numSearchResults",
        "preset_id": "presetId",
        "provider_name": "providerName",
        "provider_responses": "providerResponses",
        "request_id": "requestId",
        "response_cache_source_id": "responseCacheSourceId",
        "service_tier": "serviceTier",
        "session_id": "sessionId",
        "tokens_completion": "tokensCompletion",
        "tokens_prompt": "tokensPrompt",
        "total_cost": "totalCost",
        "upstream_id": "upstreamId",
        "upstream_inference_cost": "upstreamInferenceCost",
        "user_agent": "userAgent",
        "web_search_engine": "webSearchEngine",
    });
});
export function generationResponseDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GenerationResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenerationResponseData' from JSON`);
}
/** @internal */
export const GenerationResponse$inboundSchema = z.object({
    data: z.lazy(() => GenerationResponseData$inboundSchema),
});
export function generationResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GenerationResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenerationResponse' from JSON`);
}
