// @openrouter/sdk/esm/models/guardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 240a4b8d7a7d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ContentFilterBuiltinEntry$inboundSchema, } from "./contentfilterbuiltinentry.js";
import { ContentFilterEntry$inboundSchema, } from "./contentfilterentry.js";
import { GuardrailInterval$inboundSchema, } from "./guardrailinterval.js";
/** @internal */
export const Guardrail$inboundSchema = z.object({
    allowed_models: z.nullable(z.array(z.string())).optional(),
    allowed_providers: z.nullable(z.array(z.string())).optional(),
    content_filter_builtins: z.nullable(z.array(ContentFilterBuiltinEntry$inboundSchema)).optional(),
    content_filters: z.nullable(z.array(ContentFilterEntry$inboundSchema))
        .optional(),
    created_at: z.string(),
    description: z.nullable(z.string()).optional(),
    enforce_zdr: z.nullable(z.boolean()).optional(),
    enforce_zdr_anthropic: z.nullable(z.boolean()).optional(),
    enforce_zdr_google: z.nullable(z.boolean()).optional(),
    enforce_zdr_openai: z.nullable(z.boolean()).optional(),
    enforce_zdr_other: z.nullable(z.boolean()).optional(),
    enforce_zdr_xai: z.nullable(z.boolean()).optional(),
    id: z.string(),
    ignored_models: z.nullable(z.array(z.string())).optional(),
    ignored_providers: z.nullable(z.array(z.string())).optional(),
    include_byok_in_budgets: z.boolean(),
    limit_usd: z.nullable(z.number()).optional(),
    name: z.string(),
    reset_interval: z.nullable(GuardrailInterval$inboundSchema).optional(),
    updated_at: z.nullable(z.string()).optional(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "allowed_models": "allowedModels",
        "allowed_providers": "allowedProviders",
        "content_filter_builtins": "contentFilterBuiltins",
        "content_filters": "contentFilters",
        "created_at": "createdAt",
        "enforce_zdr": "enforceZdr",
        "enforce_zdr_anthropic": "enforceZdrAnthropic",
        "enforce_zdr_google": "enforceZdrGoogle",
        "enforce_zdr_openai": "enforceZdrOpenai",
        "enforce_zdr_other": "enforceZdrOther",
        "enforce_zdr_xai": "enforceZdrXai",
        "ignored_models": "ignoredModels",
        "ignored_providers": "ignoredProviders",
        "include_byok_in_budgets": "includeByokInBudgets",
        "limit_usd": "limitUsd",
        "reset_interval": "resetInterval",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function guardrailFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Guardrail$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Guardrail' from JSON`);
}
