// @openrouter/sdk/esm/models/imagegenerationusage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4f128ace6b39
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { AnthropicCacheCreation$inboundSchema, } from "./anthropiccachecreation.js";
import { AnthropicSpeed$inboundSchema, } from "./anthropicspeed.js";
import { AnthropicUsageIteration$inboundSchema, } from "./anthropicusageiteration.js";
import { CostDetails$inboundSchema } from "./costdetails.js";
/** @internal */
export const ImageGenerationUsageCompletionTokensDetails$inboundSchema = z.object({
    audio_tokens: z.nullable(z.int()).optional(),
    image_tokens: z.nullable(z.int()).optional(),
    reasoning_tokens: z.nullable(z.int()).optional(),
}).transform((v) => {
    return remap$(v, {
        "audio_tokens": "audioTokens",
        "image_tokens": "imageTokens",
        "reasoning_tokens": "reasoningTokens",
    });
});
export function imageGenerationUsageCompletionTokensDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenerationUsageCompletionTokensDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenerationUsageCompletionTokensDetails' from JSON`);
}
/** @internal */
export const ImageGenerationUsagePromptTokensDetails$inboundSchema = z.object({
    audio_tokens: z.nullable(z.int()).optional(),
    cache_write_tokens: z.nullable(z.int()).optional(),
    cached_tokens: z.nullable(z.int()).optional(),
    file_tokens: z.nullable(z.int()).optional(),
    video_tokens: z.nullable(z.int()).optional(),
}).transform((v) => {
    return remap$(v, {
        "audio_tokens": "audioTokens",
        "cache_write_tokens": "cacheWriteTokens",
        "cached_tokens": "cachedTokens",
        "file_tokens": "fileTokens",
        "video_tokens": "videoTokens",
    });
});
export function imageGenerationUsagePromptTokensDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenerationUsagePromptTokensDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenerationUsagePromptTokensDetails' from JSON`);
}
/** @internal */
export const ServerToolUse$inboundSchema = z
    .object({
    tool_calls_executed: z.nullable(z.int()).optional(),
    tool_calls_requested: z.nullable(z.int()).optional(),
    web_search_requests: z.nullable(z.int()).optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_calls_executed": "toolCallsExecuted",
        "tool_calls_requested": "toolCallsRequested",
        "web_search_requests": "webSearchRequests",
    });
});
export function serverToolUseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ServerToolUse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ServerToolUse' from JSON`);
}
/** @internal */
export const ImageGenerationUsage$inboundSchema = z.object({
    cache_creation: z.nullable(AnthropicCacheCreation$inboundSchema).optional(),
    completion_tokens: z.int(),
    completion_tokens_details: z.nullable(z.lazy(() => ImageGenerationUsageCompletionTokensDetails$inboundSchema)).optional(),
    cost: z.nullable(z.number()).optional(),
    cost_details: z.nullable(CostDetails$inboundSchema).optional(),
    is_byok: z.boolean().optional(),
    iterations: z.nullable(z.array(AnthropicUsageIteration$inboundSchema))
        .optional(),
    prompt_tokens: z.int(),
    prompt_tokens_details: z.nullable(z.lazy(() => ImageGenerationUsagePromptTokensDetails$inboundSchema)).optional(),
    server_tool_use: z.nullable(z.lazy(() => ServerToolUse$inboundSchema))
        .optional(),
    service_tier: z.nullable(z.string()).optional(),
    speed: z.nullable(AnthropicSpeed$inboundSchema).optional(),
    total_tokens: z.int(),
}).transform((v) => {
    return remap$(v, {
        "cache_creation": "cacheCreation",
        "completion_tokens": "completionTokens",
        "completion_tokens_details": "completionTokensDetails",
        "cost_details": "costDetails",
        "is_byok": "isByok",
        "prompt_tokens": "promptTokens",
        "prompt_tokens_details": "promptTokensDetails",
        "server_tool_use": "serverToolUse",
        "service_tier": "serviceTier",
        "total_tokens": "totalTokens",
    });
});
export function imageGenerationUsageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenerationUsage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenerationUsage' from JSON`);
}
