// @openrouter/sdk/esm/models/imagegenstreamerrorevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3ba1d6d94b96
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const ImageGenStreamErrorEventError$inboundSchema = z.object({
    code: z.nullable(z.string()).optional(),
    message: z.string(),
    param: z.nullable(z.string()).optional(),
    type: z.nullable(z.string()).optional(),
});
export function imageGenStreamErrorEventErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenStreamErrorEventError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenStreamErrorEventError' from JSON`);
}
/** @internal */
export const ImageGenStreamErrorEvent$inboundSchema = z.object({
    error: z.lazy(() => ImageGenStreamErrorEventError$inboundSchema),
    type: z.literal("error"),
});
export function imageGenStreamErrorEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenStreamErrorEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenStreamErrorEvent' from JSON`);
}
