// @openrouter/sdk/esm/models/imagegenerationrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ea1ffc0d2470
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ContentPartImage$outboundSchema, } from "./contentpartimage.js";
import { ImageGenerationProviderPreferences$outboundSchema, } from "./imagegenerationproviderpreferences.js";
/**
 * Normalized aspect ratio of the generated image. Providers clamp to their supported subset.
 */
export const ImageGenerationRequestAspectRatio = {
    Eleven: "1:1",
    Twelve: "1:2",
    Fourteen: "1:4",
    Eighteen: "1:8",
    TwentyOne: "2:1",
    TwentyThree: "2:3",
    ThirtyTwo: "3:2",
    ThirtyFour: "3:4",
    FortyOne: "4:1",
    FortyThree: "4:3",
    FortyFive: "4:5",
    FiftyFour: "5:4",
    EightyOne: "8:1",
    NineHundredAndSixteen: "9:16",
    OneHundredAndSixtyNine: "16:9",
    NineHundredAndNineteenDot5: "9:19.5",
    NineteenDot59: "19.5:9",
    NineHundredAndTwenty: "9:20",
    TwoHundredAndNine: "20:9",
    NineHundredAndTwentyOne: "9:21",
    TwoHundredAndNineteen: "21:9",
    Auto: "auto",
};
/**
 * Background treatment. `transparent` requires an output_format that supports alpha (png or webp).
 */
export const ImageGenerationRequestBackground = {
    Auto: "auto",
    Transparent: "transparent",
    Opaque: "opaque",
};
/**
 * Encoding of the returned image bytes. Most models produce raster formats (png, jpeg, webp). SVG is supported by vectorization models (e.g. Quiver) — the SVG markup is UTF-8 base64-encoded in `b64_json`.
 */
export const ImageGenerationRequestOutputFormat = {
    Png: "png",
    Jpeg: "jpeg",
    Webp: "webp",
    Svg: "svg",
};
/**
 * Rendering quality. Providers without a quality knob ignore this.
 */
export const ImageGenerationRequestQuality = {
    Auto: "auto",
    Low: "low",
    Medium: "medium",
    High: "high",
};
/**
 * Normalized resolution tier of the generated image. Concrete pixel dimensions are derived per-provider.
 */
export const ImageGenerationRequestResolution = {
    FiveHundredAndTwelve: "512",
    OneK: "1K",
    TwoK: "2K",
    FourK: "4K",
};
/** @internal */
export const ImageGenerationRequestAspectRatio$outboundSchema = openEnums.outboundSchema(ImageGenerationRequestAspectRatio);
/** @internal */
export const ImageGenerationRequestBackground$outboundSchema = openEnums.outboundSchema(ImageGenerationRequestBackground);
/** @internal */
export const ImageGenerationRequestOutputFormat$outboundSchema = openEnums.outboundSchema(ImageGenerationRequestOutputFormat);
/** @internal */
export const ImageGenerationRequestQuality$outboundSchema = openEnums.outboundSchema(ImageGenerationRequestQuality);
/** @internal */
export const ImageGenerationRequestResolution$outboundSchema = openEnums.outboundSchema(ImageGenerationRequestResolution);
/** @internal */
export const ImageGenerationRequest$outboundSchema = z.object({
    aspectRatio: ImageGenerationRequestAspectRatio$outboundSchema.optional(),
    background: ImageGenerationRequestBackground$outboundSchema.optional(),
    inputReferences: z.array(ContentPartImage$outboundSchema).optional(),
    model: z.string(),
    n: z.int().optional(),
    outputCompression: z.int().optional(),
    outputFormat: ImageGenerationRequestOutputFormat$outboundSchema.optional(),
    prompt: z.string(),
    provider: ImageGenerationProviderPreferences$outboundSchema.optional(),
    quality: ImageGenerationRequestQuality$outboundSchema.optional(),
    resolution: ImageGenerationRequestResolution$outboundSchema.optional(),
    seed: z.int().optional(),
    size: z.string().optional(),
    stream: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        aspectRatio: "aspect_ratio",
        inputReferences: "input_references",
        outputCompression: "output_compression",
        outputFormat: "output_format",
    });
});
export function imageGenerationRequestToJSON(imageGenerationRequest) {
    return JSON.stringify(ImageGenerationRequest$outboundSchema.parse(imageGenerationRequest));
}
