// @openrouter/sdk/esm/models/imagegenerationservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1efbad50f534
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ImageGenerationServerToolBackground = {
    Transparent: "transparent",
    Opaque: "opaque",
    Auto: "auto",
};
export const InputFidelity = {
    High: "high",
    Low: "low",
};
export const Moderation = {
    Auto: "auto",
    Low: "low",
};
export const ImageGenerationServerToolOutputFormat = {
    Png: "png",
    Webp: "webp",
    Jpeg: "jpeg",
};
export const ImageGenerationServerToolQuality = {
    Low: "low",
    Medium: "medium",
    High: "high",
    Auto: "auto",
};
export const ImageGenerationServerToolType = {
    ImageGeneration: "image_generation",
};
/** @internal */
export const ImageGenerationServerToolBackground$inboundSchema = openEnums.inboundSchema(ImageGenerationServerToolBackground);
/** @internal */
export const ImageGenerationServerToolBackground$outboundSchema = openEnums.outboundSchema(ImageGenerationServerToolBackground);
/** @internal */
export const InputFidelity$inboundSchema = openEnums.inboundSchema(InputFidelity);
/** @internal */
export const InputFidelity$outboundSchema = openEnums.outboundSchema(InputFidelity);
/** @internal */
export const InputImageMask$inboundSchema = z.object({
    file_id: z.string().optional(),
    image_url: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "file_id": "fileId",
        "image_url": "imageUrl",
    });
});
/** @internal */
export const InputImageMask$outboundSchema = z.object({
    fileId: z.string().optional(),
    imageUrl: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        fileId: "file_id",
        imageUrl: "image_url",
    });
});
export function inputImageMaskToJSON(inputImageMask) {
    return JSON.stringify(InputImageMask$outboundSchema.parse(inputImageMask));
}
export function inputImageMaskFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InputImageMask$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InputImageMask' from JSON`);
}
/** @internal */
export const Moderation$inboundSchema = openEnums.inboundSchema(Moderation);
/** @internal */
export const Moderation$outboundSchema = openEnums.outboundSchema(Moderation);
/** @internal */
export const ImageGenerationServerToolOutputFormat$inboundSchema = openEnums.inboundSchema(ImageGenerationServerToolOutputFormat);
/** @internal */
export const ImageGenerationServerToolOutputFormat$outboundSchema = openEnums.outboundSchema(ImageGenerationServerToolOutputFormat);
/** @internal */
export const ImageGenerationServerToolQuality$inboundSchema = openEnums.inboundSchema(ImageGenerationServerToolQuality);
/** @internal */
export const ImageGenerationServerToolQuality$outboundSchema = openEnums.outboundSchema(ImageGenerationServerToolQuality);
/** @internal */
export const ImageGenerationServerToolType$inboundSchema = z.enum(ImageGenerationServerToolType);
/** @internal */
export const ImageGenerationServerToolType$outboundSchema = ImageGenerationServerToolType$inboundSchema;
/** @internal */
export const ImageGenerationServerTool$inboundSchema = z.object({
    background: ImageGenerationServerToolBackground$inboundSchema.optional(),
    input_fidelity: z.nullable(InputFidelity$inboundSchema).optional(),
    input_image_mask: z.lazy(() => InputImageMask$inboundSchema).optional(),
    model: z.string().optional(),
    moderation: Moderation$inboundSchema.optional(),
    output_compression: z.int().optional(),
    output_format: ImageGenerationServerToolOutputFormat$inboundSchema.optional(),
    partial_images: z.int().optional(),
    quality: ImageGenerationServerToolQuality$inboundSchema.optional(),
    size: z.string().optional(),
    type: ImageGenerationServerToolType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "input_fidelity": "inputFidelity",
        "input_image_mask": "inputImageMask",
        "output_compression": "outputCompression",
        "output_format": "outputFormat",
        "partial_images": "partialImages",
    });
});
/** @internal */
export const ImageGenerationServerTool$outboundSchema = z.object({
    background: ImageGenerationServerToolBackground$outboundSchema.optional(),
    inputFidelity: z.nullable(InputFidelity$outboundSchema).optional(),
    inputImageMask: z.lazy(() => InputImageMask$outboundSchema).optional(),
    model: z.string().optional(),
    moderation: Moderation$outboundSchema.optional(),
    outputCompression: z.int().optional(),
    outputFormat: ImageGenerationServerToolOutputFormat$outboundSchema.optional(),
    partialImages: z.int().optional(),
    quality: ImageGenerationServerToolQuality$outboundSchema.optional(),
    size: z.string().optional(),
    type: ImageGenerationServerToolType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        inputFidelity: "input_fidelity",
        inputImageMask: "input_image_mask",
        outputCompression: "output_compression",
        outputFormat: "output_format",
        partialImages: "partial_images",
    });
});
export function imageGenerationServerToolToJSON(imageGenerationServerTool) {
    return JSON.stringify(ImageGenerationServerTool$outboundSchema.parse(imageGenerationServerTool));
}
export function imageGenerationServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenerationServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenerationServerTool' from JSON`);
}
