// @openrouter/sdk/esm/models/imagestreamingresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 845661ccae1f
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { ImageGenCompletedEvent$inboundSchema, } from "./imagegencompletedevent.js";
import { ImageGenPartialImageEvent$inboundSchema, } from "./imagegenpartialimageevent.js";
import { ImageGenStreamErrorEvent$inboundSchema, } from "./imagegenstreamerrorevent.js";
import { ImageGenTextChunkEvent$inboundSchema, } from "./imagegentextchunkevent.js";
/** @internal */
export const ImageStreamingResponseData$inboundSchema = discriminatedUnion("type", {
    ["image_generation.partial_image"]: ImageGenPartialImageEvent$inboundSchema,
    ["image_generation.text_chunk"]: ImageGenTextChunkEvent$inboundSchema,
    ["image_generation.completed"]: ImageGenCompletedEvent$inboundSchema,
    error: ImageGenStreamErrorEvent$inboundSchema,
});
export function imageStreamingResponseDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageStreamingResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageStreamingResponseData' from JSON`);
}
/** @internal */
export const ImageStreamingResponse$inboundSchema = z.object({
    data: z.unknown().transform((v, ctx) => {
        if (typeof v !== "string")
            return v;
        try {
            return JSON.parse(v);
        }
        catch (err) {
            ctx.addIssue({
                input: v,
                code: "custom",
                message: `malformed json: ${err}`,
            });
            return z.NEVER;
        }
    }).pipe(discriminatedUnion("type", {
        ["image_generation.partial_image"]: ImageGenPartialImageEvent$inboundSchema,
        ["image_generation.text_chunk"]: ImageGenTextChunkEvent$inboundSchema,
        ["image_generation.completed"]: ImageGenCompletedEvent$inboundSchema,
        error: ImageGenStreamErrorEvent$inboundSchema,
    })),
});
export function imageStreamingResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageStreamingResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageStreamingResponse' from JSON`);
}
