// @openrouter/sdk/esm/models/imagepricingentry.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7a0d3f05b12c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Billable = {
    OutputImage: "output_image",
    InputImage: "input_image",
    InputFont: "input_font",
    InputReference: "input_reference",
    InputText: "input_text",
};
export const Unit = {
    Image: "image",
    Megapixel: "megapixel",
    Token: "token",
};
/** @internal */
export const Billable$inboundSchema = openEnums
    .inboundSchema(Billable);
/** @internal */
export const Unit$inboundSchema = openEnums
    .inboundSchema(Unit);
/** @internal */
export const ImagePricingEntry$inboundSchema = z.object({
    billable: Billable$inboundSchema,
    cost_usd: z.number(),
    unit: Unit$inboundSchema,
    variant: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "cost_usd": "costUsd",
    });
});
export function imagePricingEntryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImagePricingEntry$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImagePricingEntry' from JSON`);
}
