// @openrouter/sdk/esm/models/imagegenerationresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 18aeba0f13cc
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ImageGenerationUsage$inboundSchema, } from "./imagegenerationusage.js";
/** @internal */
export const ImageGenerationResponseData$inboundSchema = z.object({
    b64_json: z.string(),
    media_type: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "b64_json": "b64Json",
        "media_type": "mediaType",
    });
});
export function imageGenerationResponseDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenerationResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenerationResponseData' from JSON`);
}
/** @internal */
export const ImageGenerationResponse$inboundSchema = z.object({
    created: z.int(),
    data: z.array(z.lazy(() => ImageGenerationResponseData$inboundSchema)),
    usage: ImageGenerationUsage$inboundSchema.optional(),
});
export function imageGenerationResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenerationResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenerationResponse' from JSON`);
}
