// @openrouter/sdk/esm/models/imageendpoint.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e13b5353df71
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { CapabilityDescriptor$inboundSchema, } from "./capabilitydescriptor.js";
import { ImagePricingEntry$inboundSchema, } from "./imagepricingentry.js";
/** @internal */
export const ImageEndpoint$inboundSchema = z
    .object({
    allowed_passthrough_parameters: z.array(z.string()),
    pricing: z.array(ImagePricingEntry$inboundSchema),
    provider_name: z.string(),
    provider_slug: z.string(),
    provider_tag: z.nullable(z.string()),
    supported_parameters: z.record(z.string(), CapabilityDescriptor$inboundSchema),
    supports_streaming: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "allowed_passthrough_parameters": "allowedPassthroughParameters",
        "provider_name": "providerName",
        "provider_slug": "providerSlug",
        "provider_tag": "providerTag",
        "supported_parameters": "supportedParameters",
        "supports_streaming": "supportsStreaming",
    });
});
export function imageEndpointFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageEndpoint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageEndpoint' from JSON`);
}
