// @openrouter/sdk/esm/models/imagegentextchunkevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4de0c36d6255
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
/**
 * The generation phase this chunk belongs to. `content` is the renderable output; `reasoning` and `draft` are intermediate provider phases.
 */
export const ImageGenTextChunkEventPhase = {
    Content: "content",
    Reasoning: "reasoning",
    Draft: "draft",
};
/** @internal */
export const ImageGenTextChunkEventPhase$inboundSchema = openEnums.inboundSchema(ImageGenTextChunkEventPhase);
/** @internal */
export const ImageGenTextChunkEvent$inboundSchema = z.object({
    phase: ImageGenTextChunkEventPhase$inboundSchema,
    text: z.string(),
    type: z.literal("image_generation.text_chunk"),
});
export function imageGenTextChunkEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenTextChunkEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenTextChunkEvent' from JSON`);
}
