// @openrouter/sdk/esm/models/imagemodellistitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 12f1f7100366
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { CapabilityDescriptor$inboundSchema, } from "./capabilitydescriptor.js";
import { ImageModelArchitecture$inboundSchema, } from "./imagemodelarchitecture.js";
/** @internal */
export const ImageModelListItem$inboundSchema = z.object({
    architecture: ImageModelArchitecture$inboundSchema,
    created: z.int(),
    description: z.string(),
    endpoints: z.string(),
    id: z.string(),
    name: z.string(),
    supported_parameters: z.record(z.string(), CapabilityDescriptor$inboundSchema),
    supports_streaming: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "supported_parameters": "supportedParameters",
        "supports_streaming": "supportsStreaming",
    });
});
export function imageModelListItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageModelListItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageModelListItem' from JSON`);
}
