// @openrouter/sdk/esm/models/inputsunion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3aff2165a17d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { AdditionalToolsItem$outboundSchema, } from "./additionaltoolsitem.js";
import { AgentMessageItem$outboundSchema, } from "./agentmessageitem.js";
import { ApplyPatchCallItem$outboundSchema, } from "./applypatchcallitem.js";
import { ApplyPatchCallOutputItem$outboundSchema, } from "./applypatchcalloutputitem.js";
import { CompactionItem$outboundSchema, } from "./compactionitem.js";
import { ContextCompactionItem$outboundSchema, } from "./contextcompactionitem.js";
import { CustomToolCallItem$outboundSchema, } from "./customtoolcallitem.js";
import { CustomToolCallOutputItem$outboundSchema, } from "./customtoolcalloutputitem.js";
import { EasyInputMessage$outboundSchema, } from "./easyinputmessage.js";
import { FunctionCallItem$outboundSchema, } from "./functioncallitem.js";
import { FunctionCallOutputItem$outboundSchema, } from "./functioncalloutputitem.js";
import { InputMessageItem$outboundSchema, } from "./inputmessageitem.js";
import { ItemReferenceItem$outboundSchema, } from "./itemreferenceitem.js";
import { LocalShellCallItem$outboundSchema, } from "./localshellcallitem.js";
import { LocalShellCallOutputItem$outboundSchema, } from "./localshellcalloutputitem.js";
import { McpApprovalRequestItem$outboundSchema, } from "./mcpapprovalrequestitem.js";
import { McpApprovalResponseItem$outboundSchema, } from "./mcpapprovalresponseitem.js";
import { McpCallItem$outboundSchema, } from "./mcpcallitem.js";
import { McpListToolsItem$outboundSchema, } from "./mcplisttoolsitem.js";
import { OpenAIResponsesRefusalContent$outboundSchema, } from "./openairesponsesrefusalcontent.js";
import { OutputAdvisorServerToolItem$outboundSchema, } from "./outputadvisorservertoolitem.js";
import { OutputApplyPatchServerToolItem$outboundSchema, } from "./outputapplypatchservertoolitem.js";
import { OutputBashServerToolItem$outboundSchema, } from "./outputbashservertoolitem.js";
import { OutputBrowserUseServerToolItem$outboundSchema, } from "./outputbrowseruseservertoolitem.js";
import { OutputCodeInterpreterCallItem$outboundSchema, } from "./outputcodeinterpretercallitem.js";
import { OutputCodeInterpreterServerToolItem$outboundSchema, } from "./outputcodeinterpreterservertoolitem.js";
import { OutputComputerCallItem$outboundSchema, } from "./outputcomputercallitem.js";
import { OutputCustomToolCallItem$outboundSchema, } from "./outputcustomtoolcallitem.js";
import { OutputDatetimeItem$outboundSchema, } from "./outputdatetimeitem.js";
import { OutputFileSearchCallItem$outboundSchema, } from "./outputfilesearchcallitem.js";
import { OutputFileSearchServerToolItem$outboundSchema, } from "./outputfilesearchservertoolitem.js";
import { OutputFilesServerToolItem$outboundSchema, } from "./outputfilesservertoolitem.js";
import { OutputFunctionCallItem$outboundSchema, } from "./outputfunctioncallitem.js";
import { OutputFusionServerToolItem$outboundSchema, } from "./outputfusionservertoolitem.js";
import { OutputImageGenerationCallItem$outboundSchema, } from "./outputimagegenerationcallitem.js";
import { OutputImageGenerationServerToolItem$outboundSchema, } from "./outputimagegenerationservertoolitem.js";
import { OutputMcpServerToolItem$outboundSchema, } from "./outputmcpservertoolitem.js";
import { OutputMemoryServerToolItem$outboundSchema, } from "./outputmemoryservertoolitem.js";
import { OutputSearchModelsServerToolItem$outboundSchema, } from "./outputsearchmodelsservertoolitem.js";
import { OutputSubagentServerToolItem$outboundSchema, } from "./outputsubagentservertoolitem.js";
import { OutputTextEditorServerToolItem$outboundSchema, } from "./outputtexteditorservertoolitem.js";
import { OutputToolSearchServerToolItem$outboundSchema, } from "./outputtoolsearchservertoolitem.js";
import { OutputWebFetchServerToolItem$outboundSchema, } from "./outputwebfetchservertoolitem.js";
import { OutputWebSearchCallItem$outboundSchema, } from "./outputwebsearchcallitem.js";
import { OutputWebSearchServerToolItem$outboundSchema, } from "./outputwebsearchservertoolitem.js";
import { ReasoningFormat$outboundSchema, } from "./reasoningformat.js";
import { ReasoningItem$outboundSchema, } from "./reasoningitem.js";
import { ReasoningSummaryText$outboundSchema, } from "./reasoningsummarytext.js";
import { ReasoningTextContent$outboundSchema, } from "./reasoningtextcontent.js";
import { ResponseOutputText$outboundSchema, } from "./responseoutputtext.js";
import { ShellCallItem$outboundSchema, } from "./shellcallitem.js";
import { ShellCallOutputItem$outboundSchema, } from "./shellcalloutputitem.js";
export const InputsStatusInProgress2 = {
    InProgress: "in_progress",
};
export const InputsStatusIncomplete2 = {
    Incomplete: "incomplete",
};
export const InputsStatusCompleted2 = {
    Completed: "completed",
};
export const InputsTypeReasoning = {
    Reasoning: "reasoning",
};
export const InputsPhaseFinalAnswer = {
    FinalAnswer: "final_answer",
};
export const InputsPhaseCommentary = {
    Commentary: "commentary",
};
export const InputsRole = {
    Assistant: "assistant",
};
export const InputsStatusInProgress1 = {
    InProgress: "in_progress",
};
export const InputsStatusIncomplete1 = {
    Incomplete: "incomplete",
};
export const InputsStatusCompleted1 = {
    Completed: "completed",
};
export const InputsTypeMessage = {
    Message: "message",
};
/** @internal */
export const InputsStatusInProgress2$outboundSchema = z.enum(InputsStatusInProgress2);
/** @internal */
export const InputsStatusIncomplete2$outboundSchema = z.enum(InputsStatusIncomplete2);
/** @internal */
export const InputsStatusCompleted2$outboundSchema = z.enum(InputsStatusCompleted2);
/** @internal */
export const InputsStatusUnion2$outboundSchema = z.union([
    InputsStatusCompleted2$outboundSchema,
    InputsStatusIncomplete2$outboundSchema,
    InputsStatusInProgress2$outboundSchema,
]);
export function inputsStatusUnion2ToJSON(inputsStatusUnion2) {
    return JSON.stringify(InputsStatusUnion2$outboundSchema.parse(inputsStatusUnion2));
}
/** @internal */
export const InputsTypeReasoning$outboundSchema = z.enum(InputsTypeReasoning);
/** @internal */
export const InputsReasoning$outboundSchema = z.object({
    content: z.nullable(z.array(ReasoningTextContent$outboundSchema)).optional(),
    encryptedContent: z.nullable(z.string()).optional(),
    id: z.string(),
    status: z.union([
        InputsStatusCompleted2$outboundSchema,
        InputsStatusIncomplete2$outboundSchema,
        InputsStatusInProgress2$outboundSchema,
    ]).optional(),
    summary: z.nullable(z.array(ReasoningSummaryText$outboundSchema)),
    type: InputsTypeReasoning$outboundSchema,
    format: z.nullable(ReasoningFormat$outboundSchema).optional(),
    signature: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        encryptedContent: "encrypted_content",
    });
});
export function inputsReasoningToJSON(inputsReasoning) {
    return JSON.stringify(InputsReasoning$outboundSchema.parse(inputsReasoning));
}
/** @internal */
export const InputsContent1$outboundSchema = z.union([
    ResponseOutputText$outboundSchema,
    OpenAIResponsesRefusalContent$outboundSchema,
]);
export function inputsContent1ToJSON(inputsContent1) {
    return JSON.stringify(InputsContent1$outboundSchema.parse(inputsContent1));
}
/** @internal */
export const InputsContent2$outboundSchema = z.union([
    z.array(z.union([
        ResponseOutputText$outboundSchema,
        OpenAIResponsesRefusalContent$outboundSchema,
    ])),
    z.string(),
]);
export function inputsContent2ToJSON(inputsContent2) {
    return JSON.stringify(InputsContent2$outboundSchema.parse(inputsContent2));
}
/** @internal */
export const InputsPhaseFinalAnswer$outboundSchema = z.enum(InputsPhaseFinalAnswer);
/** @internal */
export const InputsPhaseCommentary$outboundSchema = z.enum(InputsPhaseCommentary);
/** @internal */
export const InputsPhaseUnion$outboundSchema = z.union([
    InputsPhaseCommentary$outboundSchema,
    InputsPhaseFinalAnswer$outboundSchema,
]);
export function inputsPhaseUnionToJSON(inputsPhaseUnion) {
    return JSON.stringify(InputsPhaseUnion$outboundSchema.parse(inputsPhaseUnion));
}
/** @internal */
export const InputsRole$outboundSchema = z.enum(InputsRole);
/** @internal */
export const InputsStatusInProgress1$outboundSchema = z.enum(InputsStatusInProgress1);
/** @internal */
export const InputsStatusIncomplete1$outboundSchema = z.enum(InputsStatusIncomplete1);
/** @internal */
export const InputsStatusCompleted1$outboundSchema = z.enum(InputsStatusCompleted1);
/** @internal */
export const InputsStatusUnion1$outboundSchema = z.union([
    InputsStatusCompleted1$outboundSchema,
    InputsStatusIncomplete1$outboundSchema,
    InputsStatusInProgress1$outboundSchema,
]);
export function inputsStatusUnion1ToJSON(inputsStatusUnion1) {
    return JSON.stringify(InputsStatusUnion1$outboundSchema.parse(inputsStatusUnion1));
}
/** @internal */
export const InputsTypeMessage$outboundSchema = z.enum(InputsTypeMessage);
/** @internal */
export const InputsMessage$outboundSchema = z.object({
    content: z.nullable(z.union([
        z.array(z.union([
            ResponseOutputText$outboundSchema,
            OpenAIResponsesRefusalContent$outboundSchema,
        ])),
        z.string(),
    ])),
    id: z.string(),
    phase: z.nullable(z.union([
        InputsPhaseCommentary$outboundSchema,
        InputsPhaseFinalAnswer$outboundSchema,
    ])).optional(),
    role: InputsRole$outboundSchema,
    status: z.union([
        InputsStatusCompleted1$outboundSchema,
        InputsStatusIncomplete1$outboundSchema,
        InputsStatusInProgress1$outboundSchema,
    ]).optional(),
    type: InputsTypeMessage$outboundSchema,
});
export function inputsMessageToJSON(inputsMessage) {
    return JSON.stringify(InputsMessage$outboundSchema.parse(inputsMessage));
}
/** @internal */
export const InputsUnion1$outboundSchema = z.union([
    OutputCodeInterpreterCallItem$outboundSchema,
    FunctionCallItem$outboundSchema,
    LocalShellCallItem$outboundSchema,
    McpApprovalRequestItem$outboundSchema,
    McpCallItem$outboundSchema,
    ApplyPatchCallItem$outboundSchema,
    z.lazy(() => InputsMessage$outboundSchema),
    OutputFunctionCallItem$outboundSchema,
    OutputCustomToolCallItem$outboundSchema,
    OutputFileSearchCallItem$outboundSchema,
    OutputComputerCallItem$outboundSchema,
    OutputDatetimeItem$outboundSchema,
    McpListToolsItem$outboundSchema,
    CustomToolCallItem$outboundSchema,
    AgentMessageItem$outboundSchema,
    ReasoningItem$outboundSchema,
    FunctionCallOutputItem$outboundSchema,
    ApplyPatchCallOutputItem$outboundSchema,
    z.lazy(() => InputsReasoning$outboundSchema),
    OutputWebSearchCallItem$outboundSchema,
    OutputImageGenerationCallItem$outboundSchema,
    LocalShellCallOutputItem$outboundSchema,
    ShellCallItem$outboundSchema,
    ShellCallOutputItem$outboundSchema,
    McpApprovalResponseItem$outboundSchema,
    CustomToolCallOutputItem$outboundSchema,
    AdditionalToolsItem$outboundSchema,
    OutputWebSearchServerToolItem$outboundSchema,
    OutputCodeInterpreterServerToolItem$outboundSchema,
    OutputFileSearchServerToolItem$outboundSchema,
    OutputImageGenerationServerToolItem$outboundSchema,
    OutputBrowserUseServerToolItem$outboundSchema,
    OutputBashServerToolItem$outboundSchema,
    OutputTextEditorServerToolItem$outboundSchema,
    OutputApplyPatchServerToolItem$outboundSchema,
    OutputWebFetchServerToolItem$outboundSchema,
    OutputToolSearchServerToolItem$outboundSchema,
    OutputMemoryServerToolItem$outboundSchema,
    OutputMcpServerToolItem$outboundSchema,
    OutputSearchModelsServerToolItem$outboundSchema,
    OutputFusionServerToolItem$outboundSchema,
    OutputAdvisorServerToolItem$outboundSchema,
    OutputSubagentServerToolItem$outboundSchema,
    OutputFilesServerToolItem$outboundSchema,
    CompactionItem$outboundSchema,
    ItemReferenceItem$outboundSchema,
    EasyInputMessage$outboundSchema,
    InputMessageItem$outboundSchema,
    ContextCompactionItem$outboundSchema,
]);
export function inputsUnion1ToJSON(inputsUnion1) {
    return JSON.stringify(InputsUnion1$outboundSchema.parse(inputsUnion1));
}
/** @internal */
export const InputsUnion$outboundSchema = z.union([
    z.string(),
    z.array(z.union([
        OutputCodeInterpreterCallItem$outboundSchema,
        FunctionCallItem$outboundSchema,
        LocalShellCallItem$outboundSchema,
        McpApprovalRequestItem$outboundSchema,
        McpCallItem$outboundSchema,
        ApplyPatchCallItem$outboundSchema,
        z.lazy(() => InputsMessage$outboundSchema),
        OutputFunctionCallItem$outboundSchema,
        OutputCustomToolCallItem$outboundSchema,
        OutputFileSearchCallItem$outboundSchema,
        OutputComputerCallItem$outboundSchema,
        OutputDatetimeItem$outboundSchema,
        McpListToolsItem$outboundSchema,
        CustomToolCallItem$outboundSchema,
        AgentMessageItem$outboundSchema,
        ReasoningItem$outboundSchema,
        FunctionCallOutputItem$outboundSchema,
        ApplyPatchCallOutputItem$outboundSchema,
        z.lazy(() => InputsReasoning$outboundSchema),
        OutputWebSearchCallItem$outboundSchema,
        OutputImageGenerationCallItem$outboundSchema,
        LocalShellCallOutputItem$outboundSchema,
        ShellCallItem$outboundSchema,
        ShellCallOutputItem$outboundSchema,
        McpApprovalResponseItem$outboundSchema,
        CustomToolCallOutputItem$outboundSchema,
        AdditionalToolsItem$outboundSchema,
        OutputWebSearchServerToolItem$outboundSchema,
        OutputCodeInterpreterServerToolItem$outboundSchema,
        OutputFileSearchServerToolItem$outboundSchema,
        OutputImageGenerationServerToolItem$outboundSchema,
        OutputBrowserUseServerToolItem$outboundSchema,
        OutputBashServerToolItem$outboundSchema,
        OutputTextEditorServerToolItem$outboundSchema,
        OutputApplyPatchServerToolItem$outboundSchema,
        OutputWebFetchServerToolItem$outboundSchema,
        OutputToolSearchServerToolItem$outboundSchema,
        OutputMemoryServerToolItem$outboundSchema,
        OutputMcpServerToolItem$outboundSchema,
        OutputSearchModelsServerToolItem$outboundSchema,
        OutputFusionServerToolItem$outboundSchema,
        OutputAdvisorServerToolItem$outboundSchema,
        OutputSubagentServerToolItem$outboundSchema,
        OutputFilesServerToolItem$outboundSchema,
        CompactionItem$outboundSchema,
        ItemReferenceItem$outboundSchema,
        EasyInputMessage$outboundSchema,
        InputMessageItem$outboundSchema,
        ContextCompactionItem$outboundSchema,
    ])),
]);
export function inputsUnionToJSON(inputsUnion) {
    return JSON.stringify(InputsUnion$outboundSchema.parse(inputsUnion));
}
