// @openrouter/sdk/esm/models/inputtext.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8d6c10a240ae
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { PromptCacheBreakpoint$inboundSchema, PromptCacheBreakpoint$outboundSchema, } from "./promptcachebreakpoint.js";
/** @internal */
export const InputText$inboundSchema = z.object({
    prompt_cache_breakpoint: z.nullable(PromptCacheBreakpoint$inboundSchema)
        .optional(),
    text: z.string(),
    type: z.literal("input_text"),
}).transform((v) => {
    return remap$(v, {
        "prompt_cache_breakpoint": "promptCacheBreakpoint",
    });
});
/** @internal */
export const InputText$outboundSchema = z.object({
    promptCacheBreakpoint: z.nullable(PromptCacheBreakpoint$outboundSchema)
        .optional(),
    text: z.string(),
    type: z.literal("input_text"),
}).transform((v) => {
    return remap$(v, {
        promptCacheBreakpoint: "prompt_cache_breakpoint",
    });
});
export function inputTextToJSON(inputText) {
    return JSON.stringify(InputText$outboundSchema.parse(inputText));
}
export function inputTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InputText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InputText' from JSON`);
}
