// @openrouter/sdk/esm/models/listendpointsresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5178e92a44ba
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { InputModality$inboundSchema } from "./inputmodality.js";
import { InstructType$inboundSchema } from "./instructtype.js";
import { ModelGroup$inboundSchema } from "./modelgroup.js";
import { OutputModality$inboundSchema, } from "./outputmodality.js";
import { PublicEndpoint$inboundSchema, } from "./publicendpoint.js";
/** @internal */
export const Architecture$inboundSchema = z
    .object({
    input_modalities: z.array(InputModality$inboundSchema),
    instruct_type: z.nullable(InstructType$inboundSchema),
    modality: z.nullable(z.string()),
    output_modalities: z.array(OutputModality$inboundSchema),
    tokenizer: z.nullable(ModelGroup$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "input_modalities": "inputModalities",
        "instruct_type": "instructType",
        "output_modalities": "outputModalities",
    });
});
export function architectureFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Architecture$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Architecture' from JSON`);
}
/** @internal */
export const ListEndpointsResponse$inboundSchema = z.object({
    architecture: z.lazy(() => Architecture$inboundSchema),
    created: z.int(),
    description: z.string(),
    endpoints: z.array(PublicEndpoint$inboundSchema),
    id: z.string(),
    name: z.string(),
});
export function listEndpointsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListEndpointsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListEndpointsResponse' from JSON`);
}
