// @openrouter/sdk/esm/models/localshellcallitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 01cd50ccbc7a
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const TypeExec = {
    Exec: "exec",
};
export const TypeLocalShellCall = {
    LocalShellCall: "local_shell_call",
};
/** @internal */
export const TypeExec$outboundSchema = z.enum(TypeExec);
/** @internal */
export const LocalShellCallItemAction$outboundSchema = z.object({
    command: z.array(z.string()),
    env: z.record(z.string(), z.string()),
    timeoutMs: z.nullable(z.int()).optional(),
    type: TypeExec$outboundSchema,
    user: z.nullable(z.string()).optional(),
    workingDirectory: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        timeoutMs: "timeout_ms",
        workingDirectory: "working_directory",
    });
});
export function localShellCallItemActionToJSON(localShellCallItemAction) {
    return JSON.stringify(LocalShellCallItemAction$outboundSchema.parse(localShellCallItemAction));
}
/** @internal */
export const TypeLocalShellCall$outboundSchema = z.enum(TypeLocalShellCall);
/** @internal */
export const LocalShellCallItem$outboundSchema = z.object({
    action: z.lazy(() => LocalShellCallItemAction$outboundSchema),
    callId: z.string(),
    id: z.string(),
    status: ToolCallStatus$outboundSchema,
    type: TypeLocalShellCall$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function localShellCallItemToJSON(localShellCallItem) {
    return JSON.stringify(LocalShellCallItem$outboundSchema.parse(localShellCallItem));
}
