// @openrouter/sdk/esm/models/mcplisttoolsitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3a1a2af50e51
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
export const McpListToolsItemType = {
    McpListTools: "mcp_list_tools",
};
/** @internal */
export const McpListToolsItemTool$outboundSchema = z.object({
    annotations: z.any().optional(),
    description: z.nullable(z.string()).optional(),
    inputSchema: z.record(z.string(), z.any()),
    name: z.string(),
}).transform((v) => {
    return remap$(v, {
        inputSchema: "input_schema",
    });
});
export function mcpListToolsItemToolToJSON(mcpListToolsItemTool) {
    return JSON.stringify(McpListToolsItemTool$outboundSchema.parse(mcpListToolsItemTool));
}
/** @internal */
export const McpListToolsItemType$outboundSchema = z.enum(McpListToolsItemType);
/** @internal */
export const McpListToolsItem$outboundSchema = z.object({
    error: z.nullable(z.string()).optional(),
    id: z.string(),
    serverLabel: z.string(),
    tools: z.array(z.lazy(() => McpListToolsItemTool$outboundSchema)),
    type: McpListToolsItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        serverLabel: "server_label",
    });
});
export function mcpListToolsItemToJSON(mcpListToolsItem) {
    return JSON.stringify(McpListToolsItem$outboundSchema.parse(mcpListToolsItem));
}
