// @openrouter/sdk/esm/models/messagesrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 572964d5906d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { AnthropicAllowedCallers$outboundSchema, } from "./anthropicallowedcallers.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
import { AnthropicInputTokensClearAtLeast$outboundSchema, } from "./anthropicinputtokensclearatleast.js";
import { AnthropicInputTokensTrigger$outboundSchema, } from "./anthropicinputtokenstrigger.js";
import { AnthropicTextBlockParam$outboundSchema, } from "./anthropictextblockparam.js";
import { AnthropicThinkingDisplay$outboundSchema, } from "./anthropicthinkingdisplay.js";
import { AnthropicThinkingTurns$outboundSchema, } from "./anthropicthinkingturns.js";
import { AnthropicToolSearchToolBm25$outboundSchema, } from "./anthropictoolsearchtoolbm25.js";
import { AnthropicToolSearchToolRegex$outboundSchema, } from "./anthropictoolsearchtoolregex.js";
import { AnthropicToolUsesKeep$outboundSchema, } from "./anthropictooluseskeep.js";
import { AnthropicToolUsesTrigger$outboundSchema, } from "./anthropictoolusestrigger.js";
import { AnthropicWebSearchToolUserLocation$outboundSchema, } from "./anthropicwebsearchtooluserlocation.js";
import { AutoBetaRouterPlugin$outboundSchema, } from "./autobetarouterplugin.js";
import { AutoRouterPlugin$outboundSchema, } from "./autorouterplugin.js";
import { BashServerTool$outboundSchema, } from "./bashservertool.js";
import { ContextCompressionPlugin$outboundSchema, } from "./contextcompressionplugin.js";
import { DatetimeServerTool$outboundSchema, } from "./datetimeservertool.js";
import { FileParserPlugin$outboundSchema, } from "./fileparserplugin.js";
import { FusionPlugin$outboundSchema, } from "./fusionplugin.js";
import { ImageGenerationServerToolOpenRouter$outboundSchema, } from "./imagegenerationservertoolopenrouter.js";
import { MessagesFallbackParam$outboundSchema, } from "./messagesfallbackparam.js";
import { MessagesMessageParam$outboundSchema, } from "./messagesmessageparam.js";
import { MessagesOutputConfig$outboundSchema, } from "./messagesoutputconfig.js";
import { MessagesSearchModelsServerTool$outboundSchema, } from "./messagessearchmodelsservertool.js";
import { ModerationPlugin$outboundSchema, } from "./moderationplugin.js";
import { OpenRouterWebSearchServerTool$outboundSchema, } from "./openrouterwebsearchservertool.js";
import { ParetoRouterPlugin$outboundSchema, } from "./paretorouterplugin.js";
import { ProviderPreferences$outboundSchema, } from "./providerpreferences.js";
import { ResponseHealingPlugin$outboundSchema, } from "./responsehealingplugin.js";
import { StopServerToolsWhenCondition$outboundSchema, } from "./stopservertoolswhencondition.js";
import { TraceConfig$outboundSchema, } from "./traceconfig.js";
import { WebFetchPlugin$outboundSchema, } from "./webfetchplugin.js";
import { WebFetchServerTool$outboundSchema, } from "./webfetchservertool.js";
import { WebSearchPlugin$outboundSchema, } from "./websearchplugin.js";
export const EditTypeInputTokens = {
    InputTokens: "input_tokens",
};
export const KeepEnum = {
    All: "all",
};
export const KeepType = {
    All: "all",
};
/**
 * Controls output generation speed. When set to `fast`, uses a higher-speed inference configuration at premium pricing. Defaults to `standard` when omitted.
 */
export const Speed = {
    Fast: "fast",
    Standard: "standard",
};
export const NameAdvisor = {
    Advisor: "advisor",
};
export const TypeAdvisor20260301 = {
    Advisor20260301: "advisor_20260301",
};
export const NameWebSearch2 = {
    WebSearch: "web_search",
};
export const TypeWebSearch20260209 = {
    WebSearch20260209: "web_search_20260209",
};
export const NameWebSearch1 = {
    WebSearch: "web_search",
};
export const TypeWebSearch20250305 = {
    WebSearch20250305: "web_search_20250305",
};
export const NameStrReplaceEditor = {
    StrReplaceEditor: "str_replace_editor",
};
export const TypeTextEditor20250124 = {
    TextEditor20250124: "text_editor_20250124",
};
export const NameBash = {
    Bash: "bash",
};
export const TypeBash20250124 = {
    Bash20250124: "bash_20250124",
};
export const ToolTypeCustom = {
    Custom: "custom",
};
/** @internal */
export const EditTypeInputTokens$outboundSchema = z.enum(EditTypeInputTokens);
/** @internal */
export const TriggerInputTokens$outboundSchema = z.object({
    type: EditTypeInputTokens$outboundSchema,
    value: z.int(),
});
export function triggerInputTokensToJSON(triggerInputTokens) {
    return JSON.stringify(TriggerInputTokens$outboundSchema.parse(triggerInputTokens));
}
/** @internal */
export const EditCompact20260112$outboundSchema = z.object({
    instructions: z.nullable(z.string()).optional(),
    pauseAfterCompaction: z.boolean().optional(),
    trigger: z.nullable(z.lazy(() => TriggerInputTokens$outboundSchema))
        .optional(),
    type: z.literal("compact_20260112"),
}).transform((v) => {
    return remap$(v, {
        pauseAfterCompaction: "pause_after_compaction",
    });
});
export function editCompact20260112ToJSON(editCompact20260112) {
    return JSON.stringify(EditCompact20260112$outboundSchema.parse(editCompact20260112));
}
/** @internal */
export const KeepEnum$outboundSchema = z.enum(KeepEnum);
/** @internal */
export const KeepType$outboundSchema = z.enum(KeepType);
/** @internal */
export const KeepAll$outboundSchema = z
    .object({
    type: KeepType$outboundSchema,
});
export function keepAllToJSON(keepAll) {
    return JSON.stringify(KeepAll$outboundSchema.parse(keepAll));
}
/** @internal */
export const Keep$outboundSchema = z.union([
    AnthropicThinkingTurns$outboundSchema,
    z.lazy(() => KeepAll$outboundSchema),
    KeepEnum$outboundSchema,
]);
export function keepToJSON(keep) {
    return JSON.stringify(Keep$outboundSchema.parse(keep));
}
/** @internal */
export const EditClearThinking20251015$outboundSchema = z.object({
    keep: z.union([
        AnthropicThinkingTurns$outboundSchema,
        z.lazy(() => KeepAll$outboundSchema),
        KeepEnum$outboundSchema,
    ]).optional(),
    type: z.literal("clear_thinking_20251015"),
});
export function editClearThinking20251015ToJSON(editClearThinking20251015) {
    return JSON.stringify(EditClearThinking20251015$outboundSchema.parse(editClearThinking20251015));
}
/** @internal */
export const ClearToolInputs$outboundSchema = z.union([z.boolean(), z.array(z.string())]);
export function clearToolInputsToJSON(clearToolInputs) {
    return JSON.stringify(ClearToolInputs$outboundSchema.parse(clearToolInputs));
}
/** @internal */
export const Trigger$outboundSchema = z
    .union([
    AnthropicInputTokensTrigger$outboundSchema,
    AnthropicToolUsesTrigger$outboundSchema,
]);
export function triggerToJSON(trigger) {
    return JSON.stringify(Trigger$outboundSchema.parse(trigger));
}
/** @internal */
export const EditClearToolUses20250919$outboundSchema = z.object({
    clearAtLeast: z.nullable(AnthropicInputTokensClearAtLeast$outboundSchema)
        .optional(),
    clearToolInputs: z.nullable(z.union([z.boolean(), z.array(z.string())]))
        .optional(),
    excludeTools: z.nullable(z.array(z.string())).optional(),
    keep: AnthropicToolUsesKeep$outboundSchema.optional(),
    trigger: z.union([
        AnthropicInputTokensTrigger$outboundSchema,
        AnthropicToolUsesTrigger$outboundSchema,
    ]).optional(),
    type: z.literal("clear_tool_uses_20250919"),
}).transform((v) => {
    return remap$(v, {
        clearAtLeast: "clear_at_least",
        clearToolInputs: "clear_tool_inputs",
        excludeTools: "exclude_tools",
    });
});
export function editClearToolUses20250919ToJSON(editClearToolUses20250919) {
    return JSON.stringify(EditClearToolUses20250919$outboundSchema.parse(editClearToolUses20250919));
}
/** @internal */
export const Edit$outboundSchema = z.union([
    z.lazy(() => EditClearToolUses20250919$outboundSchema),
    z.lazy(() => EditClearThinking20251015$outboundSchema),
    z.lazy(() => EditCompact20260112$outboundSchema),
]);
export function editToJSON(edit) {
    return JSON.stringify(Edit$outboundSchema.parse(edit));
}
/** @internal */
export const ContextManagement$outboundSchema = z.object({
    edits: z.array(z.union([
        z.lazy(() => EditClearToolUses20250919$outboundSchema),
        z.lazy(() => EditClearThinking20251015$outboundSchema),
        z.lazy(() => EditCompact20260112$outboundSchema),
    ])).optional(),
});
export function contextManagementToJSON(contextManagement) {
    return JSON.stringify(ContextManagement$outboundSchema.parse(contextManagement));
}
/** @internal */
export const MessagesRequestMetadata$outboundSchema = z.object({
    userId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        userId: "user_id",
    });
});
export function messagesRequestMetadataToJSON(messagesRequestMetadata) {
    return JSON.stringify(MessagesRequestMetadata$outboundSchema.parse(messagesRequestMetadata));
}
/** @internal */
export const MessagesRequestPlugin$outboundSchema = z.union([
    AutoBetaRouterPlugin$outboundSchema,
    AutoRouterPlugin$outboundSchema,
    ContextCompressionPlugin$outboundSchema,
    FileParserPlugin$outboundSchema,
    FusionPlugin$outboundSchema,
    ModerationPlugin$outboundSchema,
    ParetoRouterPlugin$outboundSchema,
    ResponseHealingPlugin$outboundSchema,
    WebSearchPlugin$outboundSchema,
    WebFetchPlugin$outboundSchema,
]);
export function messagesRequestPluginToJSON(messagesRequestPlugin) {
    return JSON.stringify(MessagesRequestPlugin$outboundSchema.parse(messagesRequestPlugin));
}
/** @internal */
export const Speed$outboundSchema = openEnums
    .outboundSchema(Speed);
/** @internal */
export const System$outboundSchema = z
    .union([z.string(), z.array(AnthropicTextBlockParam$outboundSchema)]);
export function systemToJSON(system) {
    return JSON.stringify(System$outboundSchema.parse(system));
}
/** @internal */
export const ThinkingAdaptive$outboundSchema = z.object({
    display: z.nullable(AnthropicThinkingDisplay$outboundSchema).optional(),
    type: z.literal("adaptive"),
});
export function thinkingAdaptiveToJSON(thinkingAdaptive) {
    return JSON.stringify(ThinkingAdaptive$outboundSchema.parse(thinkingAdaptive));
}
/** @internal */
export const ThinkingDisabled$outboundSchema = z.object({
    type: z.literal("disabled"),
});
export function thinkingDisabledToJSON(thinkingDisabled) {
    return JSON.stringify(ThinkingDisabled$outboundSchema.parse(thinkingDisabled));
}
/** @internal */
export const ThinkingEnabled$outboundSchema = z.object({
    budgetTokens: z.int(),
    display: z.nullable(AnthropicThinkingDisplay$outboundSchema).optional(),
    type: z.literal("enabled"),
}).transform((v) => {
    return remap$(v, {
        budgetTokens: "budget_tokens",
    });
});
export function thinkingEnabledToJSON(thinkingEnabled) {
    return JSON.stringify(ThinkingEnabled$outboundSchema.parse(thinkingEnabled));
}
/** @internal */
export const Thinking$outboundSchema = z
    .union([
    z.lazy(() => ThinkingEnabled$outboundSchema),
    z.lazy(() => ThinkingDisabled$outboundSchema),
    z.lazy(() => ThinkingAdaptive$outboundSchema),
]);
export function thinkingToJSON(thinking) {
    return JSON.stringify(Thinking$outboundSchema.parse(thinking));
}
/** @internal */
export const ToolChoiceTool$outboundSchema = z.object({
    disableParallelToolUse: z.boolean().optional(),
    name: z.string(),
    type: z.literal("tool"),
}).transform((v) => {
    return remap$(v, {
        disableParallelToolUse: "disable_parallel_tool_use",
    });
});
export function toolChoiceToolToJSON(toolChoiceTool) {
    return JSON.stringify(ToolChoiceTool$outboundSchema.parse(toolChoiceTool));
}
/** @internal */
export const ToolChoiceNone$outboundSchema = z.object({
    type: z.literal("none"),
});
export function toolChoiceNoneToJSON(toolChoiceNone) {
    return JSON.stringify(ToolChoiceNone$outboundSchema.parse(toolChoiceNone));
}
/** @internal */
export const ToolChoiceAny$outboundSchema = z.object({
    disableParallelToolUse: z.boolean().optional(),
    type: z.literal("any"),
}).transform((v) => {
    return remap$(v, {
        disableParallelToolUse: "disable_parallel_tool_use",
    });
});
export function toolChoiceAnyToJSON(toolChoiceAny) {
    return JSON.stringify(ToolChoiceAny$outboundSchema.parse(toolChoiceAny));
}
/** @internal */
export const ToolChoiceAuto$outboundSchema = z.object({
    disableParallelToolUse: z.boolean().optional(),
    type: z.literal("auto"),
}).transform((v) => {
    return remap$(v, {
        disableParallelToolUse: "disable_parallel_tool_use",
    });
});
export function toolChoiceAutoToJSON(toolChoiceAuto) {
    return JSON.stringify(ToolChoiceAuto$outboundSchema.parse(toolChoiceAuto));
}
/** @internal */
export const ToolChoice$outboundSchema = z.union([
    z.lazy(() => ToolChoiceAuto$outboundSchema),
    z.lazy(() => ToolChoiceAny$outboundSchema),
    z.lazy(() => ToolChoiceNone$outboundSchema),
    z.lazy(() => ToolChoiceTool$outboundSchema),
]);
export function toolChoiceToJSON(toolChoice) {
    return JSON.stringify(ToolChoice$outboundSchema.parse(toolChoice));
}
/** @internal */
export const MessagesRequestTool$outboundSchema = z.object({
    type: z.string(),
    additionalProperties: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function messagesRequestToolToJSON(messagesRequestTool) {
    return JSON.stringify(MessagesRequestTool$outboundSchema.parse(messagesRequestTool));
}
/** @internal */
export const NameAdvisor$outboundSchema = z.enum(NameAdvisor);
/** @internal */
export const TypeAdvisor20260301$outboundSchema = z.enum(TypeAdvisor20260301);
/** @internal */
export const ToolAdvisor20260301$outboundSchema = z.object({
    allowedCallers: z.array(AnthropicAllowedCallers$outboundSchema).optional(),
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    caching: z.nullable(AnthropicCacheControlDirective$outboundSchema).optional(),
    deferLoading: z.boolean().optional(),
    maxUses: z.int().optional(),
    model: z.string(),
    name: NameAdvisor$outboundSchema,
    type: TypeAdvisor20260301$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        allowedCallers: "allowed_callers",
        cacheControl: "cache_control",
        deferLoading: "defer_loading",
        maxUses: "max_uses",
    });
});
export function toolAdvisor20260301ToJSON(toolAdvisor20260301) {
    return JSON.stringify(ToolAdvisor20260301$outboundSchema.parse(toolAdvisor20260301));
}
/** @internal */
export const NameWebSearch2$outboundSchema = z
    .enum(NameWebSearch2);
/** @internal */
export const TypeWebSearch20260209$outboundSchema = z.enum(TypeWebSearch20260209);
/** @internal */
export const ToolWebSearch20260209$outboundSchema = z.object({
    allowedCallers: z.array(AnthropicAllowedCallers$outboundSchema).optional(),
    allowedDomains: z.nullable(z.array(z.string())).optional(),
    blockedDomains: z.nullable(z.array(z.string())).optional(),
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    maxUses: z.nullable(z.int()).optional(),
    name: NameWebSearch2$outboundSchema,
    type: TypeWebSearch20260209$outboundSchema,
    userLocation: z.nullable(AnthropicWebSearchToolUserLocation$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        allowedCallers: "allowed_callers",
        allowedDomains: "allowed_domains",
        blockedDomains: "blocked_domains",
        cacheControl: "cache_control",
        maxUses: "max_uses",
        userLocation: "user_location",
    });
});
export function toolWebSearch20260209ToJSON(toolWebSearch20260209) {
    return JSON.stringify(ToolWebSearch20260209$outboundSchema.parse(toolWebSearch20260209));
}
/** @internal */
export const NameWebSearch1$outboundSchema = z
    .enum(NameWebSearch1);
/** @internal */
export const TypeWebSearch20250305$outboundSchema = z.enum(TypeWebSearch20250305);
/** @internal */
export const ToolWebSearch20250305$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
    blockedDomains: z.nullable(z.array(z.string())).optional(),
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    maxUses: z.nullable(z.int()).optional(),
    name: NameWebSearch1$outboundSchema,
    type: TypeWebSearch20250305$outboundSchema,
    userLocation: z.nullable(AnthropicWebSearchToolUserLocation$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
        blockedDomains: "blocked_domains",
        cacheControl: "cache_control",
        maxUses: "max_uses",
        userLocation: "user_location",
    });
});
export function toolWebSearch20250305ToJSON(toolWebSearch20250305) {
    return JSON.stringify(ToolWebSearch20250305$outboundSchema.parse(toolWebSearch20250305));
}
/** @internal */
export const NameStrReplaceEditor$outboundSchema = z.enum(NameStrReplaceEditor);
/** @internal */
export const TypeTextEditor20250124$outboundSchema = z.enum(TypeTextEditor20250124);
/** @internal */
export const ToolTextEditor20250124$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    name: NameStrReplaceEditor$outboundSchema,
    type: TypeTextEditor20250124$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function toolTextEditor20250124ToJSON(toolTextEditor20250124) {
    return JSON.stringify(ToolTextEditor20250124$outboundSchema.parse(toolTextEditor20250124));
}
/** @internal */
export const NameBash$outboundSchema = z.enum(NameBash);
/** @internal */
export const TypeBash20250124$outboundSchema = z.enum(TypeBash20250124);
/** @internal */
export const ToolBash20250124$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    name: NameBash$outboundSchema,
    type: TypeBash20250124$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function toolBash20250124ToJSON(toolBash20250124) {
    return JSON.stringify(ToolBash20250124$outboundSchema.parse(toolBash20250124));
}
/** @internal */
export const InputSchema$outboundSchema = z.object({
    properties: z.any().optional(),
    required: z.nullable(z.array(z.string())).optional(),
    type: z.string().default("object"),
    additionalProperties: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function inputSchemaToJSON(inputSchema) {
    return JSON.stringify(InputSchema$outboundSchema.parse(inputSchema));
}
/** @internal */
export const ToolTypeCustom$outboundSchema = z
    .enum(ToolTypeCustom);
/** @internal */
export const ToolCustom$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    description: z.string().optional(),
    inputSchema: z.lazy(() => InputSchema$outboundSchema),
    name: z.string(),
    type: ToolTypeCustom$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
        inputSchema: "input_schema",
    });
});
export function toolCustomToJSON(toolCustom) {
    return JSON.stringify(ToolCustom$outboundSchema.parse(toolCustom));
}
/** @internal */
export const MessagesRequestToolUnion$outboundSchema = z.union([
    z.lazy(() => ToolAdvisor20260301$outboundSchema),
    z.lazy(() => ToolCustom$outboundSchema),
    z.lazy(() => ToolBash20250124$outboundSchema),
    z.lazy(() => ToolTextEditor20250124$outboundSchema),
    z.lazy(() => ToolWebSearch20250305$outboundSchema),
    z.lazy(() => ToolWebSearch20260209$outboundSchema),
    AnthropicToolSearchToolBm25$outboundSchema,
    AnthropicToolSearchToolRegex$outboundSchema,
    BashServerTool$outboundSchema,
    DatetimeServerTool$outboundSchema,
    ImageGenerationServerToolOpenRouter$outboundSchema,
    MessagesSearchModelsServerTool$outboundSchema,
    WebFetchServerTool$outboundSchema,
    OpenRouterWebSearchServerTool$outboundSchema,
    z.lazy(() => MessagesRequestTool$outboundSchema),
]);
export function messagesRequestToolUnionToJSON(messagesRequestToolUnion) {
    return JSON.stringify(MessagesRequestToolUnion$outboundSchema.parse(messagesRequestToolUnion));
}
/** @internal */
export const MessagesRequest$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    contextManagement: z.nullable(z.lazy(() => ContextManagement$outboundSchema))
        .optional(),
    fallbacks: z.nullable(z.array(MessagesFallbackParam$outboundSchema))
        .optional(),
    maxTokens: z.int().optional(),
    messages: z.nullable(z.array(MessagesMessageParam$outboundSchema)),
    metadata: z.lazy(() => MessagesRequestMetadata$outboundSchema).optional(),
    model: z.string(),
    models: z.array(z.string()).optional(),
    outputConfig: MessagesOutputConfig$outboundSchema.optional(),
    plugins: z.array(z.union([
        AutoBetaRouterPlugin$outboundSchema,
        AutoRouterPlugin$outboundSchema,
        ContextCompressionPlugin$outboundSchema,
        FileParserPlugin$outboundSchema,
        FusionPlugin$outboundSchema,
        ModerationPlugin$outboundSchema,
        ParetoRouterPlugin$outboundSchema,
        ResponseHealingPlugin$outboundSchema,
        WebSearchPlugin$outboundSchema,
        WebFetchPlugin$outboundSchema,
    ])).optional(),
    provider: z.nullable(ProviderPreferences$outboundSchema).optional(),
    serviceTier: z.string().optional(),
    sessionId: z.string().optional(),
    speed: z.nullable(Speed$outboundSchema).optional(),
    stopSequences: z.array(z.string()).optional(),
    stopServerToolsWhen: z.array(StopServerToolsWhenCondition$outboundSchema)
        .optional(),
    stream: z.boolean().optional(),
    system: z.union([z.string(), z.array(AnthropicTextBlockParam$outboundSchema)])
        .optional(),
    temperature: z.number().optional(),
    thinking: z.union([
        z.lazy(() => ThinkingEnabled$outboundSchema),
        z.lazy(() => ThinkingDisabled$outboundSchema),
        z.lazy(() => ThinkingAdaptive$outboundSchema),
    ]).optional(),
    toolChoice: z.union([
        z.lazy(() => ToolChoiceAuto$outboundSchema),
        z.lazy(() => ToolChoiceAny$outboundSchema),
        z.lazy(() => ToolChoiceNone$outboundSchema),
        z.lazy(() => ToolChoiceTool$outboundSchema),
    ]).optional(),
    tools: z.array(z.union([
        z.lazy(() => ToolAdvisor20260301$outboundSchema),
        z.lazy(() => ToolCustom$outboundSchema),
        z.lazy(() => ToolBash20250124$outboundSchema),
        z.lazy(() => ToolTextEditor20250124$outboundSchema),
        z.lazy(() => ToolWebSearch20250305$outboundSchema),
        z.lazy(() => ToolWebSearch20260209$outboundSchema),
        AnthropicToolSearchToolBm25$outboundSchema,
        AnthropicToolSearchToolRegex$outboundSchema,
        BashServerTool$outboundSchema,
        DatetimeServerTool$outboundSchema,
        ImageGenerationServerToolOpenRouter$outboundSchema,
        MessagesSearchModelsServerTool$outboundSchema,
        WebFetchServerTool$outboundSchema,
        OpenRouterWebSearchServerTool$outboundSchema,
        z.lazy(() => MessagesRequestTool$outboundSchema),
    ])).optional(),
    topK: z.int().optional(),
    topP: z.number().optional(),
    trace: TraceConfig$outboundSchema.optional(),
    user: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
        contextManagement: "context_management",
        maxTokens: "max_tokens",
        outputConfig: "output_config",
        serviceTier: "service_tier",
        sessionId: "session_id",
        stopSequences: "stop_sequences",
        stopServerToolsWhen: "stop_server_tools_when",
        toolChoice: "tool_choice",
        topK: "top_k",
        topP: "top_p",
    });
});
export function messagesRequestToJSON(messagesRequest) {
    return JSON.stringify(MessagesRequest$outboundSchema.parse(messagesRequest));
}
