// @openrouter/sdk/esm/models/messagesmessageparam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0a3ea5a7e049
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
import { AnthropicDocumentBlockParam$outboundSchema, } from "./anthropicdocumentblockparam.js";
import { AnthropicImageBlockParam$outboundSchema, } from "./anthropicimageblockparam.js";
import { AnthropicSearchResultBlockParam$outboundSchema, } from "./anthropicsearchresultblockparam.js";
import { AnthropicTextBlockParam$outboundSchema, } from "./anthropictextblockparam.js";
import { AnthropicWebSearchResultBlockParam$outboundSchema, } from "./anthropicwebsearchresultblockparam.js";
import { MessagesAdvisorToolResultBlock$outboundSchema, } from "./messagesadvisortoolresultblock.js";
export const ErrorCode = {
    InvalidToolInput: "invalid_tool_input",
    Unavailable: "unavailable",
    MaxUsesExceeded: "max_uses_exceeded",
    TooManyRequests: "too_many_requests",
    QueryTooLong: "query_too_long",
};
export const TypeWebSearchToolResultError = {
    WebSearchToolResultError: "web_search_tool_result_error",
};
export const MessagesMessageParamRole = {
    User: "user",
    Assistant: "assistant",
    System: "system",
};
/** @internal */
export const ContentCompaction$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    content: z.nullable(z.string()),
    type: z.literal("compaction"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function contentCompactionToJSON(contentCompaction) {
    return JSON.stringify(ContentCompaction$outboundSchema.parse(contentCompaction));
}
/** @internal */
export const ErrorCode$outboundSchema = openEnums
    .outboundSchema(ErrorCode);
/** @internal */
export const TypeWebSearchToolResultError$outboundSchema = z.enum(TypeWebSearchToolResultError);
/** @internal */
export const ContentWebSearchToolResultError$outboundSchema = z.object({
    errorCode: ErrorCode$outboundSchema,
    type: TypeWebSearchToolResultError$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        errorCode: "error_code",
    });
});
export function contentWebSearchToolResultErrorToJSON(contentWebSearchToolResultError) {
    return JSON.stringify(ContentWebSearchToolResultError$outboundSchema.parse(contentWebSearchToolResultError));
}
/** @internal */
export const MessagesMessageParamContentUnion3$outboundSchema = z.union([
    z.lazy(() => ContentWebSearchToolResultError$outboundSchema),
    z.array(AnthropicWebSearchResultBlockParam$outboundSchema),
]);
export function messagesMessageParamContentUnion3ToJSON(messagesMessageParamContentUnion3) {
    return JSON.stringify(MessagesMessageParamContentUnion3$outboundSchema.parse(messagesMessageParamContentUnion3));
}
/** @internal */
export const ContentWebSearchToolResult$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    content: z.union([
        z.lazy(() => ContentWebSearchToolResultError$outboundSchema),
        z.array(AnthropicWebSearchResultBlockParam$outboundSchema),
    ]),
    toolUseId: z.string(),
    type: z.literal("web_search_tool_result"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
        toolUseId: "tool_use_id",
    });
});
export function contentWebSearchToolResultToJSON(contentWebSearchToolResult) {
    return JSON.stringify(ContentWebSearchToolResult$outboundSchema.parse(contentWebSearchToolResult));
}
/** @internal */
export const ContentServerToolUse$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    id: z.string(),
    input: z.any().optional(),
    name: z.string(),
    type: z.literal("server_tool_use"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function contentServerToolUseToJSON(contentServerToolUse) {
    return JSON.stringify(ContentServerToolUse$outboundSchema.parse(contentServerToolUse));
}
/** @internal */
export const ContentRedactedThinking$outboundSchema = z.object({
    data: z.string(),
    type: z.literal("redacted_thinking"),
});
export function contentRedactedThinkingToJSON(contentRedactedThinking) {
    return JSON.stringify(ContentRedactedThinking$outboundSchema.parse(contentRedactedThinking));
}
/** @internal */
export const ContentThinking$outboundSchema = z.object({
    signature: z.string(),
    thinking: z.string(),
    type: z.literal("thinking"),
});
export function contentThinkingToJSON(contentThinking) {
    return JSON.stringify(ContentThinking$outboundSchema.parse(contentThinking));
}
/** @internal */
export const ContentToolReference$outboundSchema = z.object({
    toolName: z.string(),
    type: z.literal("tool_reference"),
}).transform((v) => {
    return remap$(v, {
        toolName: "tool_name",
    });
});
export function contentToolReferenceToJSON(contentToolReference) {
    return JSON.stringify(ContentToolReference$outboundSchema.parse(contentToolReference));
}
/** @internal */
export const MessagesMessageParamContentUnion1$outboundSchema = z.union([
    AnthropicTextBlockParam$outboundSchema,
    AnthropicImageBlockParam$outboundSchema,
    z.lazy(() => ContentToolReference$outboundSchema),
    AnthropicSearchResultBlockParam$outboundSchema,
    AnthropicDocumentBlockParam$outboundSchema,
]);
export function messagesMessageParamContentUnion1ToJSON(messagesMessageParamContentUnion1) {
    return JSON.stringify(MessagesMessageParamContentUnion1$outboundSchema.parse(messagesMessageParamContentUnion1));
}
/** @internal */
export const MessagesMessageParamContentUnion2$outboundSchema = z.union([
    z.string(),
    z.array(z.union([
        AnthropicTextBlockParam$outboundSchema,
        AnthropicImageBlockParam$outboundSchema,
        z.lazy(() => ContentToolReference$outboundSchema),
        AnthropicSearchResultBlockParam$outboundSchema,
        AnthropicDocumentBlockParam$outboundSchema,
    ])),
]);
export function messagesMessageParamContentUnion2ToJSON(messagesMessageParamContentUnion2) {
    return JSON.stringify(MessagesMessageParamContentUnion2$outboundSchema.parse(messagesMessageParamContentUnion2));
}
/** @internal */
export const ContentToolResult$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    content: z.union([
        z.string(),
        z.array(z.union([
            AnthropicTextBlockParam$outboundSchema,
            AnthropicImageBlockParam$outboundSchema,
            z.lazy(() => ContentToolReference$outboundSchema),
            AnthropicSearchResultBlockParam$outboundSchema,
            AnthropicDocumentBlockParam$outboundSchema,
        ])),
    ]).optional(),
    isError: z.boolean().optional(),
    toolUseId: z.string(),
    type: z.literal("tool_result"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
        isError: "is_error",
        toolUseId: "tool_use_id",
    });
});
export function contentToolResultToJSON(contentToolResult) {
    return JSON.stringify(ContentToolResult$outboundSchema.parse(contentToolResult));
}
/** @internal */
export const ContentToolUse$outboundSchema = z.object({
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    id: z.string(),
    input: z.any().optional(),
    name: z.string(),
    type: z.literal("tool_use"),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function contentToolUseToJSON(contentToolUse) {
    return JSON.stringify(ContentToolUse$outboundSchema.parse(contentToolUse));
}
/** @internal */
export const MessagesMessageParamContentUnion4$outboundSchema = z.union([
    AnthropicTextBlockParam$outboundSchema,
    AnthropicImageBlockParam$outboundSchema,
    AnthropicDocumentBlockParam$outboundSchema,
    z.lazy(() => ContentToolUse$outboundSchema),
    z.lazy(() => ContentToolResult$outboundSchema),
    z.lazy(() => ContentThinking$outboundSchema),
    z.lazy(() => ContentRedactedThinking$outboundSchema),
    z.lazy(() => ContentServerToolUse$outboundSchema),
    z.lazy(() => ContentWebSearchToolResult$outboundSchema),
    AnthropicSearchResultBlockParam$outboundSchema,
    z.lazy(() => ContentCompaction$outboundSchema),
    MessagesAdvisorToolResultBlock$outboundSchema,
]);
export function messagesMessageParamContentUnion4ToJSON(messagesMessageParamContentUnion4) {
    return JSON.stringify(MessagesMessageParamContentUnion4$outboundSchema.parse(messagesMessageParamContentUnion4));
}
/** @internal */
export const MessagesMessageParamContentUnion5$outboundSchema = z.union([
    z.string(),
    z.array(z.union([
        AnthropicTextBlockParam$outboundSchema,
        AnthropicImageBlockParam$outboundSchema,
        AnthropicDocumentBlockParam$outboundSchema,
        z.lazy(() => ContentToolUse$outboundSchema),
        z.lazy(() => ContentToolResult$outboundSchema),
        z.lazy(() => ContentThinking$outboundSchema),
        z.lazy(() => ContentRedactedThinking$outboundSchema),
        z.lazy(() => ContentServerToolUse$outboundSchema),
        z.lazy(() => ContentWebSearchToolResult$outboundSchema),
        AnthropicSearchResultBlockParam$outboundSchema,
        z.lazy(() => ContentCompaction$outboundSchema),
        MessagesAdvisorToolResultBlock$outboundSchema,
    ])),
]);
export function messagesMessageParamContentUnion5ToJSON(messagesMessageParamContentUnion5) {
    return JSON.stringify(MessagesMessageParamContentUnion5$outboundSchema.parse(messagesMessageParamContentUnion5));
}
/** @internal */
export const MessagesMessageParamRole$outboundSchema = openEnums.outboundSchema(MessagesMessageParamRole);
/** @internal */
export const MessagesMessageParam$outboundSchema = z.object({
    content: z.union([
        z.string(),
        z.array(z.union([
            AnthropicTextBlockParam$outboundSchema,
            AnthropicImageBlockParam$outboundSchema,
            AnthropicDocumentBlockParam$outboundSchema,
            z.lazy(() => ContentToolUse$outboundSchema),
            z.lazy(() => ContentToolResult$outboundSchema),
            z.lazy(() => ContentThinking$outboundSchema),
            z.lazy(() => ContentRedactedThinking$outboundSchema),
            z.lazy(() => ContentServerToolUse$outboundSchema),
            z.lazy(() => ContentWebSearchToolResult$outboundSchema),
            AnthropicSearchResultBlockParam$outboundSchema,
            z.lazy(() => ContentCompaction$outboundSchema),
            MessagesAdvisorToolResultBlock$outboundSchema,
        ])),
    ]),
    role: MessagesMessageParamRole$outboundSchema,
});
export function messagesMessageParamToJSON(messagesMessageParam) {
    return JSON.stringify(MessagesMessageParam$outboundSchema.parse(messagesMessageParam));
}
