// @openrouter/sdk/esm/models/messagesoutputconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: aad3082eb908
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
/**
 * How much effort the model should put into its response. Higher effort levels may result in more thorough analysis but take longer. Valid values are `low`, `medium`, `high`, `xhigh`, or `max`.
 */
export const MessagesOutputConfigEffort = {
    Low: "low",
    Medium: "medium",
    High: "high",
    Xhigh: "xhigh",
    Max: "max",
};
export const MessagesOutputConfigTypeJSONSchema = {
    JsonSchema: "json_schema",
};
export const TypeTokens = {
    Tokens: "tokens",
};
/** @internal */
export const MessagesOutputConfigEffort$outboundSchema = openEnums.outboundSchema(MessagesOutputConfigEffort);
/** @internal */
export const MessagesOutputConfigTypeJSONSchema$outboundSchema = z.enum(MessagesOutputConfigTypeJSONSchema);
/** @internal */
export const MessagesOutputConfigFormat$outboundSchema = z.object({
    schema: z.record(z.string(), z.any()),
    type: MessagesOutputConfigTypeJSONSchema$outboundSchema,
});
export function messagesOutputConfigFormatToJSON(messagesOutputConfigFormat) {
    return JSON.stringify(MessagesOutputConfigFormat$outboundSchema.parse(messagesOutputConfigFormat));
}
/** @internal */
export const TypeTokens$outboundSchema = z.enum(TypeTokens);
/** @internal */
export const TaskBudget$outboundSchema = z.object({
    remaining: z.nullable(z.int()).optional(),
    total: z.int(),
    type: TypeTokens$outboundSchema,
});
export function taskBudgetToJSON(taskBudget) {
    return JSON.stringify(TaskBudget$outboundSchema.parse(taskBudget));
}
/** @internal */
export const MessagesOutputConfig$outboundSchema = z.object({
    effort: z.nullable(MessagesOutputConfigEffort$outboundSchema).optional(),
    format: z.nullable(z.lazy(() => MessagesOutputConfigFormat$outboundSchema))
        .optional(),
    taskBudget: z.nullable(z.lazy(() => TaskBudget$outboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        taskBudget: "task_budget",
    });
});
export function messagesOutputConfigToJSON(messagesOutputConfig) {
    return JSON.stringify(MessagesOutputConfig$outboundSchema.parse(messagesOutputConfig));
}
