// @openrouter/sdk/esm/models/modelreasoning.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c7312e55417d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ReasoningEffort$inboundSchema, } from "./reasoningeffort.js";
/**
 * Default reasoning effort when the client enables reasoning without specifying effort. Maps to `reasoning.effort` in chat requests. When `"none"`, prefer omitting effort unless the user explicitly disables reasoning.
 */
export const DefaultEffort = {
    Max: "max",
    Xhigh: "xhigh",
    High: "high",
    Medium: "medium",
    Low: "low",
    Minimal: "minimal",
    None: "none",
};
/** @internal */
export const DefaultEffort$inboundSchema = openEnums.inboundSchema(DefaultEffort);
/** @internal */
export const ModelReasoning$inboundSchema = z.object({
    default_effort: z.nullable(DefaultEffort$inboundSchema).optional(),
    default_enabled: z.boolean().optional(),
    mandatory: z.boolean(),
    supported_efforts: z.nullable(z.array(z.nullable(ReasoningEffort$inboundSchema))).optional(),
    supports_max_tokens: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        "default_effort": "defaultEffort",
        "default_enabled": "defaultEnabled",
        "supported_efforts": "supportedEfforts",
        "supports_max_tokens": "supportsMaxTokens",
    });
});
export function modelReasoningFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ModelReasoning$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ModelReasoning' from JSON`);
}
