// @openrouter/sdk/esm/models/modelslistresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ea0e62a13d8d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { Model$inboundSchema } from "./model.js";
/** @internal */
export const Links$inboundSchema = z.object({
    next: z.nullable(z.string()),
});
export function linksFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Links$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Links' from JSON`);
}
/** @internal */
export const ModelsListResponse$inboundSchema = z.object({
    data: z.array(Model$inboundSchema),
    links: z.lazy(() => Links$inboundSchema),
    total_count: z.int(),
}).transform((v) => {
    return remap$(v, {
        "total_count": "totalCount",
    });
});
export function modelsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ModelsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ModelsListResponse' from JSON`);
}
