// @openrouter/sdk/esm/models/modelarchitecture.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 57607576095f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { InputModality$inboundSchema } from "./inputmodality.js";
import { InstructType$inboundSchema } from "./instructtype.js";
import { ModelGroup$inboundSchema } from "./modelgroup.js";
import { OutputModality$inboundSchema, } from "./outputmodality.js";
/** @internal */
export const ModelArchitecture$inboundSchema = z.object({
    input_modalities: z.array(InputModality$inboundSchema),
    instruct_type: z.nullable(InstructType$inboundSchema).optional(),
    modality: z.nullable(z.string()),
    output_modalities: z.array(OutputModality$inboundSchema),
    tokenizer: ModelGroup$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "input_modalities": "inputModalities",
        "instruct_type": "instructType",
        "output_modalities": "outputModalities",
    });
});
export function modelArchitectureFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ModelArchitecture$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ModelArchitecture' from JSON`);
}
