// @openrouter/sdk/esm/models/namespacefunctiontool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ca0cabda8e0a
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const AllowedCaller = {
    Direct: "direct",
    Programmatic: "programmatic",
};
/** @internal */
export const AllowedCaller$inboundSchema = openEnums.inboundSchema(AllowedCaller);
/** @internal */
export const AllowedCaller$outboundSchema = openEnums.outboundSchema(AllowedCaller);
/** @internal */
export const NamespaceFunctionTool$inboundSchema = z.object({
    allowed_callers: z.nullable(z.array(AllowedCaller$inboundSchema)).optional(),
    defer_loading: z.boolean().optional(),
    description: z.nullable(z.string()).optional(),
    name: z.string(),
    output_schema: z.nullable(z.record(z.string(), z.any())).optional(),
    parameters: z.nullable(z.record(z.string(), z.any())).optional(),
    strict: z.nullable(z.boolean()).optional(),
    type: z.literal("function"),
}).transform((v) => {
    return remap$(v, {
        "allowed_callers": "allowedCallers",
        "defer_loading": "deferLoading",
        "output_schema": "outputSchema",
    });
});
/** @internal */
export const NamespaceFunctionTool$outboundSchema = z.object({
    allowedCallers: z.nullable(z.array(AllowedCaller$outboundSchema)).optional(),
    deferLoading: z.boolean().optional(),
    description: z.nullable(z.string()).optional(),
    name: z.string(),
    outputSchema: z.nullable(z.record(z.string(), z.any())).optional(),
    parameters: z.nullable(z.record(z.string(), z.any())).optional(),
    strict: z.nullable(z.boolean()).optional(),
    type: z.literal("function"),
}).transform((v) => {
    return remap$(v, {
        allowedCallers: "allowed_callers",
        deferLoading: "defer_loading",
        outputSchema: "output_schema",
    });
});
export function namespaceFunctionToolToJSON(namespaceFunctionTool) {
    return JSON.stringify(NamespaceFunctionTool$outboundSchema.parse(namespaceFunctionTool));
}
export function namespaceFunctionToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NamespaceFunctionTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NamespaceFunctionTool' from JSON`);
}
