// @openrouter/sdk/esm/models/namespacetool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b430de21d6c4
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { CustomTool$inboundSchema, CustomTool$outboundSchema, } from "./customtool.js";
import { NamespaceFunctionTool$inboundSchema, NamespaceFunctionTool$outboundSchema, } from "./namespacefunctiontool.js";
export const NamespaceToolType = {
    Namespace: "namespace",
};
/** @internal */
export const NamespaceToolTool$inboundSchema = discriminatedUnion("type", {
    function: NamespaceFunctionTool$inboundSchema,
    custom: CustomTool$inboundSchema.and(z.object({ type: z.literal("custom") })),
});
/** @internal */
export const NamespaceToolTool$outboundSchema = z.union([
    NamespaceFunctionTool$outboundSchema,
    CustomTool$outboundSchema.and(z.object({ type: z.literal("custom") })),
]);
export function namespaceToolToolToJSON(namespaceToolTool) {
    return JSON.stringify(NamespaceToolTool$outboundSchema.parse(namespaceToolTool));
}
export function namespaceToolToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NamespaceToolTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NamespaceToolTool' from JSON`);
}
/** @internal */
export const NamespaceToolType$inboundSchema = z.enum(NamespaceToolType);
/** @internal */
export const NamespaceToolType$outboundSchema = NamespaceToolType$inboundSchema;
/** @internal */
export const NamespaceTool$inboundSchema = z
    .object({
    description: z.string(),
    name: z.string(),
    tools: z.array(discriminatedUnion("type", {
        function: NamespaceFunctionTool$inboundSchema,
        custom: CustomTool$inboundSchema.and(z.object({ type: z.literal("custom") })),
    })),
    type: NamespaceToolType$inboundSchema,
});
/** @internal */
export const NamespaceTool$outboundSchema = z.object({
    description: z.string(),
    name: z.string(),
    tools: z.array(z.union([
        NamespaceFunctionTool$outboundSchema,
        CustomTool$outboundSchema.and(z.object({ type: z.literal("custom") })),
    ])),
    type: NamespaceToolType$outboundSchema,
});
export function namespaceToolToJSON(namespaceTool) {
    return JSON.stringify(NamespaceTool$outboundSchema.parse(namespaceTool));
}
export function namespaceToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NamespaceTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NamespaceTool' from JSON`);
}
