// @openrouter/sdk/esm/models/observabilityopikdestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 96b0714087b1
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityOpikDestinationConfig$inboundSchema = z.object({
    apiKey: z.string(),
    headers: z.record(z.string(), z.string()).optional(),
    projectName: z.string(),
    workspace: z.string(),
});
export function observabilityOpikDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityOpikDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityOpikDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityOpikDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityOpikDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("opik"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityOpikDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityOpikDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityOpikDestination' from JSON`);
}
