// @openrouter/sdk/esm/models/observabilitywebhookdestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 934fbf633458
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
export const Method = {
    Post: "POST",
    Put: "PUT",
};
/** @internal */
export const Method$inboundSchema = openEnums
    .inboundSchema(Method);
/** @internal */
export const ObservabilityWebhookDestinationConfig$inboundSchema = z.object({
    headers: z.record(z.string(), z.string()).optional(),
    method: Method$inboundSchema.default("POST"),
    url: z.string(),
});
export function observabilityWebhookDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityWebhookDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityWebhookDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityWebhookDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityWebhookDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("webhook"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityWebhookDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityWebhookDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityWebhookDestination' from JSON`);
}
