// @openrouter/sdk/esm/models/observabilitys3destination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a9e50c2b3d64
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityS3DestinationConfig$inboundSchema = z.object({
    accessKeyId: z.string(),
    bucketName: z.string(),
    endpoint: z.string().optional(),
    headers: z.record(z.string(), z.string()).optional(),
    pathTemplate: z.string().default("{prefix}/{date}"),
    prefix: z.string().default("openrouter-traces"),
    region: z.string().optional(),
    secretAccessKey: z.string(),
    sessionToken: z.string().optional(),
});
export function observabilityS3DestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityS3DestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityS3DestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityS3Destination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityS3DestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("s3"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityS3DestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityS3Destination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityS3Destination' from JSON`);
}
