// @openrouter/sdk/esm/models/observabilitysnowflakedestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9b4968dabbd6
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilitySnowflakeDestinationConfig$inboundSchema = z.object({
    account: z.string(),
    database: z.string().default("SNOWFLAKE_LEARNING_DB"),
    headers: z.record(z.string(), z.string()).optional(),
    schema: z.string().default("PUBLIC"),
    table: z.string().default("OPENROUTER_TRACES"),
    token: z.string(),
    warehouse: z.string().default("COMPUTE_WH"),
});
export function observabilitySnowflakeDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilitySnowflakeDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilitySnowflakeDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilitySnowflakeDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilitySnowflakeDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("snowflake"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilitySnowflakeDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilitySnowflakeDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilitySnowflakeDestination' from JSON`);
}
