// @openrouter/sdk/esm/models/observabilityrampdestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3991fa386ded
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityRampDestinationConfig$inboundSchema = z.object({
    apiKey: z.string(),
    baseUrl: z.string().default("https://api.ramp.com/developer/v1/ai-usage/openrouter"),
    headers: z.record(z.string(), z.string()).optional(),
});
export function observabilityRampDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityRampDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityRampDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityRampDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityRampDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("ramp"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityRampDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityRampDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityRampDestination' from JSON`);
}
