// @openrouter/sdk/esm/models/observabilityotelcollectordestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a10e04b8d635
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityOtelCollectorDestinationConfig$inboundSchema = z.object({
    endpoint: z.string(),
    headers: z.record(z.string(), z.string()).optional(),
});
export function observabilityOtelCollectorDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityOtelCollectorDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityOtelCollectorDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityOtelCollectorDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityOtelCollectorDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("otel-collector"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityOtelCollectorDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityOtelCollectorDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityOtelCollectorDestination' from JSON`);
}
