// @openrouter/sdk/esm/models/observabilitydestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 701b467a91ab
 */
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { ObservabilityArizeDestination$inboundSchema, } from "./observabilityarizedestination.js";
import { ObservabilityBraintrustDestination$inboundSchema, } from "./observabilitybraintrustdestination.js";
import { ObservabilityClickhouseDestination$inboundSchema, } from "./observabilityclickhousedestination.js";
import { ObservabilityDatadogDestination$inboundSchema, } from "./observabilitydatadogdestination.js";
import { ObservabilityGrafanaDestination$inboundSchema, } from "./observabilitygrafanadestination.js";
import { ObservabilityLangfuseDestination$inboundSchema, } from "./observabilitylangfusedestination.js";
import { ObservabilityLangsmithDestination$inboundSchema, } from "./observabilitylangsmithdestination.js";
import { ObservabilityNewrelicDestination$inboundSchema, } from "./observabilitynewrelicdestination.js";
import { ObservabilityOpikDestination$inboundSchema, } from "./observabilityopikdestination.js";
import { ObservabilityOtelCollectorDestination$inboundSchema, } from "./observabilityotelcollectordestination.js";
import { ObservabilityPosthogDestination$inboundSchema, } from "./observabilityposthogdestination.js";
import { ObservabilityRampDestination$inboundSchema, } from "./observabilityrampdestination.js";
import { ObservabilityS3Destination$inboundSchema, } from "./observabilitys3destination.js";
import { ObservabilitySentryDestination$inboundSchema, } from "./observabilitysentrydestination.js";
import { ObservabilitySnowflakeDestination$inboundSchema, } from "./observabilitysnowflakedestination.js";
import { ObservabilityWeaveDestination$inboundSchema, } from "./observabilityweavedestination.js";
import { ObservabilityWebhookDestination$inboundSchema, } from "./observabilitywebhookdestination.js";
/** @internal */
export const ObservabilityDestination$inboundSchema = discriminatedUnion("type", {
    arize: ObservabilityArizeDestination$inboundSchema,
    braintrust: ObservabilityBraintrustDestination$inboundSchema,
    clickhouse: ObservabilityClickhouseDestination$inboundSchema,
    datadog: ObservabilityDatadogDestination$inboundSchema,
    grafana: ObservabilityGrafanaDestination$inboundSchema,
    langfuse: ObservabilityLangfuseDestination$inboundSchema,
    langsmith: ObservabilityLangsmithDestination$inboundSchema,
    newrelic: ObservabilityNewrelicDestination$inboundSchema,
    opik: ObservabilityOpikDestination$inboundSchema,
    ["otel-collector"]: ObservabilityOtelCollectorDestination$inboundSchema,
    posthog: ObservabilityPosthogDestination$inboundSchema,
    ramp: ObservabilityRampDestination$inboundSchema,
    s3: ObservabilityS3Destination$inboundSchema,
    sentry: ObservabilitySentryDestination$inboundSchema,
    snowflake: ObservabilitySnowflakeDestination$inboundSchema,
    weave: ObservabilityWeaveDestination$inboundSchema,
    webhook: ObservabilityWebhookDestination$inboundSchema,
});
export function observabilityDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityDestination' from JSON`);
}
