// @openrouter/sdk/esm/models/observabilityclickhousedestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 628ce2e2989d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityClickhouseDestinationConfig$inboundSchema = z.object({
    database: z.string(),
    headers: z.record(z.string(), z.string()).optional(),
    host: z.string(),
    password: z.string(),
    table: z.string().default("OPENROUTER_TRACES"),
    username: z.string(),
});
export function observabilityClickhouseDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityClickhouseDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityClickhouseDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityClickhouseDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityClickhouseDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("clickhouse"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityClickhouseDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityClickhouseDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityClickhouseDestination' from JSON`);
}
