// @openrouter/sdk/esm/models/observabilityarizedestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 631c7deb0d5e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityArizeDestinationConfig$inboundSchema = z.object({
    apiKey: z.string(),
    baseUrl: z.string().default("https://otlp.arize.com"),
    headers: z.record(z.string(), z.string()).optional(),
    modelId: z.string(),
    spaceKey: z.string(),
});
export function observabilityArizeDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityArizeDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityArizeDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityArizeDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityArizeDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("arize"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityArizeDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityArizeDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityArizeDestination' from JSON`);
}
