// @openrouter/sdk/esm/models/observabilityweavedestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: beb41ac4527c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityWeaveDestinationConfig$inboundSchema = z.object({
    apiKey: z.string(),
    baseUrl: z.string().default("https://trace.wandb.ai"),
    entity: z.string(),
    headers: z.record(z.string(), z.string()).optional(),
    project: z.string(),
});
export function observabilityWeaveDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityWeaveDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityWeaveDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityWeaveDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityWeaveDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("weave"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityWeaveDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityWeaveDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityWeaveDestination' from JSON`);
}
