// @openrouter/sdk/esm/models/observabilitybraintrustdestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 37b935380e25
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityBraintrustDestinationConfig$inboundSchema = z.object({
    apiKey: z.string(),
    baseUrl: z.string().default("https://api.braintrust.dev"),
    headers: z.record(z.string(), z.string()).optional(),
    projectId: z.string(),
});
export function observabilityBraintrustDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityBraintrustDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityBraintrustDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityBraintrustDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityBraintrustDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("braintrust"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityBraintrustDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityBraintrustDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityBraintrustDestination' from JSON`);
}
