// @openrouter/sdk/esm/models/observabilityposthogdestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7b13c8a68c92
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityPosthogDestinationConfig$inboundSchema = z.object({
    apiKey: z.string(),
    endpoint: z.string().default("https://us.i.posthog.com"),
    headers: z.record(z.string(), z.string()).optional(),
});
export function observabilityPosthogDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityPosthogDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityPosthogDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityPosthogDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityPosthogDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("posthog"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityPosthogDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityPosthogDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityPosthogDestination' from JSON`);
}
