// @openrouter/sdk/esm/models/observabilitydatadogdestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 791210a608e6
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityDatadogDestinationConfig$inboundSchema = z.object({
    apiKey: z.string(),
    headers: z.record(z.string(), z.string()).optional(),
    mlApp: z.string(),
    url: z.string().default("https://api.us5.datadoghq.com"),
});
export function observabilityDatadogDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityDatadogDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityDatadogDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityDatadogDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityDatadogDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("datadog"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityDatadogDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityDatadogDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityDatadogDestination' from JSON`);
}
