// @openrouter/sdk/esm/models/observabilitylangfusedestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 866372fcdbce
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityLangfuseDestinationConfig$inboundSchema = z.object({
    baseUrl: z.string().default("https://us.cloud.langfuse.com"),
    headers: z.record(z.string(), z.string()).optional(),
    publicKey: z.string(),
    secretKey: z.string(),
});
export function observabilityLangfuseDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityLangfuseDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityLangfuseDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityLangfuseDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityLangfuseDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("langfuse"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityLangfuseDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityLangfuseDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityLangfuseDestination' from JSON`);
}
