// @openrouter/sdk/esm/models/observabilitygrafanadestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 26c17318c1d5
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
/** @internal */
export const ObservabilityGrafanaDestinationConfig$inboundSchema = z.object({
    apiKey: z.string(),
    baseUrl: z.string().default("https://otlp-gateway-prod-us-west-0.grafana.net"),
    headers: z.record(z.string(), z.string()).optional(),
    instanceId: z.string(),
});
export function observabilityGrafanaDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityGrafanaDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityGrafanaDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityGrafanaDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityGrafanaDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("grafana"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityGrafanaDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityGrafanaDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityGrafanaDestination' from JSON`);
}
