// @openrouter/sdk/esm/models/observabilitynewrelicdestination.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 49ee9e98072a
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ObservabilityFilterRulesConfig$inboundSchema, } from "./observabilityfilterrulesconfig.js";
export const Region = {
    Us: "us",
    Eu: "eu",
};
/** @internal */
export const Region$inboundSchema = openEnums
    .inboundSchema(Region);
/** @internal */
export const ObservabilityNewrelicDestinationConfig$inboundSchema = z.object({
    headers: z.record(z.string(), z.string()).optional(),
    licenseKey: z.string(),
    region: Region$inboundSchema.default("us"),
});
export function observabilityNewrelicDestinationConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityNewrelicDestinationConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityNewrelicDestinationConfig' from JSON`);
}
/** @internal */
export const ObservabilityNewrelicDestination$inboundSchema = z.object({
    api_key_hashes: z.nullable(z.array(z.string())),
    config: z.lazy(() => ObservabilityNewrelicDestinationConfig$inboundSchema),
    created_at: z.string(),
    enabled: z.boolean(),
    filter_rules: z.nullable(ObservabilityFilterRulesConfig$inboundSchema),
    id: z.string(),
    name: z.nullable(z.string()),
    privacy_mode: z.boolean(),
    sampling_rate: z.number(),
    type: z.literal("newrelic"),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "api_key_hashes": "apiKeyHashes",
        "created_at": "createdAt",
        "filter_rules": "filterRules",
        "privacy_mode": "privacyMode",
        "sampling_rate": "samplingRate",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function observabilityNewrelicDestinationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ObservabilityNewrelicDestination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObservabilityNewrelicDestination' from JSON`);
}
