// @openrouter/sdk/esm/models/openroutermetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8a5688777dfa
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { EndpointsMetadata$inboundSchema, } from "./endpointsmetadata.js";
import { PipelineStage$inboundSchema } from "./pipelinestage.js";
import { RouterAttempt$inboundSchema } from "./routerattempt.js";
import { RouterParams$inboundSchema } from "./routerparams.js";
import { RoutingStrategy$inboundSchema, } from "./routingstrategy.js";
/** @internal */
export const OpenRouterMetadata$inboundSchema = z.object({
    attempt: z.int(),
    attempts: z.array(RouterAttempt$inboundSchema).optional(),
    endpoints: EndpointsMetadata$inboundSchema,
    is_byok: z.boolean(),
    params: RouterParams$inboundSchema.optional(),
    pipeline: z.array(PipelineStage$inboundSchema).optional(),
    region: z.nullable(z.string()),
    requested: z.string(),
    strategy: RoutingStrategy$inboundSchema,
    summary: z.string(),
}).transform((v) => {
    return remap$(v, {
        "is_byok": "isByok",
    });
});
export function openRouterMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenRouterMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenRouterMetadata' from JSON`);
}
