// @openrouter/sdk/esm/models/openairesponsecustomtoolcalloutput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7ce70f788343
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { InputFile$inboundSchema } from "./inputfile.js";
import { InputImage$inboundSchema } from "./inputimage.js";
import { InputText$inboundSchema } from "./inputtext.js";
export const OpenAIResponseCustomToolCallOutputType = {
    CustomToolCallOutput: "custom_tool_call_output",
};
/** @internal */
export const OpenAIResponseCustomToolCallOutputOutput1$inboundSchema = discriminatedUnion("type", {
    input_file: InputFile$inboundSchema,
    input_image: InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
    input_text: InputText$inboundSchema,
});
export function openAIResponseCustomToolCallOutputOutput1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponseCustomToolCallOutputOutput1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponseCustomToolCallOutputOutput1' from JSON`);
}
/** @internal */
export const OpenAIResponseCustomToolCallOutputOutput2$inboundSchema = z.union([
    z.string(),
    z.array(discriminatedUnion("type", {
        input_file: InputFile$inboundSchema,
        input_image: InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
        input_text: InputText$inboundSchema,
    })),
]);
export function openAIResponseCustomToolCallOutputOutput2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponseCustomToolCallOutputOutput2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponseCustomToolCallOutputOutput2' from JSON`);
}
/** @internal */
export const OpenAIResponseCustomToolCallOutputType$inboundSchema = z.enum(OpenAIResponseCustomToolCallOutputType);
/** @internal */
export const OpenAIResponseCustomToolCallOutput$inboundSchema = z.object({
    call_id: z.string(),
    id: z.string().optional(),
    output: z.union([
        z.string(),
        z.array(discriminatedUnion("type", {
            input_file: InputFile$inboundSchema,
            input_image: InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
            input_text: InputText$inboundSchema,
        })),
    ]),
    type: OpenAIResponseCustomToolCallOutputType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
export function openAIResponseCustomToolCallOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponseCustomToolCallOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponseCustomToolCallOutput' from JSON`);
}
