// @openrouter/sdk/esm/models/openairesponsecustomtoolcall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: af21a6dcaaac
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const OpenAIResponseCustomToolCallType = {
    CustomToolCall: "custom_tool_call",
};
/** @internal */
export const OpenAIResponseCustomToolCallType$inboundSchema = z.enum(OpenAIResponseCustomToolCallType);
/** @internal */
export const OpenAIResponseCustomToolCall$inboundSchema = z.object({
    call_id: z.string(),
    id: z.string().optional(),
    input: z.string(),
    name: z.string(),
    namespace: z.string().optional(),
    type: OpenAIResponseCustomToolCallType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
export function openAIResponseCustomToolCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponseCustomToolCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponseCustomToolCall' from JSON`);
}
