// @openrouter/sdk/esm/models/openresponsesresult.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 95391b93fcbf
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { ApiErrorType$inboundSchema } from "./apierrortype.js";
import { ApplyPatchServerTool$inboundSchema, } from "./applypatchservertool.js";
import { BaseInputsUnion$inboundSchema, } from "./baseinputsunion.js";
import { BaseReasoningConfig$inboundSchema, } from "./basereasoningconfig.js";
import { CodeInterpreterServerTool$inboundSchema, } from "./codeinterpreterservertool.js";
import { CodexLocalShellTool$inboundSchema, } from "./codexlocalshelltool.js";
import { ComputerUseServerTool$inboundSchema, } from "./computeruseservertool.js";
import { CustomTool$inboundSchema } from "./customtool.js";
import { FileSearchServerTool$inboundSchema, } from "./filesearchservertool.js";
import { ImageGenerationServerTool$inboundSchema, } from "./imagegenerationservertool.js";
import { IncompleteDetails$inboundSchema, } from "./incompletedetails.js";
import { LegacyWebSearchServerTool$inboundSchema, } from "./legacywebsearchservertool.js";
import { McpServerTool$inboundSchema } from "./mcpservertool.js";
import { NamespaceTool$inboundSchema } from "./namespacetool.js";
import { OpenAIResponsesResponseStatus$inboundSchema, } from "./openairesponsesresponsestatus.js";
import { OpenAIResponsesToolChoiceUnion$inboundSchema, } from "./openairesponsestoolchoiceunion.js";
import { OpenRouterMetadata$inboundSchema, } from "./openroutermetadata.js";
import { OutputItems$inboundSchema } from "./outputitems.js";
import { Preview20250311WebSearchServerTool$inboundSchema, } from "./preview20250311websearchservertool.js";
import { PreviewWebSearchServerTool$inboundSchema, } from "./previewwebsearchservertool.js";
import { PromptCacheOptions$inboundSchema, } from "./promptcacheoptions.js";
import { ResponsesErrorField$inboundSchema, } from "./responseserrorfield.js";
import { ServerToolUseDetails$inboundSchema, } from "./servertoolusedetails.js";
import { ShellServerTool$inboundSchema, } from "./shellservertool.js";
import { StoredPromptTemplate$inboundSchema, } from "./storedprompttemplate.js";
import { TextExtendedConfig$inboundSchema, } from "./textextendedconfig.js";
import { Truncation$inboundSchema } from "./truncation.js";
import { WebSearchServerTool$inboundSchema, } from "./websearchservertool.js";
export const OpenResponsesResultObject = {
    Response: "response",
};
/** @internal */
export const OpenResponsesResultObject$inboundSchema = z.enum(OpenResponsesResultObject);
/** @internal */
export const OpenResponsesResultToolFunction$inboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    name: z.string(),
    parameters: z.nullable(z.record(z.string(), z.any())),
    strict: z.nullable(z.boolean()).optional(),
    type: z.literal("function"),
});
export function openResponsesResultToolFunctionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesResultToolFunction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesResultToolFunction' from JSON`);
}
/** @internal */
export const OpenResponsesResultToolUnion$inboundSchema = discriminatedUnion("type", {
    function: z.lazy(() => OpenResponsesResultToolFunction$inboundSchema),
    web_search_preview: PreviewWebSearchServerTool$inboundSchema.and(z.object({ type: z.literal("web_search_preview") })),
    web_search_preview_2025_03_11: Preview20250311WebSearchServerTool$inboundSchema.and(z.object({ type: z.literal("web_search_preview_2025_03_11") })),
    web_search: LegacyWebSearchServerTool$inboundSchema.and(z.object({ type: z.literal("web_search") })),
    web_search_2025_08_26: WebSearchServerTool$inboundSchema.and(z.object({ type: z.literal("web_search_2025_08_26") })),
    file_search: FileSearchServerTool$inboundSchema.and(z.object({ type: z.literal("file_search") })),
    computer_use_preview: ComputerUseServerTool$inboundSchema.and(z.object({ type: z.literal("computer_use_preview") })),
    code_interpreter: CodeInterpreterServerTool$inboundSchema.and(z.object({ type: z.literal("code_interpreter") })),
    mcp: McpServerTool$inboundSchema.and(z.object({ type: z.literal("mcp") })),
    image_generation: ImageGenerationServerTool$inboundSchema.and(z.object({ type: z.literal("image_generation") })),
    local_shell: CodexLocalShellTool$inboundSchema.and(z.object({ type: z.literal("local_shell") })),
    shell: ShellServerTool$inboundSchema.and(z.object({ type: z.literal("shell") })),
    apply_patch: ApplyPatchServerTool$inboundSchema.and(z.object({ type: z.literal("apply_patch") })),
    custom: CustomTool$inboundSchema.and(z.object({ type: z.literal("custom") })),
    namespace: NamespaceTool$inboundSchema.and(z.object({ type: z.literal("namespace") })),
});
export function openResponsesResultToolUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesResultToolUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesResultToolUnion' from JSON`);
}
/** @internal */
export const InputTokensDetails$inboundSchema = z.object({
    cache_write_tokens: z.nullable(z.int()).optional(),
    cached_tokens: z.int(),
}).transform((v) => {
    return remap$(v, {
        "cache_write_tokens": "cacheWriteTokens",
        "cached_tokens": "cachedTokens",
    });
});
export function inputTokensDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InputTokensDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InputTokensDetails' from JSON`);
}
/** @internal */
export const OutputTokensDetails$inboundSchema = z.object({
    reasoning_tokens: z.int(),
}).transform((v) => {
    return remap$(v, {
        "reasoning_tokens": "reasoningTokens",
    });
});
export function outputTokensDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputTokensDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputTokensDetails' from JSON`);
}
/** @internal */
export const UsageCostDetails$inboundSchema = z.object({
    upstream_inference_cost: z.nullable(z.number()).optional(),
    upstream_inference_input_cost: z.number(),
    upstream_inference_output_cost: z.number(),
}).transform((v) => {
    return remap$(v, {
        "upstream_inference_cost": "upstreamInferenceCost",
        "upstream_inference_input_cost": "upstreamInferenceInputCost",
        "upstream_inference_output_cost": "upstreamInferenceOutputCost",
    });
});
export function usageCostDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UsageCostDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UsageCostDetails' from JSON`);
}
/** @internal */
export const Usage$inboundSchema = z.object({
    input_tokens: z.int(),
    input_tokens_details: z.lazy(() => InputTokensDetails$inboundSchema),
    output_tokens: z.int(),
    output_tokens_details: z.lazy(() => OutputTokensDetails$inboundSchema),
    total_tokens: z.int(),
    cost: z.nullable(z.number()).optional(),
    cost_details: z.lazy(() => UsageCostDetails$inboundSchema).optional(),
    is_byok: z.boolean().optional(),
    server_tool_use_details: z.nullable(ServerToolUseDetails$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "input_tokens": "inputTokens",
        "input_tokens_details": "inputTokensDetails",
        "output_tokens": "outputTokens",
        "output_tokens_details": "outputTokensDetails",
        "total_tokens": "totalTokens",
        "cost_details": "costDetails",
        "is_byok": "isByok",
        "server_tool_use_details": "serverToolUseDetails",
    });
});
export function usageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Usage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Usage' from JSON`);
}
/** @internal */
export const OpenResponsesResult$inboundSchema = z.object({
    background: z.nullable(z.boolean()).optional(),
    completed_at: z.nullable(z.int()),
    created_at: z.int(),
    error: z.nullable(ResponsesErrorField$inboundSchema),
    frequency_penalty: z.nullable(z.number()),
    id: z.string(),
    incomplete_details: z.nullable(IncompleteDetails$inboundSchema),
    instructions: z.nullable(BaseInputsUnion$inboundSchema),
    max_output_tokens: z.nullable(z.int()).optional(),
    max_tool_calls: z.nullable(z.int()).optional(),
    metadata: z.nullable(z.record(z.string(), z.string())),
    model: z.string(),
    object: OpenResponsesResultObject$inboundSchema,
    output: z.array(OutputItems$inboundSchema),
    output_text: z.string().optional(),
    parallel_tool_calls: z.boolean(),
    presence_penalty: z.nullable(z.number()),
    previous_response_id: z.nullable(z.string()).optional(),
    prompt: z.nullable(StoredPromptTemplate$inboundSchema).optional(),
    prompt_cache_key: z.nullable(z.string()).optional(),
    prompt_cache_options: z.nullable(PromptCacheOptions$inboundSchema).optional(),
    reasoning: z.nullable(BaseReasoningConfig$inboundSchema).optional(),
    safety_identifier: z.nullable(z.string()).optional(),
    service_tier: z.nullable(z.string()).optional(),
    status: OpenAIResponsesResponseStatus$inboundSchema,
    store: z.boolean().optional(),
    temperature: z.nullable(z.number()),
    text: TextExtendedConfig$inboundSchema.optional(),
    tool_choice: OpenAIResponsesToolChoiceUnion$inboundSchema,
    tools: z.array(discriminatedUnion("type", {
        function: z.lazy(() => OpenResponsesResultToolFunction$inboundSchema),
        web_search_preview: PreviewWebSearchServerTool$inboundSchema.and(z.object({ type: z.literal("web_search_preview") })),
        web_search_preview_2025_03_11: Preview20250311WebSearchServerTool$inboundSchema.and(z.object({ type: z.literal("web_search_preview_2025_03_11") })),
        web_search: LegacyWebSearchServerTool$inboundSchema.and(z.object({ type: z.literal("web_search") })),
        web_search_2025_08_26: WebSearchServerTool$inboundSchema.and(z.object({ type: z.literal("web_search_2025_08_26") })),
        file_search: FileSearchServerTool$inboundSchema.and(z.object({ type: z.literal("file_search") })),
        computer_use_preview: ComputerUseServerTool$inboundSchema.and(z.object({ type: z.literal("computer_use_preview") })),
        code_interpreter: CodeInterpreterServerTool$inboundSchema.and(z.object({ type: z.literal("code_interpreter") })),
        mcp: McpServerTool$inboundSchema.and(z.object({ type: z.literal("mcp") })),
        image_generation: ImageGenerationServerTool$inboundSchema.and(z.object({ type: z.literal("image_generation") })),
        local_shell: CodexLocalShellTool$inboundSchema.and(z.object({ type: z.literal("local_shell") })),
        shell: ShellServerTool$inboundSchema.and(z.object({ type: z.literal("shell") })),
        apply_patch: ApplyPatchServerTool$inboundSchema.and(z.object({ type: z.literal("apply_patch") })),
        custom: CustomTool$inboundSchema.and(z.object({ type: z.literal("custom") })),
        namespace: NamespaceTool$inboundSchema.and(z.object({ type: z.literal("namespace") })),
    })),
    top_logprobs: z.nullable(z.int()).optional(),
    top_p: z.nullable(z.number()),
    truncation: z.nullable(Truncation$inboundSchema).optional(),
    usage: z.nullable(z.lazy(() => Usage$inboundSchema)).optional(),
    user: z.nullable(z.string()).optional(),
    error_type: ApiErrorType$inboundSchema.optional(),
    openrouter_metadata: OpenRouterMetadata$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "completed_at": "completedAt",
        "created_at": "createdAt",
        "frequency_penalty": "frequencyPenalty",
        "incomplete_details": "incompleteDetails",
        "max_output_tokens": "maxOutputTokens",
        "max_tool_calls": "maxToolCalls",
        "output_text": "outputText",
        "parallel_tool_calls": "parallelToolCalls",
        "presence_penalty": "presencePenalty",
        "previous_response_id": "previousResponseId",
        "prompt_cache_key": "promptCacheKey",
        "prompt_cache_options": "promptCacheOptions",
        "safety_identifier": "safetyIdentifier",
        "service_tier": "serviceTier",
        "tool_choice": "toolChoice",
        "top_logprobs": "topLogprobs",
        "top_p": "topP",
        "error_type": "errorType",
        "openrouter_metadata": "openrouterMetadata",
    });
});
export function openResponsesResultFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesResult$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesResult' from JSON`);
}
