// @openrouter/sdk/esm/models/outputwebfetchservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a28b106c9b49
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputWebFetchServerToolItemType = {
    OpenrouterWebFetch: "openrouter:web_fetch",
};
/** @internal */
export const OutputWebFetchServerToolItemType$inboundSchema = z.enum(OutputWebFetchServerToolItemType);
/** @internal */
export const OutputWebFetchServerToolItemType$outboundSchema = OutputWebFetchServerToolItemType$inboundSchema;
/** @internal */
export const OutputWebFetchServerToolItem$inboundSchema = z.object({
    content: z.string().optional(),
    error: z.string().optional(),
    httpStatus: z.int().optional(),
    id: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    title: z.string().optional(),
    type: OutputWebFetchServerToolItemType$inboundSchema,
    url: z.string().optional(),
});
/** @internal */
export const OutputWebFetchServerToolItem$outboundSchema = z.object({
    content: z.string().optional(),
    error: z.string().optional(),
    httpStatus: z.int().optional(),
    id: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    title: z.string().optional(),
    type: OutputWebFetchServerToolItemType$outboundSchema,
    url: z.string().optional(),
});
export function outputWebFetchServerToolItemToJSON(outputWebFetchServerToolItem) {
    return JSON.stringify(OutputWebFetchServerToolItem$outboundSchema.parse(outputWebFetchServerToolItem));
}
export function outputWebFetchServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputWebFetchServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputWebFetchServerToolItem' from JSON`);
}
