// @openrouter/sdk/esm/models/outputshellcalloutputitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a933ffbb5346
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { ShellCallStatus$inboundSchema, } from "./shellcallstatus.js";
/** @internal */
export const OutcomeTimeout$inboundSchema = z.object({
    type: z.literal("timeout"),
});
export function outcomeTimeoutFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutcomeTimeout$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutcomeTimeout' from JSON`);
}
/** @internal */
export const OutcomeExit$inboundSchema = z
    .object({
    exit_code: z.int(),
    type: z.literal("exit"),
}).transform((v) => {
    return remap$(v, {
        "exit_code": "exitCode",
    });
});
export function outcomeExitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutcomeExit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutcomeExit' from JSON`);
}
/** @internal */
export const Outcome$inboundSchema = discriminatedUnion("type", {
    exit: z.lazy(() => OutcomeExit$inboundSchema),
    timeout: z.lazy(() => OutcomeTimeout$inboundSchema),
});
export function outcomeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Outcome$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Outcome' from JSON`);
}
/** @internal */
export const OutputShellCallOutputItemOutput$inboundSchema = z.object({
    outcome: discriminatedUnion("type", {
        exit: z.lazy(() => OutcomeExit$inboundSchema),
        timeout: z.lazy(() => OutcomeTimeout$inboundSchema),
    }),
    stderr: z.string(),
    stdout: z.string(),
});
export function outputShellCallOutputItemOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputShellCallOutputItemOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputShellCallOutputItemOutput' from JSON`);
}
/** @internal */
export const OutputShellCallOutputItem$inboundSchema = z.object({
    call_id: z.string(),
    id: z.string(),
    max_output_length: z.nullable(z.int()).optional(),
    output: z.array(z.lazy(() => OutputShellCallOutputItemOutput$inboundSchema)),
    status: ShellCallStatus$inboundSchema,
    type: z.literal("shell_call_output"),
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
        "max_output_length": "maxOutputLength",
    });
});
export function outputShellCallOutputItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputShellCallOutputItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputShellCallOutputItem' from JSON`);
}
