// @openrouter/sdk/esm/models/outputapplypatchservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 761046712899
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ApplyPatchCallOperation$inboundSchema, ApplyPatchCallOperation$outboundSchema, } from "./applypatchcalloperation.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputApplyPatchServerToolItemType = {
    OpenrouterApplyPatch: "openrouter:apply_patch",
};
/** @internal */
export const OutputApplyPatchServerToolItemType$inboundSchema = z.enum(OutputApplyPatchServerToolItemType);
/** @internal */
export const OutputApplyPatchServerToolItemType$outboundSchema = OutputApplyPatchServerToolItemType$inboundSchema;
/** @internal */
export const OutputApplyPatchServerToolItem$inboundSchema = z.object({
    call_id: z.string().optional(),
    id: z.string().optional(),
    operation: ApplyPatchCallOperation$inboundSchema.optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputApplyPatchServerToolItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
/** @internal */
export const OutputApplyPatchServerToolItem$outboundSchema = z.object({
    callId: z.string().optional(),
    id: z.string().optional(),
    operation: ApplyPatchCallOperation$outboundSchema.optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputApplyPatchServerToolItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function outputApplyPatchServerToolItemToJSON(outputApplyPatchServerToolItem) {
    return JSON.stringify(OutputApplyPatchServerToolItem$outboundSchema.parse(outputApplyPatchServerToolItem));
}
export function outputApplyPatchServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputApplyPatchServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputApplyPatchServerToolItem' from JSON`);
}
