// @openrouter/sdk/esm/models/outputmemoryservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2fb051c8b4e9
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const ActionEnum = {
    Read: "read",
    Write: "write",
    Delete: "delete",
};
export const OutputMemoryServerToolItemType = {
    OpenrouterMemory: "openrouter:memory",
};
/** @internal */
export const ActionEnum$inboundSchema = openEnums.inboundSchema(ActionEnum);
/** @internal */
export const ActionEnum$outboundSchema = openEnums.outboundSchema(ActionEnum);
/** @internal */
export const OutputMemoryServerToolItemType$inboundSchema = z.enum(OutputMemoryServerToolItemType);
/** @internal */
export const OutputMemoryServerToolItemType$outboundSchema = OutputMemoryServerToolItemType$inboundSchema;
/** @internal */
export const OutputMemoryServerToolItem$inboundSchema = z.object({
    action: ActionEnum$inboundSchema.optional(),
    id: z.string().optional(),
    key: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputMemoryServerToolItemType$inboundSchema,
    value: z.any().optional(),
});
/** @internal */
export const OutputMemoryServerToolItem$outboundSchema = z.object({
    action: ActionEnum$outboundSchema.optional(),
    id: z.string().optional(),
    key: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputMemoryServerToolItemType$outboundSchema,
    value: z.any().optional(),
});
export function outputMemoryServerToolItemToJSON(outputMemoryServerToolItem) {
    return JSON.stringify(OutputMemoryServerToolItem$outboundSchema.parse(outputMemoryServerToolItem));
}
export function outputMemoryServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMemoryServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMemoryServerToolItem' from JSON`);
}
