// @openrouter/sdk/esm/models/outputfusionservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 30bcfe965217
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { FusionAnalysisResult$inboundSchema, FusionAnalysisResult$outboundSchema, } from "./fusionanalysisresult.js";
import { FusionSource$inboundSchema, FusionSource$outboundSchema, } from "./fusionsource.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputFusionServerToolItemType = {
    OpenrouterFusion: "openrouter:fusion",
};
/** @internal */
export const FailedModel$inboundSchema = z
    .object({
    error: z.string(),
    model: z.string(),
    status_code: z.int().optional(),
}).transform((v) => {
    return remap$(v, {
        "status_code": "statusCode",
    });
});
/** @internal */
export const FailedModel$outboundSchema = z.object({
    error: z.string(),
    model: z.string(),
    statusCode: z.int().optional(),
}).transform((v) => {
    return remap$(v, {
        statusCode: "status_code",
    });
});
export function failedModelToJSON(failedModel) {
    return JSON.stringify(FailedModel$outboundSchema.parse(failedModel));
}
export function failedModelFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FailedModel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FailedModel' from JSON`);
}
/** @internal */
export const ResponseT$inboundSchema = z.object({
    content: z.string().optional(),
    model: z.string(),
});
/** @internal */
export const ResponseT$outboundSchema = z.object({
    content: z.string().optional(),
    model: z.string(),
});
export function responseToJSON(responseT) {
    return JSON.stringify(ResponseT$outboundSchema.parse(responseT));
}
export function responseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseT' from JSON`);
}
/** @internal */
export const OutputFusionServerToolItemType$inboundSchema = z.enum(OutputFusionServerToolItemType);
/** @internal */
export const OutputFusionServerToolItemType$outboundSchema = OutputFusionServerToolItemType$inboundSchema;
/** @internal */
export const OutputFusionServerToolItem$inboundSchema = z.object({
    analysis: FusionAnalysisResult$inboundSchema.optional(),
    error: z.string().optional(),
    failed_models: z.array(z.lazy(() => FailedModel$inboundSchema)).optional(),
    failure_reason: z.string().optional(),
    id: z.string().optional(),
    responses: z.array(z.lazy(() => ResponseT$inboundSchema)).optional(),
    sources: z.array(FusionSource$inboundSchema).optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputFusionServerToolItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "failed_models": "failedModels",
        "failure_reason": "failureReason",
    });
});
/** @internal */
export const OutputFusionServerToolItem$outboundSchema = z.object({
    analysis: FusionAnalysisResult$outboundSchema.optional(),
    error: z.string().optional(),
    failedModels: z.array(z.lazy(() => FailedModel$outboundSchema)).optional(),
    failureReason: z.string().optional(),
    id: z.string().optional(),
    responses: z.array(z.lazy(() => ResponseT$outboundSchema)).optional(),
    sources: z.array(FusionSource$outboundSchema).optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputFusionServerToolItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        failedModels: "failed_models",
        failureReason: "failure_reason",
    });
});
export function outputFusionServerToolItemToJSON(outputFusionServerToolItem) {
    return JSON.stringify(OutputFusionServerToolItem$outboundSchema.parse(outputFusionServerToolItem));
}
export function outputFusionServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputFusionServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputFusionServerToolItem' from JSON`);
}
