// @openrouter/sdk/esm/models/outputsubagentservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9161bc044a98
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputSubagentServerToolItemType = {
    OpenrouterSubagent: "openrouter:subagent",
};
/** @internal */
export const OutputSubagentServerToolItemType$inboundSchema = z.enum(OutputSubagentServerToolItemType);
/** @internal */
export const OutputSubagentServerToolItemType$outboundSchema = OutputSubagentServerToolItemType$inboundSchema;
/** @internal */
export const OutputSubagentServerToolItem$inboundSchema = z.object({
    error: z.string().optional(),
    id: z.string().optional(),
    instance_name: z.string().optional(),
    model: z.string().optional(),
    name: z.string().optional(),
    outcome: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    task_description: z.string().optional(),
    task_name: z.string().optional(),
    type: OutputSubagentServerToolItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "instance_name": "instanceName",
        "task_description": "taskDescription",
        "task_name": "taskName",
    });
});
/** @internal */
export const OutputSubagentServerToolItem$outboundSchema = z.object({
    error: z.string().optional(),
    id: z.string().optional(),
    instanceName: z.string().optional(),
    model: z.string().optional(),
    name: z.string().optional(),
    outcome: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    taskDescription: z.string().optional(),
    taskName: z.string().optional(),
    type: OutputSubagentServerToolItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        instanceName: "instance_name",
        taskDescription: "task_description",
        taskName: "task_name",
    });
});
export function outputSubagentServerToolItemToJSON(outputSubagentServerToolItem) {
    return JSON.stringify(OutputSubagentServerToolItem$outboundSchema.parse(outputSubagentServerToolItem));
}
export function outputSubagentServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputSubagentServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputSubagentServerToolItem' from JSON`);
}
