// @openrouter/sdk/esm/models/outputmcpservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: dd859086c9c7
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputMcpServerToolItemType = {
    OpenrouterMcp: "openrouter:mcp",
};
/** @internal */
export const OutputMcpServerToolItemType$inboundSchema = z.enum(OutputMcpServerToolItemType);
/** @internal */
export const OutputMcpServerToolItemType$outboundSchema = OutputMcpServerToolItemType$inboundSchema;
/** @internal */
export const OutputMcpServerToolItem$inboundSchema = z.object({
    id: z.string().optional(),
    serverLabel: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    toolName: z.string().optional(),
    type: OutputMcpServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputMcpServerToolItem$outboundSchema = z.object({
    id: z.string().optional(),
    serverLabel: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    toolName: z.string().optional(),
    type: OutputMcpServerToolItemType$outboundSchema,
});
export function outputMcpServerToolItemToJSON(outputMcpServerToolItem) {
    return JSON.stringify(OutputMcpServerToolItem$outboundSchema.parse(outputMcpServerToolItem));
}
export function outputMcpServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMcpServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMcpServerToolItem' from JSON`);
}
