// @openrouter/sdk/esm/models/outputfilesservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 66f4fffdcaca
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputFilesServerToolItemType = {
    OpenrouterFiles: "openrouter:files",
};
/** @internal */
export const OutputFilesServerToolItemType$inboundSchema = z.enum(OutputFilesServerToolItemType);
/** @internal */
export const OutputFilesServerToolItemType$outboundSchema = OutputFilesServerToolItemType$inboundSchema;
/** @internal */
export const OutputFilesServerToolItem$inboundSchema = z.object({
    error: z.string().optional(),
    file_id: z.string().optional(),
    filename: z.string().optional(),
    id: z.string().optional(),
    operation: z.string().optional(),
    result: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputFilesServerToolItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "file_id": "fileId",
    });
});
/** @internal */
export const OutputFilesServerToolItem$outboundSchema = z.object({
    error: z.string().optional(),
    fileId: z.string().optional(),
    filename: z.string().optional(),
    id: z.string().optional(),
    operation: z.string().optional(),
    result: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputFilesServerToolItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        fileId: "file_id",
    });
});
export function outputFilesServerToolItemToJSON(outputFilesServerToolItem) {
    return JSON.stringify(OutputFilesServerToolItem$outboundSchema.parse(outputFilesServerToolItem));
}
export function outputFilesServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputFilesServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputFilesServerToolItem' from JSON`);
}
