// @openrouter/sdk/esm/models/outputcomputercallitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: fd4439fdf358
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const OutputComputerCallItemStatus = {
    Completed: "completed",
    Incomplete: "incomplete",
    InProgress: "in_progress",
};
export const OutputComputerCallItemType = {
    ComputerCall: "computer_call",
};
/** @internal */
export const PendingSafetyCheck$inboundSchema = z.object({
    code: z.string(),
    id: z.string(),
    message: z.string(),
});
/** @internal */
export const PendingSafetyCheck$outboundSchema = z.object({
    code: z.string(),
    id: z.string(),
    message: z.string(),
});
export function pendingSafetyCheckToJSON(pendingSafetyCheck) {
    return JSON.stringify(PendingSafetyCheck$outboundSchema.parse(pendingSafetyCheck));
}
export function pendingSafetyCheckFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PendingSafetyCheck$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PendingSafetyCheck' from JSON`);
}
/** @internal */
export const OutputComputerCallItemStatus$inboundSchema = openEnums.inboundSchema(OutputComputerCallItemStatus);
/** @internal */
export const OutputComputerCallItemStatus$outboundSchema = openEnums.outboundSchema(OutputComputerCallItemStatus);
/** @internal */
export const OutputComputerCallItemType$inboundSchema = z.enum(OutputComputerCallItemType);
/** @internal */
export const OutputComputerCallItemType$outboundSchema = OutputComputerCallItemType$inboundSchema;
/** @internal */
export const OutputComputerCallItem$inboundSchema = z.object({
    action: z.any().optional(),
    call_id: z.string(),
    id: z.string().optional(),
    pending_safety_checks: z.array(z.lazy(() => PendingSafetyCheck$inboundSchema)),
    status: OutputComputerCallItemStatus$inboundSchema,
    type: OutputComputerCallItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
        "pending_safety_checks": "pendingSafetyChecks",
    });
});
/** @internal */
export const OutputComputerCallItem$outboundSchema = z.object({
    action: z.any().optional(),
    callId: z.string(),
    id: z.string().optional(),
    pendingSafetyChecks: z.array(z.lazy(() => PendingSafetyCheck$outboundSchema)),
    status: OutputComputerCallItemStatus$outboundSchema,
    type: OutputComputerCallItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
        pendingSafetyChecks: "pending_safety_checks",
    });
});
export function outputComputerCallItemToJSON(outputComputerCallItem) {
    return JSON.stringify(OutputComputerCallItem$outboundSchema.parse(outputComputerCallItem));
}
export function outputComputerCallItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputComputerCallItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputComputerCallItem' from JSON`);
}
