// @openrouter/sdk/esm/models/outputsearchmodelsservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e7518c133cf6
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputSearchModelsServerToolItemType = {
    OpenrouterExperimentalSearchModels: "openrouter:experimental__search_models",
};
/** @internal */
export const OutputSearchModelsServerToolItemType$inboundSchema = z.enum(OutputSearchModelsServerToolItemType);
/** @internal */
export const OutputSearchModelsServerToolItemType$outboundSchema = OutputSearchModelsServerToolItemType$inboundSchema;
/** @internal */
export const OutputSearchModelsServerToolItem$inboundSchema = z.object({
    arguments: z.string().optional(),
    id: z.string().optional(),
    query: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputSearchModelsServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputSearchModelsServerToolItem$outboundSchema = z.object({
    arguments: z.string().optional(),
    id: z.string().optional(),
    query: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputSearchModelsServerToolItemType$outboundSchema,
});
export function outputSearchModelsServerToolItemToJSON(outputSearchModelsServerToolItem) {
    return JSON.stringify(OutputSearchModelsServerToolItem$outboundSchema.parse(outputSearchModelsServerToolItem));
}
export function outputSearchModelsServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputSearchModelsServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputSearchModelsServerToolItem' from JSON`);
}
