// @openrouter/sdk/esm/models/outputbashservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8eeee25f4818
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputBashServerToolItemType = {
    OpenrouterBash: "openrouter:bash",
};
/** @internal */
export const OutputBashServerToolItemType$inboundSchema = z.enum(OutputBashServerToolItemType);
/** @internal */
export const OutputBashServerToolItemType$outboundSchema = OutputBashServerToolItemType$inboundSchema;
/** @internal */
export const OutputBashServerToolItem$inboundSchema = z.object({
    arguments: z.string().optional(),
    call_id: z.string().optional(),
    command: z.string().optional(),
    exitCode: z.int().optional(),
    id: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    stderr: z.string().optional(),
    stdout: z.string().optional(),
    type: OutputBashServerToolItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
/** @internal */
export const OutputBashServerToolItem$outboundSchema = z.object({
    arguments: z.string().optional(),
    callId: z.string().optional(),
    command: z.string().optional(),
    exitCode: z.int().optional(),
    id: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    stderr: z.string().optional(),
    stdout: z.string().optional(),
    type: OutputBashServerToolItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function outputBashServerToolItemToJSON(outputBashServerToolItem) {
    return JSON.stringify(OutputBashServerToolItem$outboundSchema.parse(outputBashServerToolItem));
}
export function outputBashServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputBashServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputBashServerToolItem' from JSON`);
}
