// @openrouter/sdk/esm/models/outputbrowseruseservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5ad5586d0533
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputBrowserUseServerToolItemType = {
    OpenrouterBrowserUse: "openrouter:browser_use",
};
/** @internal */
export const OutputBrowserUseServerToolItemType$inboundSchema = z.enum(OutputBrowserUseServerToolItemType);
/** @internal */
export const OutputBrowserUseServerToolItemType$outboundSchema = OutputBrowserUseServerToolItemType$inboundSchema;
/** @internal */
export const OutputBrowserUseServerToolItem$inboundSchema = z.object({
    action: z.string().optional(),
    id: z.string().optional(),
    screenshotB64: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputBrowserUseServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputBrowserUseServerToolItem$outboundSchema = z.object({
    action: z.string().optional(),
    id: z.string().optional(),
    screenshotB64: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputBrowserUseServerToolItemType$outboundSchema,
});
export function outputBrowserUseServerToolItemToJSON(outputBrowserUseServerToolItem) {
    return JSON.stringify(OutputBrowserUseServerToolItem$outboundSchema.parse(outputBrowserUseServerToolItem));
}
export function outputBrowserUseServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputBrowserUseServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputBrowserUseServerToolItem' from JSON`);
}
