// @openrouter/sdk/esm/models/outputcustomtoolcallitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 71596c10c79b
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const OutputCustomToolCallItemType = {
    CustomToolCall: "custom_tool_call",
};
/** @internal */
export const OutputCustomToolCallItemType$inboundSchema = z.enum(OutputCustomToolCallItemType);
/** @internal */
export const OutputCustomToolCallItemType$outboundSchema = OutputCustomToolCallItemType$inboundSchema;
/** @internal */
export const OutputCustomToolCallItem$inboundSchema = z.object({
    call_id: z.string(),
    id: z.string().optional(),
    input: z.string(),
    name: z.string(),
    namespace: z.string().optional(),
    type: OutputCustomToolCallItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
/** @internal */
export const OutputCustomToolCallItem$outboundSchema = z.object({
    callId: z.string(),
    id: z.string().optional(),
    input: z.string(),
    name: z.string(),
    namespace: z.string().optional(),
    type: OutputCustomToolCallItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function outputCustomToolCallItemToJSON(outputCustomToolCallItem) {
    return JSON.stringify(OutputCustomToolCallItem$outboundSchema.parse(outputCustomToolCallItem));
}
export function outputCustomToolCallItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputCustomToolCallItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputCustomToolCallItem' from JSON`);
}
