// @openrouter/sdk/esm/models/outputfilesearchservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3e5afbd64572
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputFileSearchServerToolItemType = {
    OpenrouterFileSearch: "openrouter:file_search",
};
/** @internal */
export const OutputFileSearchServerToolItemType$inboundSchema = z.enum(OutputFileSearchServerToolItemType);
/** @internal */
export const OutputFileSearchServerToolItemType$outboundSchema = OutputFileSearchServerToolItemType$inboundSchema;
/** @internal */
export const OutputFileSearchServerToolItem$inboundSchema = z.object({
    id: z.string().optional(),
    queries: z.array(z.string()).optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputFileSearchServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputFileSearchServerToolItem$outboundSchema = z.object({
    id: z.string().optional(),
    queries: z.array(z.string()).optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputFileSearchServerToolItemType$outboundSchema,
});
export function outputFileSearchServerToolItemToJSON(outputFileSearchServerToolItem) {
    return JSON.stringify(OutputFileSearchServerToolItem$outboundSchema.parse(outputFileSearchServerToolItem));
}
export function outputFileSearchServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputFileSearchServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputFileSearchServerToolItem' from JSON`);
}
