// @openrouter/sdk/esm/models/outputtoolsearchservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 54b4dacceeab
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputToolSearchServerToolItemType = {
    OpenrouterToolSearch: "openrouter:tool_search",
};
/** @internal */
export const OutputToolSearchServerToolItemType$inboundSchema = z.enum(OutputToolSearchServerToolItemType);
/** @internal */
export const OutputToolSearchServerToolItemType$outboundSchema = OutputToolSearchServerToolItemType$inboundSchema;
/** @internal */
export const OutputToolSearchServerToolItem$inboundSchema = z.object({
    id: z.string().optional(),
    query: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputToolSearchServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputToolSearchServerToolItem$outboundSchema = z.object({
    id: z.string().optional(),
    query: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputToolSearchServerToolItemType$outboundSchema,
});
export function outputToolSearchServerToolItemToJSON(outputToolSearchServerToolItem) {
    return JSON.stringify(OutputToolSearchServerToolItem$outboundSchema.parse(outputToolSearchServerToolItem));
}
export function outputToolSearchServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputToolSearchServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputToolSearchServerToolItem' from JSON`);
}
