// @openrouter/sdk/esm/models/outputwebsearchservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 92114d9bad95
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputWebSearchServerToolItemTypeURL = {
    Url: "url",
};
export const OutputWebSearchServerToolItemTypeSearch = {
    Search: "search",
};
export const OutputWebSearchServerToolItemTypeOpenrouterWebSearch = {
    OpenrouterWebSearch: "openrouter:web_search",
};
/** @internal */
export const OutputWebSearchServerToolItemTypeURL$inboundSchema = z.enum(OutputWebSearchServerToolItemTypeURL);
/** @internal */
export const OutputWebSearchServerToolItemTypeURL$outboundSchema = OutputWebSearchServerToolItemTypeURL$inboundSchema;
/** @internal */
export const OutputWebSearchServerToolItemSource$inboundSchema = z.object({
    type: OutputWebSearchServerToolItemTypeURL$inboundSchema,
    url: z.string(),
});
/** @internal */
export const OutputWebSearchServerToolItemSource$outboundSchema = z.object({
    type: OutputWebSearchServerToolItemTypeURL$outboundSchema,
    url: z.string(),
});
export function outputWebSearchServerToolItemSourceToJSON(outputWebSearchServerToolItemSource) {
    return JSON.stringify(OutputWebSearchServerToolItemSource$outboundSchema.parse(outputWebSearchServerToolItemSource));
}
export function outputWebSearchServerToolItemSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputWebSearchServerToolItemSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputWebSearchServerToolItemSource' from JSON`);
}
/** @internal */
export const OutputWebSearchServerToolItemTypeSearch$inboundSchema = z.enum(OutputWebSearchServerToolItemTypeSearch);
/** @internal */
export const OutputWebSearchServerToolItemTypeSearch$outboundSchema = OutputWebSearchServerToolItemTypeSearch$inboundSchema;
/** @internal */
export const OutputWebSearchServerToolItemAction$inboundSchema = z.object({
    query: z.string(),
    sources: z.array(z.lazy(() => OutputWebSearchServerToolItemSource$inboundSchema)).optional(),
    type: OutputWebSearchServerToolItemTypeSearch$inboundSchema,
});
/** @internal */
export const OutputWebSearchServerToolItemAction$outboundSchema = z.object({
    query: z.string(),
    sources: z.array(z.lazy(() => OutputWebSearchServerToolItemSource$outboundSchema)).optional(),
    type: OutputWebSearchServerToolItemTypeSearch$outboundSchema,
});
export function outputWebSearchServerToolItemActionToJSON(outputWebSearchServerToolItemAction) {
    return JSON.stringify(OutputWebSearchServerToolItemAction$outboundSchema.parse(outputWebSearchServerToolItemAction));
}
export function outputWebSearchServerToolItemActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputWebSearchServerToolItemAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputWebSearchServerToolItemAction' from JSON`);
}
/** @internal */
export const OutputWebSearchServerToolItemTypeOpenrouterWebSearch$inboundSchema = z
    .enum(OutputWebSearchServerToolItemTypeOpenrouterWebSearch);
/** @internal */
export const OutputWebSearchServerToolItemTypeOpenrouterWebSearch$outboundSchema = OutputWebSearchServerToolItemTypeOpenrouterWebSearch$inboundSchema;
/** @internal */
export const OutputWebSearchServerToolItem$inboundSchema = z.object({
    action: z.lazy(() => OutputWebSearchServerToolItemAction$inboundSchema)
        .optional(),
    id: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputWebSearchServerToolItemTypeOpenrouterWebSearch$inboundSchema,
});
/** @internal */
export const OutputWebSearchServerToolItem$outboundSchema = z.object({
    action: z.lazy(() => OutputWebSearchServerToolItemAction$outboundSchema)
        .optional(),
    id: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputWebSearchServerToolItemTypeOpenrouterWebSearch$outboundSchema,
});
export function outputWebSearchServerToolItemToJSON(outputWebSearchServerToolItem) {
    return JSON.stringify(OutputWebSearchServerToolItem$outboundSchema.parse(outputWebSearchServerToolItem));
}
export function outputWebSearchServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputWebSearchServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputWebSearchServerToolItem' from JSON`);
}
