// @openrouter/sdk/esm/models/outputtexteditorservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6c55837a703d
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const Command = {
    View: "view",
    Create: "create",
    StrReplace: "str_replace",
    Insert: "insert",
};
export const OutputTextEditorServerToolItemType = {
    OpenrouterTextEditor: "openrouter:text_editor",
};
/** @internal */
export const Command$inboundSchema = openEnums
    .inboundSchema(Command);
/** @internal */
export const Command$outboundSchema = openEnums
    .outboundSchema(Command);
/** @internal */
export const OutputTextEditorServerToolItemType$inboundSchema = z.enum(OutputTextEditorServerToolItemType);
/** @internal */
export const OutputTextEditorServerToolItemType$outboundSchema = OutputTextEditorServerToolItemType$inboundSchema;
/** @internal */
export const OutputTextEditorServerToolItem$inboundSchema = z.object({
    command: Command$inboundSchema.optional(),
    filePath: z.string().optional(),
    id: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputTextEditorServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputTextEditorServerToolItem$outboundSchema = z.object({
    command: Command$outboundSchema.optional(),
    filePath: z.string().optional(),
    id: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputTextEditorServerToolItemType$outboundSchema,
});
export function outputTextEditorServerToolItemToJSON(outputTextEditorServerToolItem) {
    return JSON.stringify(OutputTextEditorServerToolItem$outboundSchema.parse(outputTextEditorServerToolItem));
}
export function outputTextEditorServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputTextEditorServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputTextEditorServerToolItem' from JSON`);
}
