// @openrouter/sdk/esm/models/outputcodeinterpreterservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 16f87f6046c4
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputCodeInterpreterServerToolItemType = {
    OpenrouterCodeInterpreter: "openrouter:code_interpreter",
};
/** @internal */
export const OutputCodeInterpreterServerToolItemType$inboundSchema = z.enum(OutputCodeInterpreterServerToolItemType);
/** @internal */
export const OutputCodeInterpreterServerToolItemType$outboundSchema = OutputCodeInterpreterServerToolItemType$inboundSchema;
/** @internal */
export const OutputCodeInterpreterServerToolItem$inboundSchema = z.object({
    code: z.string().optional(),
    exitCode: z.int().optional(),
    id: z.string().optional(),
    language: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    stderr: z.string().optional(),
    stdout: z.string().optional(),
    type: OutputCodeInterpreterServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputCodeInterpreterServerToolItem$outboundSchema = z.object({
    code: z.string().optional(),
    exitCode: z.int().optional(),
    id: z.string().optional(),
    language: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    stderr: z.string().optional(),
    stdout: z.string().optional(),
    type: OutputCodeInterpreterServerToolItemType$outboundSchema,
});
export function outputCodeInterpreterServerToolItemToJSON(outputCodeInterpreterServerToolItem) {
    return JSON.stringify(OutputCodeInterpreterServerToolItem$outboundSchema.parse(outputCodeInterpreterServerToolItem));
}
export function outputCodeInterpreterServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputCodeInterpreterServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputCodeInterpreterServerToolItem' from JSON`);
}
