// @openrouter/sdk/esm/models/outputimagegenerationservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1ea34f16eb56
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputImageGenerationServerToolItemType = {
    OpenrouterImageGeneration: "openrouter:image_generation",
};
/** @internal */
export const OutputImageGenerationServerToolItemType$inboundSchema = z.enum(OutputImageGenerationServerToolItemType);
/** @internal */
export const OutputImageGenerationServerToolItemType$outboundSchema = OutputImageGenerationServerToolItemType$inboundSchema;
/** @internal */
export const OutputImageGenerationServerToolItem$inboundSchema = z.object({
    id: z.string().optional(),
    imageB64: z.string().optional(),
    imageUrl: z.string().optional(),
    prompt: z.string().optional(),
    result: z.nullable(z.string()).optional(),
    revisedPrompt: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputImageGenerationServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputImageGenerationServerToolItem$outboundSchema = z.object({
    id: z.string().optional(),
    imageB64: z.string().optional(),
    imageUrl: z.string().optional(),
    prompt: z.string().optional(),
    result: z.nullable(z.string()).optional(),
    revisedPrompt: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputImageGenerationServerToolItemType$outboundSchema,
});
export function outputImageGenerationServerToolItemToJSON(outputImageGenerationServerToolItem) {
    return JSON.stringify(OutputImageGenerationServerToolItem$outboundSchema.parse(outputImageGenerationServerToolItem));
}
export function outputImageGenerationServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputImageGenerationServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputImageGenerationServerToolItem' from JSON`);
}
