// @openrouter/sdk/esm/models/outputcodeinterpretercallitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ee080e2eccae
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const TypeCodeInterpreterCall = {
    CodeInterpreterCall: "code_interpreter_call",
};
/** @internal */
export const OutputLogs$inboundSchema = z
    .object({
    logs: z.string(),
    type: z.literal("logs"),
});
/** @internal */
export const OutputLogs$outboundSchema = z.object({
    logs: z.string(),
    type: z.literal("logs"),
});
export function outputLogsToJSON(outputLogs) {
    return JSON.stringify(OutputLogs$outboundSchema.parse(outputLogs));
}
export function outputLogsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputLogs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputLogs' from JSON`);
}
/** @internal */
export const OutputImage$inboundSchema = z
    .object({
    type: z.literal("image"),
    url: z.string(),
});
/** @internal */
export const OutputImage$outboundSchema = z.object({
    type: z.literal("image"),
    url: z.string(),
});
export function outputImageToJSON(outputImage) {
    return JSON.stringify(OutputImage$outboundSchema.parse(outputImage));
}
export function outputImageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputImage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputImage' from JSON`);
}
/** @internal */
export const OutputCodeInterpreterCallItemOutputUnion$inboundSchema = discriminatedUnion("type", {
    image: z.lazy(() => OutputImage$inboundSchema),
    logs: z.lazy(() => OutputLogs$inboundSchema),
});
/** @internal */
export const OutputCodeInterpreterCallItemOutputUnion$outboundSchema = z.union([
    z.lazy(() => OutputImage$outboundSchema),
    z.lazy(() => OutputLogs$outboundSchema),
]);
export function outputCodeInterpreterCallItemOutputUnionToJSON(outputCodeInterpreterCallItemOutputUnion) {
    return JSON.stringify(OutputCodeInterpreterCallItemOutputUnion$outboundSchema.parse(outputCodeInterpreterCallItemOutputUnion));
}
export function outputCodeInterpreterCallItemOutputUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputCodeInterpreterCallItemOutputUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputCodeInterpreterCallItemOutputUnion' from JSON`);
}
/** @internal */
export const TypeCodeInterpreterCall$inboundSchema = z.enum(TypeCodeInterpreterCall);
/** @internal */
export const TypeCodeInterpreterCall$outboundSchema = TypeCodeInterpreterCall$inboundSchema;
/** @internal */
export const OutputCodeInterpreterCallItem$inboundSchema = z.object({
    code: z.nullable(z.string()),
    container_id: z.string(),
    id: z.string(),
    outputs: z.nullable(z.array(discriminatedUnion("type", {
        image: z.lazy(() => OutputImage$inboundSchema),
        logs: z.lazy(() => OutputLogs$inboundSchema),
    }))),
    status: ToolCallStatus$inboundSchema,
    type: TypeCodeInterpreterCall$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "container_id": "containerId",
    });
});
/** @internal */
export const OutputCodeInterpreterCallItem$outboundSchema = z.object({
    code: z.nullable(z.string()),
    containerId: z.string(),
    id: z.string(),
    outputs: z.nullable(z.array(z.union([
        z.lazy(() => OutputImage$outboundSchema),
        z.lazy(() => OutputLogs$outboundSchema),
    ]))),
    status: ToolCallStatus$outboundSchema,
    type: TypeCodeInterpreterCall$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        containerId: "container_id",
    });
});
export function outputCodeInterpreterCallItemToJSON(outputCodeInterpreterCallItem) {
    return JSON.stringify(OutputCodeInterpreterCallItem$outboundSchema.parse(outputCodeInterpreterCallItem));
}
export function outputCodeInterpreterCallItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputCodeInterpreterCallItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputCodeInterpreterCallItem' from JSON`);
}
