// @openrouter/sdk/esm/models/outputshellcallitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d8b35b2daab6
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ShellCallStatus$inboundSchema, } from "./shellcallstatus.js";
/** @internal */
export const OutputShellCallItemAction$inboundSchema = z.object({
    commands: z.array(z.string()),
    max_output_length: z.nullable(z.int()),
    timeout_ms: z.nullable(z.int()),
}).transform((v) => {
    return remap$(v, {
        "max_output_length": "maxOutputLength",
        "timeout_ms": "timeoutMs",
    });
});
export function outputShellCallItemActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputShellCallItemAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputShellCallItemAction' from JSON`);
}
/** @internal */
export const OutputShellCallItem$inboundSchema = z.object({
    action: z.lazy(() => OutputShellCallItemAction$inboundSchema).optional(),
    call_id: z.string(),
    id: z.string(),
    status: ShellCallStatus$inboundSchema,
    type: z.literal("shell_call"),
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
export function outputShellCallItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputShellCallItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputShellCallItem' from JSON`);
}
