// @openrouter/sdk/esm/models/outputadvisorservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0c9d6ff77975
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputAdvisorServerToolItemType = {
    OpenrouterAdvisor: "openrouter:advisor",
};
/** @internal */
export const OutputAdvisorServerToolItemType$inboundSchema = z.enum(OutputAdvisorServerToolItemType);
/** @internal */
export const OutputAdvisorServerToolItemType$outboundSchema = OutputAdvisorServerToolItemType$inboundSchema;
/** @internal */
export const OutputAdvisorServerToolItem$inboundSchema = z.object({
    advice: z.string().optional(),
    error: z.string().optional(),
    id: z.string().optional(),
    instance_name: z.string().optional(),
    model: z.string().optional(),
    prompt: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputAdvisorServerToolItemType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "instance_name": "instanceName",
    });
});
/** @internal */
export const OutputAdvisorServerToolItem$outboundSchema = z.object({
    advice: z.string().optional(),
    error: z.string().optional(),
    id: z.string().optional(),
    instanceName: z.string().optional(),
    model: z.string().optional(),
    prompt: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputAdvisorServerToolItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        instanceName: "instance_name",
    });
});
export function outputAdvisorServerToolItemToJSON(outputAdvisorServerToolItem) {
    return JSON.stringify(OutputAdvisorServerToolItem$outboundSchema.parse(outputAdvisorServerToolItem));
}
export function outputAdvisorServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputAdvisorServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputAdvisorServerToolItem' from JSON`);
}
