// @openrouter/sdk/esm/models/paretorouterplugin.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4e557119b1dd
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
/**
 * Price source for the Pareto frontier cost axis and for enforcing max_price. "prompt" uses catalog list price (endpoint.pricing.prompt). "weighted_avg" uses traffic-weighted effective input price from ClickHouse, falling back to prompt price for models without traffic data. Defaults to "prompt".
 */
export const PriceSource = {
    Prompt: "prompt",
    WeightedAvg: "weighted_avg",
};
/** @internal */
export const PriceSource$outboundSchema = openEnums.outboundSchema(PriceSource);
/** @internal */
export const ParetoRouterPlugin$outboundSchema = z.object({
    enabled: z.boolean().optional(),
    id: z.literal("pareto-router"),
    maxPrice: z.number().optional(),
    minCodingScore: z.number().optional(),
    priceSource: PriceSource$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        maxPrice: "max_price",
        minCodingScore: "min_coding_score",
        priceSource: "price_source",
    });
});
export function paretoRouterPluginToJSON(paretoRouterPlugin) {
    return JSON.stringify(ParetoRouterPlugin$outboundSchema.parse(paretoRouterPlugin));
}
