// @openrouter/sdk/esm/models/pipelinestage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d737d34295bb
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { PipelineStageType$inboundSchema, } from "./pipelinestagetype.js";
/** @internal */
export const PipelineStage$inboundSchema = z
    .object({
    cost_usd: z.nullable(z.number()).optional(),
    data: z.record(z.string(), z.any()).optional(),
    guardrail_id: z.string().optional(),
    guardrail_scope: z.string().optional(),
    name: z.string(),
    summary: z.string().optional(),
    type: PipelineStageType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "cost_usd": "costUsd",
        "guardrail_id": "guardrailId",
        "guardrail_scope": "guardrailScope",
    });
});
export function pipelineStageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PipelineStage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PipelineStage' from JSON`);
}
