// @openrouter/sdk/esm/models/presetdesignatedversion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 259a18622bbe
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const PresetDesignatedVersion$inboundSchema = z.object({
    config: z.record(z.string(), z.any()),
    created_at: z.string(),
    creator_id: z.string(),
    id: z.string(),
    preset_id: z.string(),
    system_prompt: z.nullable(z.string()),
    updated_at: z.string(),
    version: z.int(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "creator_id": "creatorId",
        "preset_id": "presetId",
        "system_prompt": "systemPrompt",
        "updated_at": "updatedAt",
    });
});
export function presetDesignatedVersionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PresetDesignatedVersion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PresetDesignatedVersion' from JSON`);
}
