// @openrouter/sdk/esm/models/preset.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f9cbbbd64494
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { PresetStatus$inboundSchema } from "./presetstatus.js";
/** @internal */
export const Preset$inboundSchema = z.object({
    created_at: z.string(),
    creator_user_id: z.nullable(z.string()),
    description: z.nullable(z.string()),
    designated_version_id: z.nullable(z.string()),
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    status: PresetStatus$inboundSchema,
    status_updated_at: z.nullable(z.string()),
    updated_at: z.string(),
    workspace_id: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "creator_user_id": "creatorUserId",
        "designated_version_id": "designatedVersionId",
        "status_updated_at": "statusUpdatedAt",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function presetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Preset$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Preset' from JSON`);
}
