// @openrouter/sdk/esm/models/preview20250311websearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c171c186ccac
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { PreviewWebSearchUserLocation$inboundSchema, PreviewWebSearchUserLocation$outboundSchema, } from "./previewwebsearchuserlocation.js";
import { SearchContextSizeEnum$inboundSchema, SearchContextSizeEnum$outboundSchema, } from "./searchcontextsizeenum.js";
import { WebSearchDomainFilter$inboundSchema, WebSearchDomainFilter$outboundSchema, } from "./websearchdomainfilter.js";
import { WebSearchEngineEnum$inboundSchema, WebSearchEngineEnum$outboundSchema, } from "./websearchengineenum.js";
export const Preview20250311WebSearchServerToolType = {
    WebSearchPreview20250311: "web_search_preview_2025_03_11",
};
/** @internal */
export const Preview20250311WebSearchServerToolType$inboundSchema = z.enum(Preview20250311WebSearchServerToolType);
/** @internal */
export const Preview20250311WebSearchServerToolType$outboundSchema = Preview20250311WebSearchServerToolType$inboundSchema;
/** @internal */
export const Preview20250311WebSearchServerTool$inboundSchema = z.object({
    engine: WebSearchEngineEnum$inboundSchema.optional(),
    filters: z.nullable(WebSearchDomainFilter$inboundSchema).optional(),
    max_results: z.int().optional(),
    max_uses: z.int().optional(),
    search_context_size: SearchContextSizeEnum$inboundSchema.optional(),
    type: Preview20250311WebSearchServerToolType$inboundSchema,
    user_location: z.nullable(PreviewWebSearchUserLocation$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "max_results": "maxResults",
        "max_uses": "maxUses",
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
    });
});
/** @internal */
export const Preview20250311WebSearchServerTool$outboundSchema = z.object({
    engine: WebSearchEngineEnum$outboundSchema.optional(),
    filters: z.nullable(WebSearchDomainFilter$outboundSchema).optional(),
    maxResults: z.int().optional(),
    maxUses: z.int().optional(),
    searchContextSize: SearchContextSizeEnum$outboundSchema.optional(),
    type: Preview20250311WebSearchServerToolType$outboundSchema,
    userLocation: z.nullable(PreviewWebSearchUserLocation$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxUses: "max_uses",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
    });
});
export function preview20250311WebSearchServerToolToJSON(preview20250311WebSearchServerTool) {
    return JSON.stringify(Preview20250311WebSearchServerTool$outboundSchema.parse(preview20250311WebSearchServerTool));
}
export function preview20250311WebSearchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Preview20250311WebSearchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Preview20250311WebSearchServerTool' from JSON`);
}
