// @openrouter/sdk/esm/models/previewwebsearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6b7147491ff2
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { PreviewWebSearchUserLocation$inboundSchema, PreviewWebSearchUserLocation$outboundSchema, } from "./previewwebsearchuserlocation.js";
import { SearchContextSizeEnum$inboundSchema, SearchContextSizeEnum$outboundSchema, } from "./searchcontextsizeenum.js";
import { WebSearchDomainFilter$inboundSchema, WebSearchDomainFilter$outboundSchema, } from "./websearchdomainfilter.js";
import { WebSearchEngineEnum$inboundSchema, WebSearchEngineEnum$outboundSchema, } from "./websearchengineenum.js";
export const PreviewWebSearchServerToolType = {
    WebSearchPreview: "web_search_preview",
};
/** @internal */
export const PreviewWebSearchServerToolType$inboundSchema = z.enum(PreviewWebSearchServerToolType);
/** @internal */
export const PreviewWebSearchServerToolType$outboundSchema = PreviewWebSearchServerToolType$inboundSchema;
/** @internal */
export const PreviewWebSearchServerTool$inboundSchema = z.object({
    engine: WebSearchEngineEnum$inboundSchema.optional(),
    filters: z.nullable(WebSearchDomainFilter$inboundSchema).optional(),
    max_results: z.int().optional(),
    max_uses: z.int().optional(),
    search_context_size: SearchContextSizeEnum$inboundSchema.optional(),
    type: PreviewWebSearchServerToolType$inboundSchema,
    user_location: z.nullable(PreviewWebSearchUserLocation$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "max_results": "maxResults",
        "max_uses": "maxUses",
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
    });
});
/** @internal */
export const PreviewWebSearchServerTool$outboundSchema = z.object({
    engine: WebSearchEngineEnum$outboundSchema.optional(),
    filters: z.nullable(WebSearchDomainFilter$outboundSchema).optional(),
    maxResults: z.int().optional(),
    maxUses: z.int().optional(),
    searchContextSize: SearchContextSizeEnum$outboundSchema.optional(),
    type: PreviewWebSearchServerToolType$outboundSchema,
    userLocation: z.nullable(PreviewWebSearchUserLocation$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxUses: "max_uses",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
    });
});
export function previewWebSearchServerToolToJSON(previewWebSearchServerTool) {
    return JSON.stringify(PreviewWebSearchServerTool$outboundSchema.parse(previewWebSearchServerTool));
}
export function previewWebSearchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PreviewWebSearchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PreviewWebSearchServerTool' from JSON`);
}
