// @openrouter/sdk/esm/models/pricingoverride.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b1116f9327ef
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const PricingOverride$inboundSchema = z.object({
    audio: z.string().optional(),
    completion: z.string().optional(),
    input_audio_cache: z.string().optional(),
    input_cache_read: z.string().optional(),
    input_cache_write: z.string().optional(),
    input_cache_write_1h: z.string().optional(),
    min_prompt_tokens: z.number().optional(),
    prompt: z.string().optional(),
    utc_end: z.number().optional(),
    utc_start: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        "input_audio_cache": "inputAudioCache",
        "input_cache_read": "inputCacheRead",
        "input_cache_write": "inputCacheWrite",
        "input_cache_write_1h": "inputCacheWrite1h",
        "min_prompt_tokens": "minPromptTokens",
        "utc_end": "utcEnd",
        "utc_start": "utcStart",
    });
});
export function pricingOverrideFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PricingOverride$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PricingOverride' from JSON`);
}
