// @openrouter/sdk/esm/models/promptcacheoptions.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: beb41f51263a
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const PromptCacheOptionsMode = {
    Explicit: "explicit",
};
/** @internal */
export const PromptCacheOptionsMode$inboundSchema = z.enum(PromptCacheOptionsMode);
/** @internal */
export const PromptCacheOptionsMode$outboundSchema = PromptCacheOptionsMode$inboundSchema;
/** @internal */
export const PromptCacheOptions$inboundSchema = z.object({
    mode: PromptCacheOptionsMode$inboundSchema,
    ttl: z.nullable(z.string()).optional(),
});
/** @internal */
export const PromptCacheOptions$outboundSchema = z.object({
    mode: PromptCacheOptionsMode$outboundSchema,
    ttl: z.nullable(z.string()).optional(),
});
export function promptCacheOptionsToJSON(promptCacheOptions) {
    return JSON.stringify(PromptCacheOptions$outboundSchema.parse(promptCacheOptions));
}
export function promptCacheOptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PromptCacheOptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PromptCacheOptions' from JSON`);
}
