// @openrouter/sdk/esm/models/promptcachebreakpoint.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f8222f76987b
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const PromptCacheBreakpointMode = {
    Explicit: "explicit",
};
/** @internal */
export const PromptCacheBreakpointMode$inboundSchema = z.enum(PromptCacheBreakpointMode);
/** @internal */
export const PromptCacheBreakpointMode$outboundSchema = PromptCacheBreakpointMode$inboundSchema;
/** @internal */
export const PromptCacheBreakpoint$inboundSchema = z.object({
    mode: PromptCacheBreakpointMode$inboundSchema,
});
/** @internal */
export const PromptCacheBreakpoint$outboundSchema = z.object({
    mode: PromptCacheBreakpointMode$outboundSchema,
});
export function promptCacheBreakpointToJSON(promptCacheBreakpoint) {
    return JSON.stringify(PromptCacheBreakpoint$outboundSchema.parse(promptCacheBreakpoint));
}
export function promptCacheBreakpointFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PromptCacheBreakpoint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PromptCacheBreakpoint' from JSON`);
}
