// @openrouter/sdk/esm/models/providerpreferences.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3a47295d7b91
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { PreferredMaxLatency$outboundSchema, } from "./preferredmaxlatency.js";
import { PreferredMinThroughput$outboundSchema, } from "./preferredminthroughput.js";
import { ProviderName$outboundSchema } from "./providername.js";
import { ProviderSort$outboundSchema } from "./providersort.js";
import { ProviderSortConfig$outboundSchema, } from "./providersortconfig.js";
import { Quantization$outboundSchema } from "./quantization.js";
/**
 * Data collection setting. If no available model provider meets the requirement, your request will return an error.
 *
 * @remarks
 * - allow: (default) allow providers which store user data non-transiently and may train on it
 *
 * - deny: use only providers which do not collect user data.
 */
export const DataCollection = {
    Deny: "deny",
    Allow: "allow",
};
/** @internal */
export const DataCollection$outboundSchema = openEnums.outboundSchema(DataCollection);
/** @internal */
export const ProviderPreferencesIgnore$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function providerPreferencesIgnoreToJSON(providerPreferencesIgnore) {
    return JSON.stringify(ProviderPreferencesIgnore$outboundSchema.parse(providerPreferencesIgnore));
}
/** @internal */
export const MaxPrice$outboundSchema = z
    .object({
    audio: z.string().optional(),
    completion: z.string().optional(),
    image: z.string().optional(),
    prompt: z.string().optional(),
    request: z.string().optional(),
});
export function maxPriceToJSON(maxPrice) {
    return JSON.stringify(MaxPrice$outboundSchema.parse(maxPrice));
}
/** @internal */
export const ProviderPreferencesOnly$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function providerPreferencesOnlyToJSON(providerPreferencesOnly) {
    return JSON.stringify(ProviderPreferencesOnly$outboundSchema.parse(providerPreferencesOnly));
}
/** @internal */
export const ProviderPreferencesOrder$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function providerPreferencesOrderToJSON(providerPreferencesOrder) {
    return JSON.stringify(ProviderPreferencesOrder$outboundSchema.parse(providerPreferencesOrder));
}
/** @internal */
export const ProviderPreferencesSort$outboundSchema = z.union([ProviderSort$outboundSchema, ProviderSortConfig$outboundSchema]);
export function providerPreferencesSortToJSON(providerPreferencesSort) {
    return JSON.stringify(ProviderPreferencesSort$outboundSchema.parse(providerPreferencesSort));
}
/** @internal */
export const ProviderPreferences$outboundSchema = z.object({
    allowFallbacks: z.nullable(z.boolean()).optional(),
    dataCollection: z.nullable(DataCollection$outboundSchema).optional(),
    enforceDistillableText: z.nullable(z.boolean()).optional(),
    ignore: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()]))).optional(),
    maxPrice: z.lazy(() => MaxPrice$outboundSchema).optional(),
    only: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()])))
        .optional(),
    order: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()])))
        .optional(),
    preferredMaxLatency: z.nullable(PreferredMaxLatency$outboundSchema)
        .optional(),
    preferredMinThroughput: z.nullable(PreferredMinThroughput$outboundSchema)
        .optional(),
    quantizations: z.nullable(z.array(Quantization$outboundSchema)).optional(),
    requireParameters: z.nullable(z.boolean()).optional(),
    sort: z.nullable(z.union([ProviderSort$outboundSchema, ProviderSortConfig$outboundSchema])).optional(),
    zdr: z.nullable(z.boolean()).optional(),
}).transform((v) => {
    return remap$(v, {
        allowFallbacks: "allow_fallbacks",
        dataCollection: "data_collection",
        enforceDistillableText: "enforce_distillable_text",
        maxPrice: "max_price",
        preferredMaxLatency: "preferred_max_latency",
        preferredMinThroughput: "preferred_min_throughput",
        requireParameters: "require_parameters",
    });
});
export function providerPreferencesToJSON(providerPreferences) {
    return JSON.stringify(ProviderPreferences$outboundSchema.parse(providerPreferences));
}
