// @openrouter/sdk/esm/models/providername.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 89e536fb023a
 */
import * as openEnums from "../types/enums.js";
export const ProviderName = {
    AkashML: "AkashML",
    Ai21: "AI21",
    AionLabs: "AionLabs",
    Alibaba: "Alibaba",
    Ambient: "Ambient",
    Baidu: "Baidu",
    AmazonBedrock: "Amazon Bedrock",
    AmazonNova: "Amazon Nova",
    Anthropic: "Anthropic",
    ArceeAI: "Arcee AI",
    AtlasCloud: "AtlasCloud",
    Avian: "Avian",
    Azure: "Azure",
    BaseTen: "BaseTen",
    BytePlus: "BytePlus",
    BlackForestLabs: "Black Forest Labs",
    Cerebras: "Cerebras",
    Chutes: "Chutes",
    Cirrascale: "Cirrascale",
    Clarifai: "Clarifai",
    Cloudflare: "Cloudflare",
    Cohere: "Cohere",
    CoreWeave: "CoreWeave",
    Crucible: "Crucible",
    Crusoe: "Crusoe",
    Darkbloom: "Darkbloom",
    Decart: "Decart",
    Deepgram: "Deepgram",
    DeepInfra: "DeepInfra",
    DeepSeek: "DeepSeek",
    DekaLLM: "DekaLLM",
    DigitalOcean: "DigitalOcean",
    Featherless: "Featherless",
    Fireworks: "Fireworks",
    FishAudio: "Fish Audio",
    Friendli: "Friendli",
    GMICloud: "GMICloud",
    Google: "Google",
    GoogleAIStudio: "Google AI Studio",
    Groq: "Groq",
    HeyGen: "HeyGen",
    Inception: "Inception",
    Inceptron: "Inceptron",
    InferenceNet: "InferenceNet",
    Ionstream: "Ionstream",
    Infermatic: "Infermatic",
    IoNet: "Io Net",
    InferactVLLM: "Inferact vLLM",
    Inflection: "Inflection",
    Liquid: "Liquid",
    Mara: "Mara",
    Mancer2: "Mancer 2",
    Meta: "Meta",
    Minimax: "Minimax",
    ModelRun: "ModelRun",
    Mistral: "Mistral",
    Modular: "Modular",
    MoonshotAI: "Moonshot AI",
    Morph: "Morph",
    NCompass: "NCompass",
    Nebius: "Nebius",
    NexAGI: "Nex AGI",
    NextBit: "NextBit",
    Novita: "Novita",
    Nvidia: "Nvidia",
    OpenAI: "OpenAI",
    OpenInference: "OpenInference",
    Parasail: "Parasail",
    Poolside: "Poolside",
    Perceptron: "Perceptron",
    Perplexity: "Perplexity",
    Phala: "Phala",
    Recraft: "Recraft",
    Reka: "Reka",
    Relace: "Relace",
    SailResearch: "Sail Research",
    SakanaAI: "Sakana AI",
    SambaNova: "SambaNova",
    Seed: "Seed",
    SiliconFlow: "SiliconFlow",
    Sourceful: "Sourceful",
    StepFun: "StepFun",
    Stealth: "Stealth",
    StreamLake: "StreamLake",
    Switchpoint: "Switchpoint",
    Tencent: "Tencent",
    Tenstorrent: "Tenstorrent",
    Together: "Together",
    Upstage: "Upstage",
    Venice: "Venice",
    Wafer: "Wafer",
    WandB: "WandB",
    Quiver: "Quiver",
    Krea: "Krea",
    Runway: "Runway",
    Xiaomi: "Xiaomi",
    XAI: "xAI",
    ZAi: "Z.AI",
    FakeProvider: "FakeProvider",
};
/** @internal */
export const ProviderName$inboundSchema = openEnums.inboundSchema(ProviderName);
/** @internal */
export const ProviderName$outboundSchema = openEnums.outboundSchema(ProviderName);
