// @openrouter/sdk/esm/models/rankingsdailymeta.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6e46d93381d7
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * Dataset version. Field names and grain are stable for the life of `v1`.
 */
export const RankingsDailyMetaVersion = {
    V1: "v1",
};
/** @internal */
export const RankingsDailyMetaVersion$inboundSchema = z.enum(RankingsDailyMetaVersion);
/** @internal */
export const RankingsDailyMeta$inboundSchema = z.object({
    as_of: z.string(),
    end_date: z.string(),
    start_date: z.string(),
    version: RankingsDailyMetaVersion$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "as_of": "asOf",
        "end_date": "endDate",
        "start_date": "startDate",
    });
});
export function rankingsDailyMetaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RankingsDailyMeta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RankingsDailyMeta' from JSON`);
}
