// @openrouter/sdk/esm/models/reasoningdetailservertoolcall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e2bb1917981b
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ReasoningFormat$inboundSchema, ReasoningFormat$outboundSchema, } from "./reasoningformat.js";
/** @internal */
export const ReasoningDetailServerToolCall$inboundSchema = z.object({
    arguments: z.string(),
    format: z.nullable(ReasoningFormat$inboundSchema).optional(),
    id: z.nullable(z.string()).optional(),
    index: z.int().optional(),
    result: z.string(),
    tool_call_id: z.nullable(z.string()).optional(),
    tool_name: z.string(),
    type: z.literal("reasoning.server_tool_call"),
}).transform((v) => {
    return remap$(v, {
        "tool_call_id": "toolCallId",
        "tool_name": "toolName",
    });
});
/** @internal */
export const ReasoningDetailServerToolCall$outboundSchema = z.object({
    arguments: z.string(),
    format: z.nullable(ReasoningFormat$outboundSchema).optional(),
    id: z.nullable(z.string()).optional(),
    index: z.int().optional(),
    result: z.string(),
    toolCallId: z.nullable(z.string()).optional(),
    toolName: z.string(),
    type: z.literal("reasoning.server_tool_call"),
}).transform((v) => {
    return remap$(v, {
        toolCallId: "tool_call_id",
        toolName: "tool_name",
    });
});
export function reasoningDetailServerToolCallToJSON(reasoningDetailServerToolCall) {
    return JSON.stringify(ReasoningDetailServerToolCall$outboundSchema.parse(reasoningDetailServerToolCall));
}
export function reasoningDetailServerToolCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ReasoningDetailServerToolCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ReasoningDetailServerToolCall' from JSON`);
}
