// @openrouter/sdk/esm/models/responsesrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 39799702f982
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { AdvisorServerToolOpenRouter$outboundSchema, } from "./advisorservertoolopenrouter.js";
import { AnthropicCacheControlDirective$outboundSchema, } from "./anthropiccachecontroldirective.js";
import { ApplyPatchServerTool$outboundSchema, } from "./applypatchservertool.js";
import { ApplyPatchServerToolOpenRouter$outboundSchema, } from "./applypatchservertoolopenrouter.js";
import { AutoBetaRouterPlugin$outboundSchema, } from "./autobetarouterplugin.js";
import { AutoRouterPlugin$outboundSchema, } from "./autorouterplugin.js";
import { BashServerTool$outboundSchema, } from "./bashservertool.js";
import { ChatDebugOptions$outboundSchema, } from "./chatdebugoptions.js";
import { CodeInterpreterServerTool$outboundSchema, } from "./codeinterpreterservertool.js";
import { CodexLocalShellTool$outboundSchema, } from "./codexlocalshelltool.js";
import { ComputerUseServerTool$outboundSchema, } from "./computeruseservertool.js";
import { ContextCompressionPlugin$outboundSchema, } from "./contextcompressionplugin.js";
import { CustomTool$outboundSchema, } from "./customtool.js";
import { DatetimeServerTool$outboundSchema, } from "./datetimeservertool.js";
import { FileParserPlugin$outboundSchema, } from "./fileparserplugin.js";
import { FileSearchServerTool$outboundSchema, } from "./filesearchservertool.js";
import { FilesServerTool$outboundSchema, } from "./filesservertool.js";
import { FusionPlugin$outboundSchema, } from "./fusionplugin.js";
import { FusionServerToolOpenRouter$outboundSchema, } from "./fusionservertoolopenrouter.js";
import { ImageConfig$outboundSchema, } from "./imageconfig.js";
import { ImageGenerationServerTool$outboundSchema, } from "./imagegenerationservertool.js";
import { ImageGenerationServerToolOpenRouter$outboundSchema, } from "./imagegenerationservertoolopenrouter.js";
import { InputsUnion$outboundSchema, } from "./inputsunion.js";
import { LegacyWebSearchServerTool$outboundSchema, } from "./legacywebsearchservertool.js";
import { McpServerTool$outboundSchema, } from "./mcpservertool.js";
import { ModerationPlugin$outboundSchema, } from "./moderationplugin.js";
import { NamespaceTool$outboundSchema, } from "./namespacetool.js";
import { OpenAIResponsesToolChoiceUnion$outboundSchema, } from "./openairesponsestoolchoiceunion.js";
import { OpenAIResponsesTruncation$outboundSchema, } from "./openairesponsestruncation.js";
import { OutputModalityEnum$outboundSchema, } from "./outputmodalityenum.js";
import { ParetoRouterPlugin$outboundSchema, } from "./paretorouterplugin.js";
import { Preview20250311WebSearchServerTool$outboundSchema, } from "./preview20250311websearchservertool.js";
import { PreviewWebSearchServerTool$outboundSchema, } from "./previewwebsearchservertool.js";
import { PromptCacheOptions$outboundSchema, } from "./promptcacheoptions.js";
import { ProviderPreferences$outboundSchema, } from "./providerpreferences.js";
import { ReasoningContext$outboundSchema, } from "./reasoningcontext.js";
import { ReasoningEffort$outboundSchema, } from "./reasoningeffort.js";
import { ReasoningMode$outboundSchema, } from "./reasoningmode.js";
import { ReasoningSummaryVerbosity$outboundSchema, } from "./reasoningsummaryverbosity.js";
import { ResponseHealingPlugin$outboundSchema, } from "./responsehealingplugin.js";
import { ResponseIncludesEnum$outboundSchema, } from "./responseincludesenum.js";
import { SearchModelsServerToolOpenRouter$outboundSchema, } from "./searchmodelsservertoolopenrouter.js";
import { ShellServerTool$outboundSchema, } from "./shellservertool.js";
import { ShellServerToolOpenRouter$outboundSchema, } from "./shellservertoolopenrouter.js";
import { StopServerToolsWhenCondition$outboundSchema, } from "./stopservertoolswhencondition.js";
import { StoredPromptTemplate$outboundSchema, } from "./storedprompttemplate.js";
import { SubagentServerToolOpenRouter$outboundSchema, } from "./subagentservertoolopenrouter.js";
import { TextExtendedConfig$outboundSchema, } from "./textextendedconfig.js";
import { TraceConfig$outboundSchema, } from "./traceconfig.js";
import { WebFetchPlugin$outboundSchema, } from "./webfetchplugin.js";
import { WebFetchServerTool$outboundSchema, } from "./webfetchservertool.js";
import { WebSearchPlugin$outboundSchema, } from "./websearchplugin.js";
import { WebSearchServerTool$outboundSchema, } from "./websearchservertool.js";
import { WebSearchServerToolOpenRouter$outboundSchema, } from "./websearchservertoolopenrouter.js";
export const ResponsesRequestServiceTier = {
    Auto: "auto",
    Default: "default",
    Flex: "flex",
    Priority: "priority",
    Scale: "scale",
};
/** @internal */
export const ResponsesRequestPlugin$outboundSchema = z.union([
    AutoBetaRouterPlugin$outboundSchema,
    AutoRouterPlugin$outboundSchema,
    ContextCompressionPlugin$outboundSchema,
    FileParserPlugin$outboundSchema,
    FusionPlugin$outboundSchema,
    ModerationPlugin$outboundSchema,
    ParetoRouterPlugin$outboundSchema,
    ResponseHealingPlugin$outboundSchema,
    WebSearchPlugin$outboundSchema,
    WebFetchPlugin$outboundSchema,
]);
export function responsesRequestPluginToJSON(responsesRequestPlugin) {
    return JSON.stringify(ResponsesRequestPlugin$outboundSchema.parse(responsesRequestPlugin));
}
/** @internal */
export const ReasoningConfig$outboundSchema = z.object({
    context: z.nullable(ReasoningContext$outboundSchema).optional(),
    effort: z.nullable(ReasoningEffort$outboundSchema).optional(),
    mode: z.nullable(ReasoningMode$outboundSchema).optional(),
    summary: z.nullable(ReasoningSummaryVerbosity$outboundSchema).optional(),
    enabled: z.nullable(z.boolean()).optional(),
    maxTokens: z.nullable(z.int()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxTokens: "max_tokens",
    });
});
export function reasoningConfigToJSON(reasoningConfig) {
    return JSON.stringify(ReasoningConfig$outboundSchema.parse(reasoningConfig));
}
/** @internal */
export const ResponsesRequestServiceTier$outboundSchema = openEnums.outboundSchema(ResponsesRequestServiceTier);
/** @internal */
export const ResponsesRequestToolFunction$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    name: z.string(),
    parameters: z.nullable(z.record(z.string(), z.any())),
    strict: z.nullable(z.boolean()).optional(),
    type: z.literal("function"),
});
export function responsesRequestToolFunctionToJSON(responsesRequestToolFunction) {
    return JSON.stringify(ResponsesRequestToolFunction$outboundSchema.parse(responsesRequestToolFunction));
}
/** @internal */
export const ResponsesRequestToolUnion$outboundSchema = z.union([
    z.lazy(() => ResponsesRequestToolFunction$outboundSchema),
    PreviewWebSearchServerTool$outboundSchema.and(z.object({ type: z.literal("web_search_preview") })),
    Preview20250311WebSearchServerTool$outboundSchema.and(z.object({ type: z.literal("web_search_preview_2025_03_11") })),
    LegacyWebSearchServerTool$outboundSchema.and(z.object({ type: z.literal("web_search") })),
    WebSearchServerTool$outboundSchema.and(z.object({ type: z.literal("web_search_2025_08_26") })),
    FileSearchServerTool$outboundSchema.and(z.object({ type: z.literal("file_search") })),
    ComputerUseServerTool$outboundSchema.and(z.object({ type: z.literal("computer_use_preview") })),
    CodeInterpreterServerTool$outboundSchema.and(z.object({ type: z.literal("code_interpreter") })),
    McpServerTool$outboundSchema.and(z.object({ type: z.literal("mcp") })),
    ImageGenerationServerTool$outboundSchema.and(z.object({ type: z.literal("image_generation") })),
    CodexLocalShellTool$outboundSchema.and(z.object({ type: z.literal("local_shell") })),
    ShellServerTool$outboundSchema.and(z.object({ type: z.literal("shell") })),
    ApplyPatchServerTool$outboundSchema.and(z.object({ type: z.literal("apply_patch") })),
    CustomTool$outboundSchema.and(z.object({ type: z.literal("custom") })),
    NamespaceTool$outboundSchema.and(z.object({ type: z.literal("namespace") })),
    AdvisorServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:advisor") })),
    SubagentServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:subagent") })),
    DatetimeServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:datetime") })),
    FilesServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:files") })),
    FusionServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:fusion") })),
    ImageGenerationServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:image_generation") })),
    SearchModelsServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:experimental__search_models") })),
    WebFetchServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:web_fetch") })),
    WebSearchServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:web_search") })),
    ApplyPatchServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:apply_patch") })),
    BashServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:bash") })),
    ShellServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:shell") })),
]);
export function responsesRequestToolUnionToJSON(responsesRequestToolUnion) {
    return JSON.stringify(ResponsesRequestToolUnion$outboundSchema.parse(responsesRequestToolUnion));
}
/** @internal */
export const ResponsesRequest$outboundSchema = z.object({
    background: z.nullable(z.boolean()).optional(),
    cacheControl: AnthropicCacheControlDirective$outboundSchema.optional(),
    debug: ChatDebugOptions$outboundSchema.optional(),
    frequencyPenalty: z.nullable(z.number()).optional(),
    imageConfig: z.record(z.string(), ImageConfig$outboundSchema).optional(),
    include: z.nullable(z.array(ResponseIncludesEnum$outboundSchema)).optional(),
    input: InputsUnion$outboundSchema.optional(),
    instructions: z.nullable(z.string()).optional(),
    maxOutputTokens: z.nullable(z.int()).optional(),
    maxToolCalls: z.nullable(z.int()).optional(),
    metadata: z.nullable(z.record(z.string(), z.string())).optional(),
    modalities: z.array(OutputModalityEnum$outboundSchema).optional(),
    model: z.string().optional(),
    models: z.array(z.string()).optional(),
    parallelToolCalls: z.nullable(z.boolean()).optional(),
    plugins: z.array(z.union([
        AutoBetaRouterPlugin$outboundSchema,
        AutoRouterPlugin$outboundSchema,
        ContextCompressionPlugin$outboundSchema,
        FileParserPlugin$outboundSchema,
        FusionPlugin$outboundSchema,
        ModerationPlugin$outboundSchema,
        ParetoRouterPlugin$outboundSchema,
        ResponseHealingPlugin$outboundSchema,
        WebSearchPlugin$outboundSchema,
        WebFetchPlugin$outboundSchema,
    ])).optional(),
    presencePenalty: z.nullable(z.number()).optional(),
    previousResponseId: z.any().optional(),
    prompt: z.nullable(StoredPromptTemplate$outboundSchema).optional(),
    promptCacheKey: z.nullable(z.string()).optional(),
    promptCacheOptions: z.nullable(PromptCacheOptions$outboundSchema).optional(),
    provider: z.nullable(ProviderPreferences$outboundSchema).optional(),
    reasoning: z.nullable(z.lazy(() => ReasoningConfig$outboundSchema))
        .optional(),
    safetyIdentifier: z.nullable(z.string()).optional(),
    serviceTier: z.nullable(ResponsesRequestServiceTier$outboundSchema.default("auto")),
    sessionId: z.string().optional(),
    stopServerToolsWhen: z.array(StopServerToolsWhenCondition$outboundSchema)
        .optional(),
    store: z.literal(false).default(false),
    stream: z.boolean().default(false),
    temperature: z.nullable(z.number()).optional(),
    text: TextExtendedConfig$outboundSchema.optional(),
    toolChoice: OpenAIResponsesToolChoiceUnion$outboundSchema.optional(),
    tools: z.array(z.union([
        z.lazy(() => ResponsesRequestToolFunction$outboundSchema),
        PreviewWebSearchServerTool$outboundSchema.and(z.object({ type: z.literal("web_search_preview") })),
        Preview20250311WebSearchServerTool$outboundSchema.and(z.object({ type: z.literal("web_search_preview_2025_03_11") })),
        LegacyWebSearchServerTool$outboundSchema.and(z.object({ type: z.literal("web_search") })),
        WebSearchServerTool$outboundSchema.and(z.object({ type: z.literal("web_search_2025_08_26") })),
        FileSearchServerTool$outboundSchema.and(z.object({ type: z.literal("file_search") })),
        ComputerUseServerTool$outboundSchema.and(z.object({ type: z.literal("computer_use_preview") })),
        CodeInterpreterServerTool$outboundSchema.and(z.object({ type: z.literal("code_interpreter") })),
        McpServerTool$outboundSchema.and(z.object({ type: z.literal("mcp") })),
        ImageGenerationServerTool$outboundSchema.and(z.object({ type: z.literal("image_generation") })),
        CodexLocalShellTool$outboundSchema.and(z.object({ type: z.literal("local_shell") })),
        ShellServerTool$outboundSchema.and(z.object({ type: z.literal("shell") })),
        ApplyPatchServerTool$outboundSchema.and(z.object({ type: z.literal("apply_patch") })),
        CustomTool$outboundSchema.and(z.object({ type: z.literal("custom") })),
        NamespaceTool$outboundSchema.and(z.object({ type: z.literal("namespace") })),
        AdvisorServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:advisor") })),
        SubagentServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:subagent") })),
        DatetimeServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:datetime") })),
        FilesServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:files") })),
        FusionServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:fusion") })),
        ImageGenerationServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:image_generation") })),
        SearchModelsServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:experimental__search_models") })),
        WebFetchServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:web_fetch") })),
        WebSearchServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:web_search") })),
        ApplyPatchServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:apply_patch") })),
        BashServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:bash") })),
        ShellServerToolOpenRouter$outboundSchema.and(z.object({ type: z.literal("openrouter:shell") })),
    ])).optional(),
    topK: z.int().optional(),
    topLogprobs: z.nullable(z.int()).optional(),
    topP: z.nullable(z.number()).optional(),
    trace: TraceConfig$outboundSchema.optional(),
    truncation: z.nullable(OpenAIResponsesTruncation$outboundSchema).optional(),
    user: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
        frequencyPenalty: "frequency_penalty",
        imageConfig: "image_config",
        maxOutputTokens: "max_output_tokens",
        maxToolCalls: "max_tool_calls",
        parallelToolCalls: "parallel_tool_calls",
        presencePenalty: "presence_penalty",
        previousResponseId: "previous_response_id",
        promptCacheKey: "prompt_cache_key",
        promptCacheOptions: "prompt_cache_options",
        safetyIdentifier: "safety_identifier",
        serviceTier: "service_tier",
        sessionId: "session_id",
        stopServerToolsWhen: "stop_server_tools_when",
        toolChoice: "tool_choice",
        topK: "top_k",
        topLogprobs: "top_logprobs",
        topP: "top_p",
    });
});
export function responsesRequestToJSON(responsesRequest) {
    return JSON.stringify(ResponsesRequest$outboundSchema.parse(responsesRequest));
}
