// @openrouter/sdk/esm/models/shellcalloutputitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7fbbfc5a6ebc
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const ShellCallOutputItemType = {
    ShellCallOutput: "shell_call_output",
};
/** @internal */
export const ShellCallOutputItemOutput$outboundSchema = z.object({
    content: z.nullable(z.string()).optional(),
    exitCode: z.nullable(z.int()).optional(),
    type: z.string(),
    additionalProperties: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            exitCode: "exit_code",
            additionalProperties: null,
        }),
    };
});
export function shellCallOutputItemOutputToJSON(shellCallOutputItemOutput) {
    return JSON.stringify(ShellCallOutputItemOutput$outboundSchema.parse(shellCallOutputItemOutput));
}
/** @internal */
export const ShellCallOutputItemType$outboundSchema = z.enum(ShellCallOutputItemType);
/** @internal */
export const ShellCallOutputItem$outboundSchema = z.object({
    callId: z.string(),
    id: z.nullable(z.string()).optional(),
    maxOutputLength: z.nullable(z.int()).optional(),
    output: z.array(z.lazy(() => ShellCallOutputItemOutput$outboundSchema)),
    status: z.nullable(ToolCallStatus$outboundSchema).optional(),
    type: ShellCallOutputItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
        maxOutputLength: "max_output_length",
    });
});
export function shellCallOutputItemToJSON(shellCallOutputItem) {
    return JSON.stringify(ShellCallOutputItem$outboundSchema.parse(shellCallOutputItem));
}
