// @openrouter/sdk/esm/models/shellcallitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a94bd412c431
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const ShellCallItemType = {
    ShellCall: "shell_call",
};
/** @internal */
export const ShellCallItemAction$outboundSchema = z.object({
    commands: z.array(z.string()),
    maxOutputLength: z.nullable(z.int()).optional(),
    timeoutMs: z.nullable(z.int()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxOutputLength: "max_output_length",
        timeoutMs: "timeout_ms",
    });
});
export function shellCallItemActionToJSON(shellCallItemAction) {
    return JSON.stringify(ShellCallItemAction$outboundSchema.parse(shellCallItemAction));
}
/** @internal */
export const ShellCallItemType$outboundSchema = z.enum(ShellCallItemType);
/** @internal */
export const ShellCallItem$outboundSchema = z.object({
    action: z.lazy(() => ShellCallItemAction$outboundSchema),
    callId: z.string(),
    environment: z.any().optional(),
    id: z.nullable(z.string()).optional(),
    status: z.nullable(ToolCallStatus$outboundSchema).optional(),
    type: ShellCallItemType$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function shellCallItemToJSON(shellCallItem) {
    return JSON.stringify(ShellCallItem$outboundSchema.parse(shellCallItem));
}
