// @openrouter/sdk/esm/models/shellservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4bd6823cb793
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const ShellServerToolType = {
    Shell: "shell",
};
/** @internal */
export const ShellServerToolType$inboundSchema = z.enum(ShellServerToolType);
/** @internal */
export const ShellServerToolType$outboundSchema = ShellServerToolType$inboundSchema;
/** @internal */
export const ShellServerTool$inboundSchema = z.object({
    type: ShellServerToolType$inboundSchema,
});
/** @internal */
export const ShellServerTool$outboundSchema = z.object({
    type: ShellServerToolType$outboundSchema,
});
export function shellServerToolToJSON(shellServerTool) {
    return JSON.stringify(ShellServerTool$outboundSchema.parse(shellServerTool));
}
export function shellServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ShellServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShellServerTool' from JSON`);
}
