// @openrouter/sdk/esm/models/speechrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 96d573d68420
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ProviderOptions$outboundSchema, } from "./provideroptions.js";
/**
 * Audio output format
 */
export const SpeechRequestResponseFormat = {
    Mp3: "mp3",
    Pcm: "pcm",
};
/** @internal */
export const SpeechRequestProvider$outboundSchema = z.object({
    options: ProviderOptions$outboundSchema.optional(),
});
export function speechRequestProviderToJSON(speechRequestProvider) {
    return JSON.stringify(SpeechRequestProvider$outboundSchema.parse(speechRequestProvider));
}
/** @internal */
export const SpeechRequestResponseFormat$outboundSchema = openEnums.outboundSchema(SpeechRequestResponseFormat);
/** @internal */
export const SpeechRequest$outboundSchema = z.object({
    input: z.string(),
    model: z.string(),
    provider: z.lazy(() => SpeechRequestProvider$outboundSchema).optional(),
    responseFormat: SpeechRequestResponseFormat$outboundSchema.default("pcm"),
    speed: z.number().optional(),
    voice: z.string(),
}).transform((v) => {
    return remap$(v, {
        responseFormat: "response_format",
    });
});
export function speechRequestToJSON(speechRequest) {
    return JSON.stringify(SpeechRequest$outboundSchema.parse(speechRequest));
}
