// @openrouter/sdk/esm/models/sttrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3be8453c0517
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ProviderOptions$outboundSchema, } from "./provideroptions.js";
import { STTInputAudio$outboundSchema, } from "./sttinputaudio.js";
import { STTTimestampGranularity$outboundSchema, } from "./stttimestampgranularity.js";
/**
 * Output format. "json" (default) returns { text, usage }. "verbose_json" additionally returns task, language, duration, and segment-level timestamps; only supported by OpenAI-compatible providers.
 */
export const STTRequestResponseFormat = {
    Json: "json",
    VerboseJson: "verbose_json",
};
/** @internal */
export const STTRequestProvider$outboundSchema = z.object({
    options: ProviderOptions$outboundSchema.optional(),
});
export function sttRequestProviderToJSON(sttRequestProvider) {
    return JSON.stringify(STTRequestProvider$outboundSchema.parse(sttRequestProvider));
}
/** @internal */
export const STTRequestResponseFormat$outboundSchema = openEnums.outboundSchema(STTRequestResponseFormat);
/** @internal */
export const STTRequest$outboundSchema = z.object({
    inputAudio: STTInputAudio$outboundSchema,
    language: z.string().optional(),
    model: z.string(),
    provider: z.lazy(() => STTRequestProvider$outboundSchema).optional(),
    responseFormat: STTRequestResponseFormat$outboundSchema.optional(),
    temperature: z.number().optional(),
    timestampGranularities: z.array(STTTimestampGranularity$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        inputAudio: "input_audio",
        responseFormat: "response_format",
        timestampGranularities: "timestamp_granularities",
    });
});
export function sttRequestToJSON(sttRequest) {
    return JSON.stringify(STTRequest$outboundSchema.parse(sttRequest));
}
