// @openrouter/sdk/esm/models/sttsegment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 610157864a1c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const STTSegment$inboundSchema = z
    .object({
    avg_logprob: z.number().optional(),
    compression_ratio: z.number().optional(),
    end: z.number(),
    id: z.int(),
    no_speech_prob: z.number().optional(),
    seek: z.int().optional(),
    speaker: z.int().optional(),
    start: z.number(),
    temperature: z.number().optional(),
    text: z.string(),
    tokens: z.array(z.int()).optional(),
}).transform((v) => {
    return remap$(v, {
        "avg_logprob": "avgLogprob",
        "compression_ratio": "compressionRatio",
        "no_speech_prob": "noSpeechProb",
    });
});
export function sttSegmentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => STTSegment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'STTSegment' from JSON`);
}
