// @openrouter/sdk/esm/models/subagentservertoolconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6f8ec51577e6
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { SubagentNestedTool$outboundSchema, } from "./subagentnestedtool.js";
import { SubagentReasoning$outboundSchema, } from "./subagentreasoning.js";
/** @internal */
export const SubagentServerToolConfig$outboundSchema = z.object({
    instructions: z.string().optional(),
    maxCompletionTokens: z.int().optional(),
    maxToolCalls: z.int().optional(),
    model: z.string().optional(),
    name: z.string().optional(),
    reasoning: SubagentReasoning$outboundSchema.optional(),
    temperature: z.number().optional(),
    tools: z.array(SubagentNestedTool$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        maxCompletionTokens: "max_completion_tokens",
        maxToolCalls: "max_tool_calls",
    });
});
export function subagentServerToolConfigToJSON(subagentServerToolConfig) {
    return JSON.stringify(SubagentServerToolConfig$outboundSchema.parse(subagentServerToolConfig));
}
