// @openrouter/sdk/esm/models/subagentreasoning.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 916aba5c0712
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
/**
 * Reasoning effort level for the subagent call.
 */
export const SubagentReasoningEffort = {
    Max: "max",
    Xhigh: "xhigh",
    High: "high",
    Medium: "medium",
    Low: "low",
    Minimal: "minimal",
    None: "none",
};
/** @internal */
export const SubagentReasoningEffort$outboundSchema = openEnums.outboundSchema(SubagentReasoningEffort);
/** @internal */
export const SubagentReasoning$outboundSchema = z.object({
    effort: SubagentReasoningEffort$outboundSchema.optional(),
    maxTokens: z.int().optional(),
}).transform((v) => {
    return remap$(v, {
        maxTokens: "max_tokens",
    });
});
export function subagentReasoningToJSON(subagentReasoning) {
    return JSON.stringify(SubagentReasoning$outboundSchema.parse(subagentReasoning));
}
