// @openrouter/sdk/esm/models/submitgenerationfeedbackrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c8bd4306e20e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
/**
 * The category of feedback being reported
 */
export const Category = {
    Latency: "latency",
    Incoherence: "incoherence",
    IncorrectResponse: "incorrect_response",
    Formatting: "formatting",
    Billing: "billing",
    ApiError: "api_error",
    Other: "other",
};
/** @internal */
export const Category$outboundSchema = openEnums
    .outboundSchema(Category);
/** @internal */
export const SubmitGenerationFeedbackRequest$outboundSchema = z.object({
    category: Category$outboundSchema,
    comment: z.string().optional(),
    generationId: z.string(),
}).transform((v) => {
    return remap$(v, {
        generationId: "generation_id",
    });
});
export function submitGenerationFeedbackRequestToJSON(submitGenerationFeedbackRequest) {
    return JSON.stringify(SubmitGenerationFeedbackRequest$outboundSchema.parse(submitGenerationFeedbackRequest));
}
