// @openrouter/sdk/esm/models/taskclassificationresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e618eb0802a7
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { TaskClassificationItem$inboundSchema, } from "./taskclassificationitem.js";
import { TaskClassificationMacroCategory$inboundSchema, } from "./taskclassificationmacrocategory.js";
/** @internal */
export const TaskClassificationResponseData$inboundSchema = z.object({
    as_of: z.string(),
    classifications: z.array(TaskClassificationItem$inboundSchema),
    macro_categories: z.array(TaskClassificationMacroCategory$inboundSchema),
    window_days: z.int(),
}).transform((v) => {
    return remap$(v, {
        "as_of": "asOf",
        "macro_categories": "macroCategories",
        "window_days": "windowDays",
    });
});
export function taskClassificationResponseDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TaskClassificationResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TaskClassificationResponseData' from JSON`);
}
/** @internal */
export const TaskClassificationResponse$inboundSchema = z.object({
    data: z.lazy(() => TaskClassificationResponseData$inboundSchema),
});
export function taskClassificationResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TaskClassificationResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TaskClassificationResponse' from JSON`);
}
