// @openrouter/sdk/esm/models/taskclassificationitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 58896645e490
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { TaskClassificationModel$inboundSchema, } from "./taskclassificationmodel.js";
/** @internal */
export const TaskClassificationItem$inboundSchema = z.object({
    category_token_share: z.number(),
    category_usage_share: z.number(),
    display_name: z.string(),
    macro_category: z.string(),
    models: z.array(TaskClassificationModel$inboundSchema),
    tag: z.string(),
    token_share: z.number(),
    usage_share: z.number(),
}).transform((v) => {
    return remap$(v, {
        "category_token_share": "categoryTokenShare",
        "category_usage_share": "categoryUsageShare",
        "display_name": "displayName",
        "macro_category": "macroCategory",
        "token_share": "tokenShare",
        "usage_share": "usageShare",
    });
});
export function taskClassificationItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TaskClassificationItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TaskClassificationItem' from JSON`);
}
