// @openrouter/sdk/esm/models/unifiedbenchmarksmeta.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d146ee8b8956
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
/**
 * The source filter applied, or null when all sources are returned.
 */
export const SourceEnum = {
    ArtificialAnalysis: "artificial-analysis",
    DesignArena: "design-arena",
};
/**
 * Dataset version.
 */
export const UnifiedBenchmarksMetaVersion = {
    V1: "v1",
};
/** @internal */
export const SourceEnum$inboundSchema = openEnums.inboundSchema(SourceEnum);
/** @internal */
export const UnifiedBenchmarksMetaVersion$inboundSchema = z.enum(UnifiedBenchmarksMetaVersion);
/** @internal */
export const UnifiedBenchmarksMeta$inboundSchema = z.object({
    as_of: z.string(),
    citation: z.nullable(z.string()),
    model_count: z.int(),
    source: z.nullable(SourceEnum$inboundSchema),
    source_url: z.nullable(z.string()),
    task_type: z.nullable(z.string()),
    version: UnifiedBenchmarksMetaVersion$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "as_of": "asOf",
        "model_count": "modelCount",
        "source_url": "sourceUrl",
        "task_type": "taskType",
    });
});
export function unifiedBenchmarksMetaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UnifiedBenchmarksMeta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UnifiedBenchmarksMeta' from JSON`);
}
