// @openrouter/sdk/esm/models/unifiedbenchmarksdaitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f911314affb3
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { UnifiedBenchmarkPricing$inboundSchema, } from "./unifiedbenchmarkpricing.js";
/** @internal */
export const TournamentStats$inboundSchema = z.object({
    first_place: z.nullable(z.int()),
    fourth_place: z.nullable(z.int()),
    second_place: z.nullable(z.int()),
    third_place: z.nullable(z.int()),
    total: z.nullable(z.int()),
}).transform((v) => {
    return remap$(v, {
        "first_place": "firstPlace",
        "fourth_place": "fourthPlace",
        "second_place": "secondPlace",
        "third_place": "thirdPlace",
    });
});
export function tournamentStatsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TournamentStats$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TournamentStats' from JSON`);
}
/** @internal */
export const UnifiedBenchmarksDAItem$inboundSchema = z.object({
    arena: z.string(),
    avg_generation_time_ms: z.nullable(z.number()),
    category: z.string(),
    display_name: z.string(),
    elo: z.number(),
    model_permaslug: z.string(),
    pricing: z.nullable(UnifiedBenchmarkPricing$inboundSchema),
    source: z.literal("design-arena"),
    tournament_stats: z.lazy(() => TournamentStats$inboundSchema),
    win_rate: z.number(),
}).transform((v) => {
    return remap$(v, {
        "avg_generation_time_ms": "avgGenerationTimeMs",
        "display_name": "displayName",
        "model_permaslug": "modelPermaslug",
        "tournament_stats": "tournamentStats",
        "win_rate": "winRate",
    });
});
export function unifiedBenchmarksDAItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UnifiedBenchmarksDAItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UnifiedBenchmarksDAItem' from JSON`);
}
