// @openrouter/sdk/esm/models/unifiedbenchmarksresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b3f08556727b
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { discriminatedUnion } from "../types/discriminatedUnion.js";
import { UnifiedBenchmarksAAItem$inboundSchema, } from "./unifiedbenchmarksaaitem.js";
import { UnifiedBenchmarksDAItem$inboundSchema, } from "./unifiedbenchmarksdaitem.js";
import { UnifiedBenchmarksMeta$inboundSchema, } from "./unifiedbenchmarksmeta.js";
/** @internal */
export const UnifiedBenchmarksResponseData$inboundSchema = discriminatedUnion("source", {
    ["artificial-analysis"]: UnifiedBenchmarksAAItem$inboundSchema,
    ["design-arena"]: UnifiedBenchmarksDAItem$inboundSchema,
});
export function unifiedBenchmarksResponseDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UnifiedBenchmarksResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UnifiedBenchmarksResponseData' from JSON`);
}
/** @internal */
export const UnifiedBenchmarksResponse$inboundSchema = z.object({
    data: z.array(discriminatedUnion("source", {
        ["artificial-analysis"]: UnifiedBenchmarksAAItem$inboundSchema,
        ["design-arena"]: UnifiedBenchmarksDAItem$inboundSchema,
    })),
    meta: UnifiedBenchmarksMeta$inboundSchema,
});
export function unifiedBenchmarksResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UnifiedBenchmarksResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UnifiedBenchmarksResponse' from JSON`);
}
