// @openrouter/sdk/esm/models/unifiedbenchmarksaaitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 436660899947
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { UnifiedBenchmarkPricing$inboundSchema, } from "./unifiedbenchmarkpricing.js";
/** @internal */
export const UnifiedBenchmarksAAItem$inboundSchema = z.object({
    agentic_index: z.nullable(z.number()),
    coding_index: z.nullable(z.number()),
    display_name: z.string(),
    intelligence_index: z.nullable(z.number()),
    model_permaslug: z.string(),
    pricing: z.nullable(UnifiedBenchmarkPricing$inboundSchema),
    source: z.literal("artificial-analysis"),
}).transform((v) => {
    return remap$(v, {
        "agentic_index": "agenticIndex",
        "coding_index": "codingIndex",
        "display_name": "displayName",
        "intelligence_index": "intelligenceIndex",
        "model_permaslug": "modelPermaslug",
    });
});
export function unifiedBenchmarksAAItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UnifiedBenchmarksAAItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UnifiedBenchmarksAAItem' from JSON`);
}
