// @openrouter/sdk/esm/models/updateobservabilitydestinationrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b5f21f0b6b97
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ObservabilityFilterRulesConfigNullable$outboundSchema, } from "./observabilityfilterrulesconfignullable.js";
/** @internal */
export const UpdateObservabilityDestinationRequest$outboundSchema = z.object({
    apiKeyHashes: z.nullable(z.array(z.string())).optional(),
    config: z.record(z.string(), z.any()).optional(),
    enabled: z.boolean().optional(),
    filterRules: z.nullable(ObservabilityFilterRulesConfigNullable$outboundSchema)
        .optional(),
    name: z.string().optional(),
    privacyMode: z.boolean().optional(),
    samplingRate: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        apiKeyHashes: "api_key_hashes",
        filterRules: "filter_rules",
        privacyMode: "privacy_mode",
        samplingRate: "sampling_rate",
    });
});
export function updateObservabilityDestinationRequestToJSON(updateObservabilityDestinationRequest) {
    return JSON.stringify(UpdateObservabilityDestinationRequest$outboundSchema.parse(updateObservabilityDestinationRequest));
}
