// @openrouter/sdk/esm/models/videogenerationrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e72ea2e4c5b3
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { FrameImage$outboundSchema, } from "./frameimage.js";
import { InputReference$outboundSchema, } from "./inputreference.js";
/**
 * Aspect ratio of the generated video
 */
export const VideoGenerationRequestAspectRatio = {
    OneHundredAndSixtyNine: "16:9",
    NineHundredAndSixteen: "9:16",
    Eleven: "1:1",
    FortyThree: "4:3",
    ThirtyFour: "3:4",
    ThirtyTwo: "3:2",
    TwentyThree: "2:3",
    TwoHundredAndNineteen: "21:9",
    NineHundredAndTwentyOne: "9:21",
};
/**
 * Resolution of the generated video
 */
export const VideoGenerationRequestResolution = {
    FourHundredAndEightyp: "480p",
    SevenHundredAndTwentyp: "720p",
    OneThousandAndEightyp: "1080p",
    OneK: "1K",
    TwoK: "2K",
    FourK: "4K",
};
/** @internal */
export const VideoGenerationRequestAspectRatio$outboundSchema = openEnums.outboundSchema(VideoGenerationRequestAspectRatio);
/** @internal */
export const VideoGenerationRequestOptions$outboundSchema = z.object({
    oneai: z.record(z.string(), z.any()).optional(),
    ai21: z.record(z.string(), z.any()).optional(),
    aionLabs: z.record(z.string(), z.any()).optional(),
    akashml: z.record(z.string(), z.any()).optional(),
    alibaba: z.record(z.string(), z.any()).optional(),
    amazonBedrock: z.record(z.string(), z.any()).optional(),
    amazonNova: z.record(z.string(), z.any()).optional(),
    ambient: z.record(z.string(), z.any()).optional(),
    anthropic: z.record(z.string(), z.any()).optional(),
    anyscale: z.record(z.string(), z.any()).optional(),
    arceeAi: z.record(z.string(), z.any()).optional(),
    atlasCloud: z.record(z.string(), z.any()).optional(),
    atoma: z.record(z.string(), z.any()).optional(),
    avian: z.record(z.string(), z.any()).optional(),
    azure: z.record(z.string(), z.any()).optional(),
    baidu: z.record(z.string(), z.any()).optional(),
    baseten: z.record(z.string(), z.any()).optional(),
    blackForestLabs: z.record(z.string(), z.any()).optional(),
    byteplus: z.record(z.string(), z.any()).optional(),
    centml: z.record(z.string(), z.any()).optional(),
    cerebras: z.record(z.string(), z.any()).optional(),
    chutes: z.record(z.string(), z.any()).optional(),
    cirrascale: z.record(z.string(), z.any()).optional(),
    clarifai: z.record(z.string(), z.any()).optional(),
    cloudflare: z.record(z.string(), z.any()).optional(),
    cohere: z.record(z.string(), z.any()).optional(),
    coreweave: z.record(z.string(), z.any()).optional(),
    crofai: z.record(z.string(), z.any()).optional(),
    crucible: z.record(z.string(), z.any()).optional(),
    crusoe: z.record(z.string(), z.any()).optional(),
    darkbloom: z.record(z.string(), z.any()).optional(),
    decart: z.record(z.string(), z.any()).optional(),
    deepgram: z.record(z.string(), z.any()).optional(),
    deepinfra: z.record(z.string(), z.any()).optional(),
    deepseek: z.record(z.string(), z.any()).optional(),
    dekallm: z.record(z.string(), z.any()).optional(),
    digitalocean: z.record(z.string(), z.any()).optional(),
    enfer: z.record(z.string(), z.any()).optional(),
    fakeProvider: z.record(z.string(), z.any()).optional(),
    featherless: z.record(z.string(), z.any()).optional(),
    fireworks: z.record(z.string(), z.any()).optional(),
    fishAudio: z.record(z.string(), z.any()).optional(),
    friendli: z.record(z.string(), z.any()).optional(),
    gmicloud: z.record(z.string(), z.any()).optional(),
    googleAiStudio: z.record(z.string(), z.any()).optional(),
    googleVertex: z.record(z.string(), z.any()).optional(),
    gopomelo: z.record(z.string(), z.any()).optional(),
    groq: z.record(z.string(), z.any()).optional(),
    heygen: z.record(z.string(), z.any()).optional(),
    huggingface: z.record(z.string(), z.any()).optional(),
    hyperbolic: z.record(z.string(), z.any()).optional(),
    hyperbolicQuantized: z.record(z.string(), z.any()).optional(),
    inception: z.record(z.string(), z.any()).optional(),
    inceptron: z.record(z.string(), z.any()).optional(),
    inferactVllm: z.record(z.string(), z.any()).optional(),
    inferenceNet: z.record(z.string(), z.any()).optional(),
    infermatic: z.record(z.string(), z.any()).optional(),
    inflection: z.record(z.string(), z.any()).optional(),
    inocloud: z.record(z.string(), z.any()).optional(),
    ioNet: z.record(z.string(), z.any()).optional(),
    ionstream: z.record(z.string(), z.any()).optional(),
    klusterai: z.record(z.string(), z.any()).optional(),
    krea: z.record(z.string(), z.any()).optional(),
    lambda: z.record(z.string(), z.any()).optional(),
    lepton: z.record(z.string(), z.any()).optional(),
    liquid: z.record(z.string(), z.any()).optional(),
    lynn: z.record(z.string(), z.any()).optional(),
    lynnPrivate: z.record(z.string(), z.any()).optional(),
    mancer: z.record(z.string(), z.any()).optional(),
    mancerOld: z.record(z.string(), z.any()).optional(),
    mara: z.record(z.string(), z.any()).optional(),
    meta: z.record(z.string(), z.any()).optional(),
    minimax: z.record(z.string(), z.any()).optional(),
    mistral: z.record(z.string(), z.any()).optional(),
    modal: z.record(z.string(), z.any()).optional(),
    modelrun: z.record(z.string(), z.any()).optional(),
    modular: z.record(z.string(), z.any()).optional(),
    moonshotai: z.record(z.string(), z.any()).optional(),
    morph: z.record(z.string(), z.any()).optional(),
    ncompass: z.record(z.string(), z.any()).optional(),
    nebius: z.record(z.string(), z.any()).optional(),
    nexAgi: z.record(z.string(), z.any()).optional(),
    nextbit: z.record(z.string(), z.any()).optional(),
    nineteen: z.record(z.string(), z.any()).optional(),
    novita: z.record(z.string(), z.any()).optional(),
    nvidia: z.record(z.string(), z.any()).optional(),
    octoai: z.record(z.string(), z.any()).optional(),
    openInference: z.record(z.string(), z.any()).optional(),
    openai: z.record(z.string(), z.any()).optional(),
    parasail: z.record(z.string(), z.any()).optional(),
    perceptron: z.record(z.string(), z.any()).optional(),
    perplexity: z.record(z.string(), z.any()).optional(),
    phala: z.record(z.string(), z.any()).optional(),
    poolside: z.record(z.string(), z.any()).optional(),
    quiver: z.record(z.string(), z.any()).optional(),
    recraft: z.record(z.string(), z.any()).optional(),
    recursal: z.record(z.string(), z.any()).optional(),
    reflection: z.record(z.string(), z.any()).optional(),
    reka: z.record(z.string(), z.any()).optional(),
    relace: z.record(z.string(), z.any()).optional(),
    replicate: z.record(z.string(), z.any()).optional(),
    runway: z.record(z.string(), z.any()).optional(),
    sailResearch: z.record(z.string(), z.any()).optional(),
    sakana: z.record(z.string(), z.any()).optional(),
    sakanaAi: z.record(z.string(), z.any()).optional(),
    sambanova: z.record(z.string(), z.any()).optional(),
    sambanovaCloaked: z.record(z.string(), z.any()).optional(),
    seed: z.record(z.string(), z.any()).optional(),
    sfCompute: z.record(z.string(), z.any()).optional(),
    siliconflow: z.record(z.string(), z.any()).optional(),
    sourceful: z.record(z.string(), z.any()).optional(),
    stealth: z.record(z.string(), z.any()).optional(),
    stepfun: z.record(z.string(), z.any()).optional(),
    streamlake: z.record(z.string(), z.any()).optional(),
    switchpoint: z.record(z.string(), z.any()).optional(),
    targon: z.record(z.string(), z.any()).optional(),
    tencent: z.record(z.string(), z.any()).optional(),
    tenstorrent: z.record(z.string(), z.any()).optional(),
    together: z.record(z.string(), z.any()).optional(),
    togetherLite: z.record(z.string(), z.any()).optional(),
    ubicloud: z.record(z.string(), z.any()).optional(),
    upstage: z.record(z.string(), z.any()).optional(),
    venice: z.record(z.string(), z.any()).optional(),
    wafer: z.record(z.string(), z.any()).optional(),
    wandb: z.record(z.string(), z.any()).optional(),
    wandbLegacy: z.record(z.string(), z.any()).optional(),
    xai: z.record(z.string(), z.any()).optional(),
    xiaomi: z.record(z.string(), z.any()).optional(),
    zAi: z.record(z.string(), z.any()).optional(),
}).transform((v) => {
    return remap$(v, {
        oneai: "01ai",
        aionLabs: "aion-labs",
        amazonBedrock: "amazon-bedrock",
        amazonNova: "amazon-nova",
        arceeAi: "arcee-ai",
        atlasCloud: "atlas-cloud",
        blackForestLabs: "black-forest-labs",
        fakeProvider: "fake-provider",
        fishAudio: "fish-audio",
        googleAiStudio: "google-ai-studio",
        googleVertex: "google-vertex",
        hyperbolicQuantized: "hyperbolic-quantized",
        inferactVllm: "inferact-vllm",
        inferenceNet: "inference-net",
        ioNet: "io-net",
        lynnPrivate: "lynn-private",
        mancerOld: "mancer-old",
        nexAgi: "nex-agi",
        openInference: "open-inference",
        sailResearch: "sail-research",
        sakanaAi: "sakana-ai",
        sambanovaCloaked: "sambanova-cloaked",
        sfCompute: "sf-compute",
        togetherLite: "together-lite",
        wandbLegacy: "wandb-legacy",
        zAi: "z-ai",
    });
});
export function videoGenerationRequestOptionsToJSON(videoGenerationRequestOptions) {
    return JSON.stringify(VideoGenerationRequestOptions$outboundSchema.parse(videoGenerationRequestOptions));
}
/** @internal */
export const VideoGenerationRequestProvider$outboundSchema = z.object({
    options: z.lazy(() => VideoGenerationRequestOptions$outboundSchema)
        .optional(),
});
export function videoGenerationRequestProviderToJSON(videoGenerationRequestProvider) {
    return JSON.stringify(VideoGenerationRequestProvider$outboundSchema.parse(videoGenerationRequestProvider));
}
/** @internal */
export const VideoGenerationRequestResolution$outboundSchema = openEnums.outboundSchema(VideoGenerationRequestResolution);
/** @internal */
export const VideoGenerationRequest$outboundSchema = z.object({
    aspectRatio: VideoGenerationRequestAspectRatio$outboundSchema.optional(),
    callbackUrl: z.string().optional(),
    duration: z.int().optional(),
    frameImages: z.array(FrameImage$outboundSchema).optional(),
    generateAudio: z.boolean().optional(),
    inputReferences: z.array(InputReference$outboundSchema).optional(),
    model: z.string(),
    prompt: z.string().optional(),
    provider: z.lazy(() => VideoGenerationRequestProvider$outboundSchema)
        .optional(),
    resolution: VideoGenerationRequestResolution$outboundSchema.optional(),
    seed: z.int().optional(),
    size: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        aspectRatio: "aspect_ratio",
        callbackUrl: "callback_url",
        frameImages: "frame_images",
        generateAudio: "generate_audio",
        inputReferences: "input_references",
    });
});
export function videoGenerationRequestToJSON(videoGenerationRequest) {
    return JSON.stringify(VideoGenerationRequest$outboundSchema.parse(videoGenerationRequest));
}
