// @openrouter/sdk/esm/models/websearchplugin.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: fa32581277e2
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { WebSearchEngine$outboundSchema, } from "./websearchengine.js";
export const WebSearchPluginType = {
    Approximate: "approximate",
};
/** @internal */
export const WebSearchPluginType$outboundSchema = z.enum(WebSearchPluginType);
/** @internal */
export const UserLocation$outboundSchema = z.object({
    city: z.nullable(z.string()).optional(),
    country: z.nullable(z.string()).optional(),
    region: z.nullable(z.string()).optional(),
    timezone: z.nullable(z.string()).optional(),
    type: WebSearchPluginType$outboundSchema,
});
export function userLocationToJSON(userLocation) {
    return JSON.stringify(UserLocation$outboundSchema.parse(userLocation));
}
/** @internal */
export const WebSearchPlugin$outboundSchema = z.object({
    enabled: z.boolean().optional(),
    engine: WebSearchEngine$outboundSchema.optional(),
    excludeDomains: z.array(z.string()).optional(),
    id: z.literal("web"),
    includeDomains: z.array(z.string()).optional(),
    maxResults: z.int().optional(),
    maxUses: z.int().optional(),
    searchPrompt: z.string().optional(),
    userLocation: z.nullable(z.lazy(() => UserLocation$outboundSchema))
        .optional(),
}).transform((v) => {
    return remap$(v, {
        excludeDomains: "exclude_domains",
        includeDomains: "include_domains",
        maxResults: "max_results",
        maxUses: "max_uses",
        searchPrompt: "search_prompt",
        userLocation: "user_location",
    });
});
export function webSearchPluginToJSON(webSearchPlugin) {
    return JSON.stringify(WebSearchPlugin$outboundSchema.parse(webSearchPlugin));
}
