// @openrouter/sdk/esm/models/websearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 478da2f8e455
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { SearchContextSizeEnum$inboundSchema, SearchContextSizeEnum$outboundSchema, } from "./searchcontextsizeenum.js";
import { WebSearchDomainFilter$inboundSchema, WebSearchDomainFilter$outboundSchema, } from "./websearchdomainfilter.js";
import { WebSearchEngineEnum$inboundSchema, WebSearchEngineEnum$outboundSchema, } from "./websearchengineenum.js";
import { WebSearchUserLocation$inboundSchema, WebSearchUserLocation$outboundSchema, } from "./websearchuserlocation.js";
export const WebSearchServerToolType = {
    WebSearch20250826: "web_search_2025_08_26",
};
/** @internal */
export const WebSearchServerToolType$inboundSchema = z.enum(WebSearchServerToolType);
/** @internal */
export const WebSearchServerToolType$outboundSchema = WebSearchServerToolType$inboundSchema;
/** @internal */
export const WebSearchServerTool$inboundSchema = z.object({
    engine: WebSearchEngineEnum$inboundSchema.optional(),
    filters: z.nullable(WebSearchDomainFilter$inboundSchema).optional(),
    max_results: z.int().optional(),
    max_uses: z.int().optional(),
    search_context_size: SearchContextSizeEnum$inboundSchema.optional(),
    type: WebSearchServerToolType$inboundSchema,
    user_location: z.nullable(WebSearchUserLocation$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "max_results": "maxResults",
        "max_uses": "maxUses",
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
    });
});
/** @internal */
export const WebSearchServerTool$outboundSchema = z.object({
    engine: WebSearchEngineEnum$outboundSchema.optional(),
    filters: z.nullable(WebSearchDomainFilter$outboundSchema).optional(),
    maxResults: z.int().optional(),
    maxUses: z.int().optional(),
    searchContextSize: SearchContextSizeEnum$outboundSchema.optional(),
    type: WebSearchServerToolType$outboundSchema,
    userLocation: z.nullable(WebSearchUserLocation$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxUses: "max_uses",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
    });
});
export function webSearchServerToolToJSON(webSearchServerTool) {
    return JSON.stringify(WebSearchServerTool$outboundSchema.parse(webSearchServerTool));
}
export function webSearchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebSearchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebSearchServerTool' from JSON`);
}
