// @openrouter/sdk/esm/models/workspacemember.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 42cb4568aa7e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
/**
 * Role of the member in the workspace
 */
export const WorkspaceMemberRole = {
    Admin: "admin",
    Member: "member",
};
/** @internal */
export const WorkspaceMemberRole$inboundSchema = openEnums.inboundSchema(WorkspaceMemberRole);
/** @internal */
export const WorkspaceMember$inboundSchema = z.object({
    created_at: z.string(),
    id: z.string(),
    role: WorkspaceMemberRole$inboundSchema,
    user_id: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "user_id": "userId",
        "workspace_id": "workspaceId",
    });
});
export function workspaceMemberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WorkspaceMember$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WorkspaceMember' from JSON`);
}
