// @openrouter/sdk/esm/models/workspace.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f3a6e662c47b
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const Workspace$inboundSchema = z.object({
    created_at: z.string(),
    created_by: z.nullable(z.string()),
    default_guardrail_id: z.string(),
    default_image_model: z.nullable(z.string()),
    default_provider_sort: z.nullable(z.string()),
    default_text_model: z.nullable(z.string()),
    description: z.nullable(z.string()),
    id: z.string(),
    io_logging_api_key_ids: z.nullable(z.array(z.int())),
    io_logging_sampling_rate: z.number(),
    is_data_discount_logging_enabled: z.boolean(),
    is_observability_broadcast_enabled: z.boolean(),
    is_observability_io_logging_enabled: z.boolean(),
    name: z.string(),
    slug: z.string(),
    updated_at: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "created_by": "createdBy",
        "default_guardrail_id": "defaultGuardrailId",
        "default_image_model": "defaultImageModel",
        "default_provider_sort": "defaultProviderSort",
        "default_text_model": "defaultTextModel",
        "io_logging_api_key_ids": "ioLoggingApiKeyIds",
        "io_logging_sampling_rate": "ioLoggingSamplingRate",
        "is_data_discount_logging_enabled": "isDataDiscountLoggingEnabled",
        "is_observability_broadcast_enabled": "isObservabilityBroadcastEnabled",
        "is_observability_io_logging_enabled": "isObservabilityIoLoggingEnabled",
        "updated_at": "updatedAt",
    });
});
export function workspaceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Workspace$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Workspace' from JSON`);
}
