// @openrouter/sdk/esm/models/workspacebudget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ede50342c11d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
/**
 * Interval at which spend resets. Null means a lifetime (one-time) budget.
 */
export const ResetInterval = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/** @internal */
export const ResetInterval$inboundSchema = openEnums.inboundSchema(ResetInterval);
/** @internal */
export const WorkspaceBudget$inboundSchema = z.object({
    created_at: z.string(),
    id: z.string(),
    limit_usd: z.number(),
    reset_interval: z.nullable(ResetInterval$inboundSchema),
    updated_at: z.string(),
    workspace_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "limit_usd": "limitUsd",
        "reset_interval": "resetInterval",
        "updated_at": "updatedAt",
        "workspace_id": "workspaceId",
    });
});
export function workspaceBudgetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WorkspaceBudget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WorkspaceBudget' from JSON`);
}
