// @openrouter/sdk/esm/models/operations/createimages.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0a478467f700
 */
import * as z from "zod/v4";
import { EventStream } from "../../lib/event-streams.js";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const CreateImagesRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    imageGenerationRequest: models.ImageGenerationRequest$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        imageGenerationRequest: "ImageGenerationRequest",
    });
});
export function createImagesRequestToJSON(createImagesRequest) {
    return JSON.stringify(CreateImagesRequest$outboundSchema.parse(createImagesRequest));
}
/** @internal */
export const CreateImagesResponse$inboundSchema = z.union([
    models.ImageGenerationResponse$inboundSchema,
    z.custom(x => x instanceof ReadableStream)
        .transform(stream => {
        return new EventStream(stream, rawEvent => {
            if (rawEvent.data === "[DONE]")
                return { done: true, value: undefined };
            return {
                done: false,
                value: models.ImageStreamingResponse$inboundSchema.parse(rawEvent)
                    ?.data,
            };
        });
    }),
]);
export function createImagesResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateImagesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateImagesResponse' from JSON`);
}
