// @openrouter/sdk/esm/models/operations/createaudiotranscriptionsmultipart.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 29a4e7468387
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { blobLikeSchema } from "../../types/blobs.js";
import * as openEnums from "../../types/enums.js";
/**
 * The response format. "json" (default) returns { text, usage }; "verbose_json" additionally returns task, language, duration, and segment-level timestamps (OpenAI-compatible providers only).
 */
export const ResponseFormat = {
    Json: "json",
    VerboseJson: "verbose_json",
};
export const TimestampGranularities = {
    Word: "word",
    Segment: "segment",
};
/** @internal */
export const CreateAudioTranscriptionsMultipartFile$outboundSchema = z.object({
    fileName: z.string(),
    content: z.union([
        z.custom(x => x instanceof ReadableStream),
        z.custom(x => x instanceof Blob),
        z.custom(x => x instanceof ArrayBuffer),
        z.custom(x => x instanceof Uint8Array),
    ]),
});
export function createAudioTranscriptionsMultipartFileToJSON(createAudioTranscriptionsMultipartFile) {
    return JSON.stringify(CreateAudioTranscriptionsMultipartFile$outboundSchema.parse(createAudioTranscriptionsMultipartFile));
}
/** @internal */
export const ResponseFormat$outboundSchema = openEnums.outboundSchema(ResponseFormat);
/** @internal */
export const TimestampGranularities$outboundSchema = openEnums.outboundSchema(TimestampGranularities);
/** @internal */
export const CreateAudioTranscriptionsMultipartRequestBody$outboundSchema = z.object({
    file: z.lazy(() => CreateAudioTranscriptionsMultipartFile$outboundSchema)
        .or(blobLikeSchema),
    language: z.string().optional(),
    model: z.string(),
    responseFormat: ResponseFormat$outboundSchema.optional(),
    temperature: z.number().optional(),
    timestampGranularities: z.array(TimestampGranularities$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        responseFormat: "response_format",
        timestampGranularities: "timestamp_granularities[]",
    });
});
export function createAudioTranscriptionsMultipartRequestBodyToJSON(createAudioTranscriptionsMultipartRequestBody) {
    return JSON.stringify(CreateAudioTranscriptionsMultipartRequestBody$outboundSchema.parse(createAudioTranscriptionsMultipartRequestBody));
}
/** @internal */
export const CreateAudioTranscriptionsMultipartRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    requestBody: z.lazy(() => CreateAudioTranscriptionsMultipartRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        requestBody: "RequestBody",
    });
});
export function createAudioTranscriptionsMultipartRequestToJSON(createAudioTranscriptionsMultipartRequest) {
    return JSON.stringify(CreateAudioTranscriptionsMultipartRequest$outboundSchema.parse(createAudioTranscriptionsMultipartRequest));
}
