// @openrouter/sdk/esm/models/operations/getapprankings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: edd3364e84c1
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
import * as models from "../index.js";
/**
 * Marketplace category group to filter by (e.g. `coding`). Only apps tagged with a subcategory inside this group are returned. Mutually combinable with `subcategory` — when both are supplied the `subcategory` must belong to the `category` group.
 */
export const GetAppRankingsCategory = {
    Coding: "coding",
    Creative: "creative",
    Productivity: "productivity",
    Entertainment: "entertainment",
};
/**
 * Marketplace subcategory to filter by (e.g. `cli-agent`). Takes precedence over `category` for the actual filter; when `category` is also supplied the pair must be consistent.
 */
export const Subcategory = {
    CliAgent: "cli-agent",
    IdeExtension: "ide-extension",
    CloudAgent: "cloud-agent",
    ProgrammingApp: "programming-app",
    NativeAppBuilder: "native-app-builder",
    CreativeWriting: "creative-writing",
    VideoGen: "video-gen",
    ImageGen: "image-gen",
    AudioGen: "audio-gen",
    Roleplay: "roleplay",
    Game: "game",
    WritingAssistant: "writing-assistant",
    GeneralChat: "general-chat",
    PersonalAgent: "personal-agent",
    Legal: "legal",
};
/**
 * `popular` ranks apps by total token volume inside the date window. `trending` ranks apps by absolute excess token growth: window volume minus the average volume of the three equal-length periods immediately preceding the window. Apps with no excess growth are omitted from `trending` results.
 */
export const GetAppRankingsSort = {
    Popular: "popular",
    Trending: "trending",
};
/** @internal */
export const GetAppRankingsCategory$outboundSchema = openEnums.outboundSchema(GetAppRankingsCategory);
/** @internal */
export const Subcategory$outboundSchema = openEnums.outboundSchema(Subcategory);
/** @internal */
export const GetAppRankingsSort$outboundSchema = openEnums.outboundSchema(GetAppRankingsSort);
/** @internal */
export const GetAppRankingsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    category: GetAppRankingsCategory$outboundSchema.optional(),
    subcategory: Subcategory$outboundSchema.optional(),
    sort: GetAppRankingsSort$outboundSchema.default("popular"),
    startDate: z.string().optional(),
    endDate: z.string().optional(),
    limit: z.int().default(50),
    offset: z.nullable(z.int().default(0)),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        startDate: "start_date",
        endDate: "end_date",
    });
});
export function getAppRankingsRequestToJSON(getAppRankingsRequest) {
    return JSON.stringify(GetAppRankingsRequest$outboundSchema.parse(getAppRankingsRequest));
}
/** @internal */
export const GetAppRankingsResponse$inboundSchema = z.object({
    Result: models.AppRankingsResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function getAppRankingsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAppRankingsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAppRankingsResponse' from JSON`);
}
