// @openrouter/sdk/esm/models/operations/getanalyticsmeta.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 55adc33e474c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * Granularity identifier
 */
export const GranularityName = {
    Minute: "minute",
    Hour: "hour",
    Day: "day",
    Week: "week",
    Month: "month",
};
/**
 * How this metric value should be formatted for display (e.g. percent → multiply by 100 and append %, currency → prefix with $)
 */
export const DisplayFormat = {
    Number: "number",
    Currency: "currency",
    Percent: "percent",
    Latency: "latency",
    Throughput: "throughput",
};
/**
 * Operator identifier used in filter definitions
 */
export const OperatorName = {
    Eq: "eq",
    Neq: "neq",
    In: "in",
    NotIn: "not_in",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
};
/**
 * Whether the operator expects a single value or an array
 */
export const ValueType = {
    Scalar: "scalar",
    Array: "array",
};
/** @internal */
export const GetAnalyticsMetaRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
    });
});
export function getAnalyticsMetaRequestToJSON(getAnalyticsMetaRequest) {
    return JSON.stringify(GetAnalyticsMetaRequest$outboundSchema.parse(getAnalyticsMetaRequest));
}
/** @internal */
export const Dimension$inboundSchema = z.object({
    display_label: z.string(),
    name: z.string(),
}).transform((v) => {
    return remap$(v, {
        "display_label": "displayLabel",
    });
});
export function dimensionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Dimension$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Dimension' from JSON`);
}
/** @internal */
export const GranularityName$inboundSchema = openEnums.inboundSchema(GranularityName);
/** @internal */
export const Granularity$inboundSchema = z
    .object({
    display_label: z.string(),
    name: GranularityName$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "display_label": "displayLabel",
    });
});
export function granularityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Granularity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Granularity' from JSON`);
}
/** @internal */
export const DisplayFormat$inboundSchema = openEnums.inboundSchema(DisplayFormat);
/** @internal */
export const Metric$inboundSchema = z.object({
    display_format: DisplayFormat$inboundSchema,
    display_label: z.string(),
    is_rate: z.boolean(),
    name: z.string(),
}).transform((v) => {
    return remap$(v, {
        "display_format": "displayFormat",
        "display_label": "displayLabel",
        "is_rate": "isRate",
    });
});
export function metricFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Metric$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Metric' from JSON`);
}
/** @internal */
export const OperatorName$inboundSchema = openEnums.inboundSchema(OperatorName);
/** @internal */
export const ValueType$inboundSchema = openEnums
    .inboundSchema(ValueType);
/** @internal */
export const Operator$inboundSchema = z.object({
    name: OperatorName$inboundSchema,
    value_type: ValueType$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "value_type": "valueType",
    });
});
export function operatorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Operator$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Operator' from JSON`);
}
/** @internal */
export const GetAnalyticsMetaData$inboundSchema = z.object({
    dimensions: z.array(z.lazy(() => Dimension$inboundSchema)),
    granularities: z.array(z.lazy(() => Granularity$inboundSchema)),
    metrics: z.array(z.lazy(() => Metric$inboundSchema)),
    operators: z.array(z.lazy(() => Operator$inboundSchema)),
});
export function getAnalyticsMetaDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAnalyticsMetaData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAnalyticsMetaData' from JSON`);
}
/** @internal */
export const GetAnalyticsMetaResponse$inboundSchema = z.object({
    data: z.lazy(() => GetAnalyticsMetaData$inboundSchema),
});
export function getAnalyticsMetaResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAnalyticsMetaResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAnalyticsMetaResponse' from JSON`);
}
