// @openrouter/sdk/esm/models/operations/getbenchmarks.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 52eb1eb063ae
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as openEnums from "../../types/enums.js";
/**
 * Benchmark source to query. Determines the shape of the returned items. When omitted, returns results from all sources.
 */
export const Source = {
    ArtificialAnalysis: "artificial-analysis",
    DesignArena: "design-arena",
};
/**
 * Filter results by task type. For Artificial Analysis, maps to the corresponding index. For Design Arena, maps to the matching category.
 */
export const TaskType = {
    Coding: "coding",
    Intelligence: "intelligence",
    Agentic: "agentic",
};
/**
 * Design Arena only: arena to query. Defaults to `models` when source is `design-arena`.
 */
export const Arena = {
    Models: "models",
    Builders: "builders",
    Agents: "agents",
};
/** @internal */
export const Source$outboundSchema = openEnums
    .outboundSchema(Source);
/** @internal */
export const TaskType$outboundSchema = openEnums
    .outboundSchema(TaskType);
/** @internal */
export const Arena$outboundSchema = openEnums
    .outboundSchema(Arena);
/** @internal */
export const GetBenchmarksRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    source: Source$outboundSchema.optional(),
    taskType: TaskType$outboundSchema.optional(),
    arena: Arena$outboundSchema.optional(),
    category: z.string().optional(),
    maxResults: z.int().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        taskType: "task_type",
        maxResults: "max_results",
    });
});
export function getBenchmarksRequestToJSON(getBenchmarksRequest) {
    return JSON.stringify(GetBenchmarksRequest$outboundSchema.parse(getBenchmarksRequest));
}
