// @openrouter/sdk/esm/models/operations/getmodels.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f1343fbaaed5
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
import * as models from "../index.js";
/**
 * Filter models by use case category
 */
export const GetModelsCategory = {
    Programming: "programming",
    Roleplay: "roleplay",
    Marketing: "marketing",
    MarketingSeo: "marketing/seo",
    Technology: "technology",
    Science: "science",
    Translation: "translation",
    Legal: "legal",
    Finance: "finance",
    Health: "health",
    Trivia: "trivia",
    Academia: "academia",
};
/**
 * Sort the returned models server-side. Prefer this over fetching the full list and sorting client-side. Options: pricing-low-to-high, pricing-high-to-low (average prompt/completion price), context-high-to-low (context length), throughput-high-to-low, latency-low-to-high (recent median performance), most-popular, top-weekly (tokens processed in the last week), newest (creation date), intelligence-high-to-low, coding-high-to-low, agentic-high-to-low (Artificial Analysis indices), design-arena-elo-high-to-low (best Design Arena ELO across arenas). Models without a score for the chosen benchmark are placed last. When omitted, the existing default ordering is preserved.
 */
export const GetModelsSort = {
    MostPopular: "most-popular",
    Newest: "newest",
    TopWeekly: "top-weekly",
    PricingLowToHigh: "pricing-low-to-high",
    PricingHighToLow: "pricing-high-to-low",
    ContextHighToLow: "context-high-to-low",
    ThroughputHighToLow: "throughput-high-to-low",
    LatencyLowToHigh: "latency-low-to-high",
    IntelligenceHighToLow: "intelligence-high-to-low",
    CodingHighToLow: "coding-high-to-low",
    AgenticHighToLow: "agentic-high-to-low",
    DesignArenaEloHighToLow: "design-arena-elo-high-to-low",
};
/**
 * Filter by distillation capability. "true" returns only distillable models, "false" excludes them.
 */
export const Distillable = {
    True: "true",
    False: "false",
};
/**
 * When set to "true", return only models with zero data retention endpoints.
 */
export const Zdr = {
    True: "true",
};
/**
 * Filter to models with endpoints in the given data region. Currently only "eu" is supported.
 */
export const Region = {
    Eu: "eu",
};
/** @internal */
export const GetModelsCategory$outboundSchema = openEnums.outboundSchema(GetModelsCategory);
/** @internal */
export const GetModelsSort$outboundSchema = openEnums.outboundSchema(GetModelsSort);
/** @internal */
export const Distillable$outboundSchema = openEnums.outboundSchema(Distillable);
/** @internal */
export const Zdr$outboundSchema = z.enum(Zdr);
/** @internal */
export const Region$outboundSchema = z.enum(Region);
/** @internal */
export const GetModelsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    offset: z.nullable(z.int().default(0)),
    limit: z.int().default(500),
    category: GetModelsCategory$outboundSchema.optional(),
    supportedParameters: z.string().optional(),
    outputModalities: z.string().optional(),
    sort: GetModelsSort$outboundSchema.optional(),
    q: z.string().optional(),
    inputModalities: z.string().optional(),
    context: z.int().optional(),
    minPrice: z.nullable(z.number()).optional(),
    maxPrice: z.nullable(z.number()).optional(),
    arch: z.string().optional(),
    modelAuthors: z.string().optional(),
    providers: z.string().optional(),
    distillable: Distillable$outboundSchema.optional(),
    zdr: Zdr$outboundSchema.optional(),
    region: Region$outboundSchema.optional(),
    minOutputPrice: z.nullable(z.number()).optional(),
    maxOutputPrice: z.nullable(z.number()).optional(),
    minAgeDays: z.nullable(z.int()).optional(),
    maxAgeDays: z.nullable(z.int()).optional(),
    minIntelligenceIndex: z.nullable(z.number()).optional(),
    maxIntelligenceIndex: z.nullable(z.number()).optional(),
    minCodingIndex: z.nullable(z.number()).optional(),
    maxCodingIndex: z.nullable(z.number()).optional(),
    minAgenticIndex: z.nullable(z.number()).optional(),
    maxAgenticIndex: z.nullable(z.number()).optional(),
    minToolSuccessRate: z.nullable(z.number()).optional(),
    maxToolSuccessRate: z.nullable(z.number()).optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        supportedParameters: "supported_parameters",
        outputModalities: "output_modalities",
        inputModalities: "input_modalities",
        minPrice: "min_price",
        maxPrice: "max_price",
        modelAuthors: "model_authors",
        minOutputPrice: "min_output_price",
        maxOutputPrice: "max_output_price",
        minAgeDays: "min_age_days",
        maxAgeDays: "max_age_days",
        minIntelligenceIndex: "min_intelligence_index",
        maxIntelligenceIndex: "max_intelligence_index",
        minCodingIndex: "min_coding_index",
        maxCodingIndex: "max_coding_index",
        minAgenticIndex: "min_agentic_index",
        maxAgenticIndex: "max_agentic_index",
        minToolSuccessRate: "min_tool_success_rate",
        maxToolSuccessRate: "max_tool_success_rate",
    });
});
export function getModelsRequestToJSON(getModelsRequest) {
    return JSON.stringify(GetModelsRequest$outboundSchema.parse(getModelsRequest));
}
/** @internal */
export const GetModelsResponse$inboundSchema = z.object({
    Result: models.ModelsListResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function getModelsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetModelsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetModelsResponse' from JSON`);
}
