// @openrouter/sdk/esm/models/operations/getrankingsdaily.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b165b39aba3b
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as openEnums from "../../types/enums.js";
/**
 * Time grain of each row. `day` (default) returns the per-UTC-day series; `week` buckets by ISO week start; `month` buckets by month start. With `category` or `language_type` only `week` (default) and `month` are available — `day` is rejected with a 400 because those datasets are aggregated weekly. For those sampled datasets `period=month` buckets each week by its week-start month, so totals are approximate at month boundaries.
 */
export const Period = {
    Day: "day",
    Week: "week",
    Month: "month",
};
/**
 * Restrict to models for a modality surface: `text` / `image_output` match output modality, `image` / `audio` match input modality, and `tool_calling` keeps only rows that recorded at least one tool call. Exact dataset — cannot be combined with `category` or `language_type`.
 */
export const Modality = {
    Text: "text",
    Image: "image",
    ImageOutput: "image_output",
    Audio: "audio",
    ToolCalling: "tool_calling",
};
/**
 * Restrict to requests whose context length falls in this bucket (`1K`, `10K`, `100K`, `1M`, or `10M`). Exact dataset — cannot be combined with `category` or `language_type`.
 */
export const ContextBucket = {
    OneK: "1K",
    TenK: "10K",
    OneHundredK: "100K",
    OneM: "1M",
    TenM: "10M",
};
/**
 * Restrict to a use-case category (e.g. `programming`, `roleplay`). Sourced from a sampled, upsampled dataset, so `total_tokens` is an estimate and is aggregated weekly (the trailing weekly bucket may include traffic past `end_date`). Cannot be combined with `modality`, `context_bucket`, or `language_type`.
 */
export const GetRankingsDailyCategory = {
    Programming: "programming",
    Roleplay: "roleplay",
    Marketing: "marketing",
    MarketingSeo: "marketing/seo",
    Technology: "technology",
    Science: "science",
    Translation: "translation",
    Legal: "legal",
    Finance: "finance",
    Health: "health",
    Trivia: "trivia",
    Academia: "academia",
};
/**
 * Restrict to natural-language or programming-language tagged activity. Sourced from a sampled, upsampled dataset, so `total_tokens` is an estimate and is aggregated weekly (the trailing weekly bucket may include traffic past `end_date`). Cannot be combined with `modality`, `context_bucket`, or `category`.
 */
export const LanguageType = {
    Natural: "natural",
    Programming: "programming",
};
/** @internal */
export const Period$outboundSchema = openEnums
    .outboundSchema(Period);
/** @internal */
export const Modality$outboundSchema = openEnums
    .outboundSchema(Modality);
/** @internal */
export const ContextBucket$outboundSchema = openEnums.outboundSchema(ContextBucket);
/** @internal */
export const GetRankingsDailyCategory$outboundSchema = openEnums.outboundSchema(GetRankingsDailyCategory);
/** @internal */
export const LanguageType$outboundSchema = openEnums.outboundSchema(LanguageType);
/** @internal */
export const GetRankingsDailyRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    startDate: z.string().optional(),
    endDate: z.string().optional(),
    period: Period$outboundSchema.optional(),
    modality: Modality$outboundSchema.optional(),
    contextBucket: ContextBucket$outboundSchema.optional(),
    category: GetRankingsDailyCategory$outboundSchema.optional(),
    languageType: LanguageType$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        startDate: "start_date",
        endDate: "end_date",
        contextBucket: "context_bucket",
        languageType: "language_type",
    });
});
export function getRankingsDailyRequestToJSON(getRankingsDailyRequest) {
    return JSON.stringify(GetRankingsDailyRequest$outboundSchema.parse(getRankingsDailyRequest));
}
