// @openrouter/sdk/esm/models/operations/listobservabilitydestinations.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 611f391b0806
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const ListObservabilityDestinationsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    offset: z.nullable(z.int().default(0)),
    limit: z.int().default(50),
    workspaceId: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        workspaceId: "workspace_id",
    });
});
export function listObservabilityDestinationsRequestToJSON(listObservabilityDestinationsRequest) {
    return JSON.stringify(ListObservabilityDestinationsRequest$outboundSchema.parse(listObservabilityDestinationsRequest));
}
/** @internal */
export const ListObservabilityDestinationsResponse$inboundSchema = z.object({
    Result: models.ListObservabilityDestinationsResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listObservabilityDestinationsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListObservabilityDestinationsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListObservabilityDestinationsResponse' from JSON`);
}
