// @openrouter/sdk/esm/models/operations/listembeddingsmodels.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 52a7fdf31054
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const ListEmbeddingsModelsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    offset: z.nullable(z.int().default(0)),
    limit: z.int().default(500),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
    });
});
export function listEmbeddingsModelsRequestToJSON(listEmbeddingsModelsRequest) {
    return JSON.stringify(ListEmbeddingsModelsRequest$outboundSchema.parse(listEmbeddingsModelsRequest));
}
/** @internal */
export const ListEmbeddingsModelsResponse$inboundSchema = z.object({
    Result: models.ModelsListResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listEmbeddingsModelsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListEmbeddingsModelsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListEmbeddingsModelsResponse' from JSON`);
}
