// @openrouter/sdk/esm/models/operations/listworkspaces.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 04f9b5b78c0f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const ListWorkspacesRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    offset: z.nullable(z.int().default(0)),
    limit: z.int().default(50),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
    });
});
export function listWorkspacesRequestToJSON(listWorkspacesRequest) {
    return JSON.stringify(ListWorkspacesRequest$outboundSchema.parse(listWorkspacesRequest));
}
/** @internal */
export const ListWorkspacesResponse$inboundSchema = z.object({
    Result: models.ListWorkspacesResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listWorkspacesResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListWorkspacesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListWorkspacesResponse' from JSON`);
}
