// @openrouter/sdk/esm/models/operations/listmodelsuser.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0846aac6be02
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const ListModelsUserSecurity$outboundSchema = z.object({
    bearer: z.string(),
});
export function listModelsUserSecurityToJSON(listModelsUserSecurity) {
    return JSON.stringify(ListModelsUserSecurity$outboundSchema.parse(listModelsUserSecurity));
}
/** @internal */
export const ListModelsUserRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    offset: z.nullable(z.int().default(0)),
    limit: z.int().default(500),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
    });
});
export function listModelsUserRequestToJSON(listModelsUserRequest) {
    return JSON.stringify(ListModelsUserRequest$outboundSchema.parse(listModelsUserRequest));
}
/** @internal */
export const ListModelsUserResponse$inboundSchema = z.object({
    Result: models.ModelsListResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listModelsUserResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListModelsUserResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListModelsUserResponse' from JSON`);
}
