// @openrouter/sdk/esm/models/operations/listpresets.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 85970d436882
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const ListPresetsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    offset: z.nullable(z.int().default(0)),
    limit: z.int().default(50),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
    });
});
export function listPresetsRequestToJSON(listPresetsRequest) {
    return JSON.stringify(ListPresetsRequest$outboundSchema.parse(listPresetsRequest));
}
/** @internal */
export const ListPresetsResponse$inboundSchema = z.object({
    Result: models.ListPresetsResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listPresetsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListPresetsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListPresetsResponse' from JSON`);
}
