// @openrouter/sdk/esm/models/operations/listpresetversions.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7971ca6e4018
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const ListPresetVersionsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    slug: z.string(),
    offset: z.nullable(z.int().default(0)),
    limit: z.int().default(50),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
    });
});
export function listPresetVersionsRequestToJSON(listPresetVersionsRequest) {
    return JSON.stringify(ListPresetVersionsRequest$outboundSchema.parse(listPresetVersionsRequest));
}
/** @internal */
export const ListPresetVersionsResponse$inboundSchema = z.object({
    Result: models.ListPresetVersionsResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listPresetVersionsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListPresetVersionsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListPresetVersionsResponse' from JSON`);
}
