// @openrouter/sdk/esm/models/operations/listfiles.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9220b094caf4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const ListFilesRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    limit: z.int().optional(),
    cursor: z.string().optional(),
    workspaceId: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        workspaceId: "workspace_id",
    });
});
export function listFilesRequestToJSON(listFilesRequest) {
    return JSON.stringify(ListFilesRequest$outboundSchema.parse(listFilesRequest));
}
/** @internal */
export const ListFilesResponse$inboundSchema = z.object({
    Result: models.FileListResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listFilesResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListFilesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListFilesResponse' from JSON`);
}
