// @openrouter/sdk/esm/models/operations/listworkspacemembers.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 009ed1f62eb4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const ListWorkspaceMembersRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    id: z.string(),
    offset: z.nullable(z.int().default(0)),
    limit: z.int().default(50),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
    });
});
export function listWorkspaceMembersRequestToJSON(listWorkspaceMembersRequest) {
    return JSON.stringify(ListWorkspaceMembersRequest$outboundSchema.parse(listWorkspaceMembersRequest));
}
/** @internal */
export const ListWorkspaceMembersResponse$inboundSchema = z.object({
    Result: models.ListWorkspaceMembersResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function listWorkspaceMembersResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListWorkspaceMembersResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListWorkspaceMembersResponse' from JSON`);
}
