// @openrouter/sdk/esm/models/operations/queryanalytics.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2b9c6044e362
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
export const Direction = {
    Asc: "asc",
    Desc: "desc",
};
/** @internal */
export const ClassifierDimensions$outboundSchema = z.object({
    classifierId: z.string(),
    dimensionNames: z.array(z.string()).optional(),
    includeNulls: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        classifierId: "classifier_id",
        dimensionNames: "dimension_names",
        includeNulls: "include_nulls",
    });
});
export function classifierDimensionsToJSON(classifierDimensions) {
    return JSON.stringify(ClassifierDimensions$outboundSchema.parse(classifierDimensions));
}
/** @internal */
export const ValueClassifierFilters$outboundSchema = z.union([z.string(), z.number()]);
export function valueClassifierFiltersToJSON(valueClassifierFilters) {
    return JSON.stringify(ValueClassifierFilters$outboundSchema.parse(valueClassifierFilters));
}
/** @internal */
export const ClassifierFiltersValue$outboundSchema = z.union([
    z.string(),
    z.number(),
    z.array(z.union([z.string(), z.number()])),
]);
export function classifierFiltersValueToJSON(classifierFiltersValue) {
    return JSON.stringify(ClassifierFiltersValue$outboundSchema.parse(classifierFiltersValue));
}
/** @internal */
export const ClassifierFiltersFilter$outboundSchema = z.object({
    field: z.string(),
    operator: z.string(),
    value: z.union([
        z.string(),
        z.number(),
        z.array(z.union([z.string(), z.number()])),
    ]),
});
export function classifierFiltersFilterToJSON(classifierFiltersFilter) {
    return JSON.stringify(ClassifierFiltersFilter$outboundSchema.parse(classifierFiltersFilter));
}
/** @internal */
export const ClassifierFilters$outboundSchema = z.object({
    classifierId: z.string(),
    filters: z.array(z.lazy(() => ClassifierFiltersFilter$outboundSchema)),
}).transform((v) => {
    return remap$(v, {
        classifierId: "classifier_id",
    });
});
export function classifierFiltersToJSON(classifierFilters) {
    return JSON.stringify(ClassifierFilters$outboundSchema.parse(classifierFilters));
}
/** @internal */
export const Value2$outboundSchema = z
    .union([z.string(), z.number()]);
export function value2ToJSON(value2) {
    return JSON.stringify(Value2$outboundSchema.parse(value2));
}
/** @internal */
export const Value1$outboundSchema = z
    .union([z.string(), z.number(), z.array(z.union([z.string(), z.number()]))]);
export function value1ToJSON(value1) {
    return JSON.stringify(Value1$outboundSchema.parse(value1));
}
/** @internal */
export const Filter$outboundSchema = z
    .object({
    field: z.string(),
    operator: z.string(),
    value: z.union([
        z.string(),
        z.number(),
        z.array(z.union([z.string(), z.number()])),
    ]),
});
export function filterToJSON(filter) {
    return JSON.stringify(Filter$outboundSchema.parse(filter));
}
/** @internal */
export const Direction$outboundSchema = openEnums
    .outboundSchema(Direction);
/** @internal */
export const OrderBy$outboundSchema = z
    .object({
    direction: Direction$outboundSchema,
    field: z.string(),
});
export function orderByToJSON(orderBy) {
    return JSON.stringify(OrderBy$outboundSchema.parse(orderBy));
}
/** @internal */
export const TimeRange$outboundSchema = z.object({
    end: z.date().transform(v => v.toISOString()),
    start: z.date().transform(v => v.toISOString()),
});
export function timeRangeToJSON(timeRange) {
    return JSON.stringify(TimeRange$outboundSchema.parse(timeRange));
}
/** @internal */
export const QueryAnalyticsRequestBody$outboundSchema = z.object({
    classifierDimensions: z.lazy(() => ClassifierDimensions$outboundSchema)
        .optional(),
    classifierFilters: z.lazy(() => ClassifierFilters$outboundSchema).optional(),
    dimensions: z.array(z.string()).optional(),
    filters: z.array(z.lazy(() => Filter$outboundSchema)).optional(),
    granularity: z.string().optional(),
    groupLimit: z.int().optional(),
    limit: z.int().optional(),
    metrics: z.array(z.string()),
    orderBy: z.lazy(() => OrderBy$outboundSchema).optional(),
    timeRange: z.lazy(() => TimeRange$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        classifierDimensions: "classifier_dimensions",
        classifierFilters: "classifier_filters",
        groupLimit: "group_limit",
        orderBy: "order_by",
        timeRange: "time_range",
    });
});
export function queryAnalyticsRequestBodyToJSON(queryAnalyticsRequestBody) {
    return JSON.stringify(QueryAnalyticsRequestBody$outboundSchema.parse(queryAnalyticsRequestBody));
}
/** @internal */
export const QueryAnalyticsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    requestBody: z.lazy(() => QueryAnalyticsRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        requestBody: "RequestBody",
    });
});
export function queryAnalyticsRequestToJSON(queryAnalyticsRequest) {
    return JSON.stringify(QueryAnalyticsRequest$outboundSchema.parse(queryAnalyticsRequest));
}
/** @internal */
export const QueryAnalyticsData1$inboundSchema = z.object({});
export function queryAnalyticsData1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => QueryAnalyticsData1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryAnalyticsData1' from JSON`);
}
/** @internal */
export const Metadata$inboundSchema = z.object({
    query_time_ms: z.number(),
    row_count: z.int(),
    truncated: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "query_time_ms": "queryTimeMs",
        "row_count": "rowCount",
    });
});
export function metadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Metadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Metadata' from JSON`);
}
/** @internal */
export const QueryAnalyticsData2$inboundSchema = z.object({
    cachedAt: z.number().optional(),
    data: z.array(z.lazy(() => QueryAnalyticsData1$inboundSchema)),
    metadata: z.lazy(() => Metadata$inboundSchema),
    warnings: z.array(z.string()).optional(),
});
export function queryAnalyticsData2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => QueryAnalyticsData2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryAnalyticsData2' from JSON`);
}
/** @internal */
export const QueryAnalyticsResponse$inboundSchema = z.object({
    data: z.lazy(() => QueryAnalyticsData2$inboundSchema),
});
export function queryAnalyticsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => QueryAnalyticsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryAnalyticsResponse' from JSON`);
}
