// @openrouter/sdk/esm/models/operations/sendchatcompletionrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 52a55a780d2d
 */
import * as z from "zod/v4";
import { EventStream } from "../../lib/event-streams.js";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const SendChatCompletionRequestRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    xOpenRouterMetadata: models.MetadataLevel$outboundSchema.optional(),
    chatRequest: models.ChatRequest$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xOpenRouterMetadata: "X-OpenRouter-Metadata",
        chatRequest: "ChatRequest",
    });
});
export function sendChatCompletionRequestRequestToJSON(sendChatCompletionRequestRequest) {
    return JSON.stringify(SendChatCompletionRequestRequest$outboundSchema.parse(sendChatCompletionRequestRequest));
}
/** @internal */
export const SendChatCompletionRequestResponse$inboundSchema = z.union([
    models.ChatResult$inboundSchema,
    z.custom(x => x instanceof ReadableStream)
        .transform(stream => {
        return new EventStream(stream, rawEvent => {
            if (rawEvent.data === "[DONE]")
                return { done: true, value: undefined };
            return {
                done: false,
                value: models.ChatStreamingResponse$inboundSchema.parse(rawEvent)
                    ?.data,
            };
        });
    }),
]);
export function sendChatCompletionRequestResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SendChatCompletionRequestResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SendChatCompletionRequestResponse' from JSON`);
}
