// @openrouter/sdk/esm/models/operations/uploadfile.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e2f1b0f34a4a
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { blobLikeSchema } from "../../types/blobs.js";
/** @internal */
export const UploadFileFile$outboundSchema = z.object({
    fileName: z.string(),
    content: z.union([
        z.custom(x => x instanceof ReadableStream),
        z.custom(x => x instanceof Blob),
        z.custom(x => x instanceof ArrayBuffer),
        z.custom(x => x instanceof Uint8Array),
    ]),
});
export function uploadFileFileToJSON(uploadFileFile) {
    return JSON.stringify(UploadFileFile$outboundSchema.parse(uploadFileFile));
}
/** @internal */
export const UploadFileRequestBody$outboundSchema = z.object({
    file: z.lazy(() => UploadFileFile$outboundSchema).or(blobLikeSchema),
});
export function uploadFileRequestBodyToJSON(uploadFileRequestBody) {
    return JSON.stringify(UploadFileRequestBody$outboundSchema.parse(uploadFileRequestBody));
}
/** @internal */
export const UploadFileRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    workspaceId: z.string().optional(),
    requestBody: z.lazy(() => UploadFileRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        workspaceId: "workspace_id",
        requestBody: "RequestBody",
    });
});
export function uploadFileRequestToJSON(uploadFileRequest) {
    return JSON.stringify(UploadFileRequest$outboundSchema.parse(uploadFileRequest));
}
