// @openrouter/sdk/esm/models/errors/unauthorizedresponseerror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 79ab0861cf08
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as models from "../index.js";
import { OpenRouterError } from "./openroutererror.js";
/**
 * Unauthorized - Authentication required or invalid credentials
 */
export class UnauthorizedResponseError extends OpenRouterError {
    constructor(err, httpMeta) {
        const message = err.error?.message
            || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        if (err.openrouterMetadata != null) {
            this.openrouterMetadata = err.openrouterMetadata;
        }
        if (err.userId != null)
            this.userId = err.userId;
        this.name = "UnauthorizedResponseError";
    }
}
/** @internal */
export const UnauthorizedResponseError$inboundSchema = z.object({
    error: models.UnauthorizedResponseErrorData$inboundSchema,
    openrouter_metadata: z.nullable(z.record(z.string(), z.any())).optional(),
    user_id: z.nullable(z.string()).optional(),
    request$: z.custom(x => x instanceof Request),
    response$: z.custom(x => x instanceof Response),
    body$: z.string(),
})
    .transform((v) => {
    const remapped = remap$(v, {
        "openrouter_metadata": "openrouterMetadata",
        "user_id": "userId",
    });
    return new UnauthorizedResponseError(remapped, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
