// @openrouter/sdk/esm/funcs/datasetsGetAppRankings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2ae09697d446
 */
import { encodeFormQuery, encodeSimple } from "../lib/encodings.js";
import { matchStatusCode } from "../lib/http.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as errors from "../models/errors/index.js";
import * as operations from "../models/operations/index.js";
import { APIPromise } from "../types/async.js";
import { createPageIterator, haltIterator, } from "../types/operations.js";
/**
 * Top apps by token usage
 *
 * @remarks
 * Returns the top public apps on OpenRouter ranked by token usage inside the requested
 * date window, matching the public apps marketplace on openrouter.ai/apps. Token totals
 * are `prompt_tokens + completion_tokens`; hidden and private apps are excluded and
 * traffic from related app aliases is merged into the canonical visible app.
 *
 * `sort=popular` (default) ranks by total token volume inside the window.
 * `sort=trending` ranks by absolute excess token growth: window volume minus the average
 * volume of the three equal-length periods immediately preceding the window. Apps with
 * no excess growth are omitted, so `trending` may return fewer than `limit` rows.
 *
 * Filter with `category` (marketplace category group, e.g. `coding`) or `subcategory`
 * (e.g. `cli-agent`). Ranks are re-numbered 1..N after filtering. Page with `offset` —
 * `rank` stays absolute, so the first row of `offset=50` is `rank: 51`.
 *
 * Authenticate with any valid OpenRouter API key (same key used for inference).
 * Rate-limited to 30 requests/minute per key and 500 requests/day per account.
 *
 * When republishing or quoting this dataset, OpenRouter must be cited as:
 * "Source: OpenRouter (openrouter.ai/apps), as of {as_of}."
 *
 * Token counts come from each upstream provider's own tokenizer, so a token attributed
 * to one app is not directly comparable to a token attributed to another app whose
 * traffic flows through a different provider.
 */
export function datasetsGetAppRankings(client, request, options) {
    return new APIPromise($do(client, request, options));
}
async function $do(client, request, options) {
    const parsed = safeParse(request, (value) => operations.GetAppRankingsRequest$outboundSchema.optional().parse(value), "Input validation failed");
    if (!parsed.ok) {
        return [haltIterator(parsed), { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = null;
    const path = pathToFunc("/datasets/app-rankings")();
    const query = encodeFormQuery({
        "category": payload?.category,
        "end_date": payload?.end_date,
        "limit": payload?.limit,
        "offset": payload?.offset,
        "sort": payload?.sort,
        "start_date": payload?.start_date,
        "subcategory": payload?.subcategory,
    });
    const headers = new Headers(compactMap({
        Accept: "application/json",
        "HTTP-Referer": encodeSimple("HTTP-Referer", payload?.["HTTP-Referer"] ?? client._options.httpReferer, { explode: false, charEncoding: "none" }),
        "X-OpenRouter-Categories": encodeSimple("X-OpenRouter-Categories", payload?.appCategories ?? client._options.appCategories, { explode: false, charEncoding: "none" }),
        "X-OpenRouter-Title": encodeSimple("X-OpenRouter-Title", payload?.appTitle ?? client._options.appTitle, { explode: false, charEncoding: "none" }),
    }));
    const secConfig = await extractSecurity(client._options.apiKey);
    const securityInput = secConfig == null ? {} : { apiKey: secConfig };
    const requestSecurity = resolveGlobalSecurity(securityInput);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "getAppRankings",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: client._options.apiKey,
        retryConfig: options?.retries
            || client._options.retryConfig
            || {
                strategy: "backoff",
                backoff: {
                    initialInterval: 500,
                    maxInterval: 60000,
                    exponent: 1.5,
                    maxElapsedTime: 3600000,
                },
                retryConnectionErrors: true,
            }
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["5XX"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "GET",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        query: query,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [haltIterator(requestRes), { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        isErrorStatusCode: (statusCode) => matchStatusCode({ status: statusCode }, ["4XX", "5XX"]),
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [haltIterator(doResult), { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result, raw] = await M.match(M.json(200, operations.GetAppRankingsResponse$inboundSchema, {
        key: "Result",
    }), M.jsonErr(400, errors.BadRequestResponseError$inboundSchema), M.jsonErr(401, errors.UnauthorizedResponseError$inboundSchema), M.jsonErr(429, errors.TooManyRequestsResponseError$inboundSchema), M.jsonErr(500, errors.InternalServerResponseError$inboundSchema), M.fail("4XX"), M.fail("5XX"))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [haltIterator(result), {
                status: "complete",
                request: req,
                response,
            }];
    }
    const nextFunc = (responseData) => {
        const offset = request?.offset ?? 0;
        if (!responseData) {
            return { next: () => null };
        }
        const results = responseData.data;
        if (!Array.isArray(results) || !results.length) {
            return { next: () => null };
        }
        const limit = request?.limit ?? 50;
        if (results.length < limit) {
            return { next: () => null };
        }
        const nextOffset = offset + results.length;
        const nextVal = () => datasetsGetAppRankings(client, {
            ...request,
            offset: nextOffset,
        }, options);
        return { next: nextVal, "~next": { offset: nextOffset } };
    };
    const page = { ...result, ...nextFunc(raw) };
    return [{ ...page, ...createPageIterator(page, (v) => !v.ok) }, {
            status: "complete",
            request: req,
            response,
        }];
}
