// @openrouter/sdk/esm/funcs/datasetsGetRankingsDaily.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f0177699bfaf
 */
import { encodeFormQuery, encodeSimple } from "../lib/encodings.js";
import { matchStatusCode } from "../lib/http.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as errors from "../models/errors/index.js";
import * as models from "../models/index.js";
import * as operations from "../models/operations/index.js";
import { APIPromise } from "../types/async.js";
/**
 * Daily token totals for top 50 models
 *
 * @remarks
 * Returns the top 50 public models per day by total token usage on OpenRouter, plus a
 * single aggregated `other` row per day that sums every model outside that top 50.
 * Token totals are `prompt_tokens + completion_tokens`, matching the public rankings
 * chart on openrouter.ai/rankings.
 *
 * Each row is a distinct `(date, model_permaslug)` pair. The `other` row uses the
 * reserved permaslug `other` and is always returned last within its date, so callers
 * can compute `top-50 traffic / total daily traffic` without a second request.
 *
 * Optional filters slice the dataset. `period` (`day`/`week`/`month`) sets the time
 * grain. `modality` and `context_bucket` narrow the exact dataset by output/input
 * modality (or tool-calling activity) and request context length. `category` and
 * `language_type` instead read a sampled, upsampled dataset whose `total_tokens` are
 * weekly-grain estimates — they cannot be combined with each other or with the exact
 * filters, and reject `period=day` with a 400.
 *
 * Authenticate with any valid OpenRouter API key (same key used for inference).
 * Rate-limited to 30 requests/minute per key and 500 requests/day per account.
 *
 * When republishing or quoting this dataset, OpenRouter must be cited as:
 * "Source: OpenRouter (openrouter.ai/rankings), as of {as_of}."
 *
 * Token counts come from each upstream provider's own tokenizer (Anthropic counts
 * are as reported by Anthropic, OpenAI counts are as reported by OpenAI, etc.), so
 * a token in one row is not directly comparable to a token in another row from a
 * different provider.
 */
export function datasetsGetRankingsDaily(client, request, options) {
    return new APIPromise($do(client, request, options));
}
async function $do(client, request, options) {
    const parsed = safeParse(request, (value) => operations.GetRankingsDailyRequest$outboundSchema.optional().parse(value), "Input validation failed");
    if (!parsed.ok) {
        return [parsed, { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = null;
    const path = pathToFunc("/datasets/rankings-daily")();
    const query = encodeFormQuery({
        "category": payload?.category,
        "context_bucket": payload?.context_bucket,
        "end_date": payload?.end_date,
        "language_type": payload?.language_type,
        "modality": payload?.modality,
        "period": payload?.period,
        "start_date": payload?.start_date,
    });
    const headers = new Headers(compactMap({
        Accept: "application/json",
        "HTTP-Referer": encodeSimple("HTTP-Referer", payload?.["HTTP-Referer"] ?? client._options.httpReferer, { explode: false, charEncoding: "none" }),
        "X-OpenRouter-Categories": encodeSimple("X-OpenRouter-Categories", payload?.appCategories ?? client._options.appCategories, { explode: false, charEncoding: "none" }),
        "X-OpenRouter-Title": encodeSimple("X-OpenRouter-Title", payload?.appTitle ?? client._options.appTitle, { explode: false, charEncoding: "none" }),
    }));
    const secConfig = await extractSecurity(client._options.apiKey);
    const securityInput = secConfig == null ? {} : { apiKey: secConfig };
    const requestSecurity = resolveGlobalSecurity(securityInput);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "getRankingsDaily",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: client._options.apiKey,
        retryConfig: options?.retries
            || client._options.retryConfig
            || {
                strategy: "backoff",
                backoff: {
                    initialInterval: 500,
                    maxInterval: 60000,
                    exponent: 1.5,
                    maxElapsedTime: 3600000,
                },
                retryConnectionErrors: true,
            }
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["5XX"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "GET",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        query: query,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [requestRes, { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        isErrorStatusCode: (statusCode) => matchStatusCode({ status: statusCode }, ["4XX", "5XX"]),
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [doResult, { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result] = await M.match(M.json(200, models.RankingsDailyResponse$inboundSchema), M.jsonErr(400, errors.BadRequestResponseError$inboundSchema), M.jsonErr(401, errors.UnauthorizedResponseError$inboundSchema), M.jsonErr(429, errors.TooManyRequestsResponseError$inboundSchema), M.jsonErr(500, errors.InternalServerResponseError$inboundSchema), M.fail("4XX"), M.fail("5XX"))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [result, { status: "complete", request: req, response }];
    }
    return [result, { status: "complete", request: req, response }];
}
