// @openrouter/sdk/esm/funcs/modelsList.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: deff050b46c5
 */
import { encodeFormQuery, encodeSimple } from "../lib/encodings.js";
import { matchStatusCode } from "../lib/http.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as errors from "../models/errors/index.js";
import * as operations from "../models/operations/index.js";
import { APIPromise } from "../types/async.js";
import { createPageIterator, haltIterator, } from "../types/operations.js";
/**
 * List all models and their properties
 */
export function modelsList(client, request, options) {
    return new APIPromise($do(client, request, options));
}
async function $do(client, request, options) {
    const parsed = safeParse(request, (value) => operations.GetModelsRequest$outboundSchema.optional().parse(value), "Input validation failed");
    if (!parsed.ok) {
        return [haltIterator(parsed), { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = null;
    const path = pathToFunc("/models")();
    const query = encodeFormQuery({
        "arch": payload?.arch,
        "category": payload?.category,
        "context": payload?.context,
        "distillable": payload?.distillable,
        "input_modalities": payload?.input_modalities,
        "limit": payload?.limit,
        "max_age_days": payload?.max_age_days,
        "max_agentic_index": payload?.max_agentic_index,
        "max_coding_index": payload?.max_coding_index,
        "max_intelligence_index": payload?.max_intelligence_index,
        "max_output_price": payload?.max_output_price,
        "max_price": payload?.max_price,
        "max_tool_success_rate": payload?.max_tool_success_rate,
        "min_age_days": payload?.min_age_days,
        "min_agentic_index": payload?.min_agentic_index,
        "min_coding_index": payload?.min_coding_index,
        "min_intelligence_index": payload?.min_intelligence_index,
        "min_output_price": payload?.min_output_price,
        "min_price": payload?.min_price,
        "min_tool_success_rate": payload?.min_tool_success_rate,
        "model_authors": payload?.model_authors,
        "offset": payload?.offset,
        "output_modalities": payload?.output_modalities,
        "providers": payload?.providers,
        "q": payload?.q,
        "region": payload?.region,
        "sort": payload?.sort,
        "supported_parameters": payload?.supported_parameters,
        "zdr": payload?.zdr,
    });
    const headers = new Headers(compactMap({
        Accept: "application/json",
        "HTTP-Referer": encodeSimple("HTTP-Referer", payload?.["HTTP-Referer"] ?? client._options.httpReferer, { explode: false, charEncoding: "none" }),
        "X-OpenRouter-Categories": encodeSimple("X-OpenRouter-Categories", payload?.appCategories ?? client._options.appCategories, { explode: false, charEncoding: "none" }),
        "X-OpenRouter-Title": encodeSimple("X-OpenRouter-Title", payload?.appTitle ?? client._options.appTitle, { explode: false, charEncoding: "none" }),
    }));
    const secConfig = await extractSecurity(client._options.apiKey);
    const securityInput = secConfig == null ? {} : { apiKey: secConfig };
    const requestSecurity = resolveGlobalSecurity(securityInput);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "getModels",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: client._options.apiKey,
        retryConfig: options?.retries
            || client._options.retryConfig
            || {
                strategy: "backoff",
                backoff: {
                    initialInterval: 500,
                    maxInterval: 60000,
                    exponent: 1.5,
                    maxElapsedTime: 3600000,
                },
                retryConnectionErrors: true,
            }
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["5XX"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "GET",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        query: query,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [haltIterator(requestRes), { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        isErrorStatusCode: (statusCode) => matchStatusCode({ status: statusCode }, ["4XX", "5XX"]),
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [haltIterator(doResult), { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result, raw] = await M.match(M.json(200, operations.GetModelsResponse$inboundSchema, { key: "Result" }), M.jsonErr(400, errors.BadRequestResponseError$inboundSchema), M.jsonErr(403, errors.ForbiddenResponseError$inboundSchema), M.jsonErr(500, errors.InternalServerResponseError$inboundSchema), M.fail("4XX"), M.fail("5XX"))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [haltIterator(result), {
                status: "complete",
                request: req,
                response,
            }];
    }
    const nextFunc = (responseData) => {
        const offset = request?.offset ?? 0;
        if (!responseData) {
            return { next: () => null };
        }
        const results = responseData.data;
        if (!Array.isArray(results) || !results.length) {
            return { next: () => null };
        }
        const limit = request?.limit ?? 500;
        if (results.length < limit) {
            return { next: () => null };
        }
        const nextOffset = offset + results.length;
        const nextVal = () => modelsList(client, {
            ...request,
            offset: nextOffset,
        }, options);
        return { next: nextVal, "~next": { offset: nextOffset } };
    };
    const page = { ...result, ...nextFunc(raw) };
    return [{ ...page, ...createPageIterator(page, (v) => !v.ok) }, {
            status: "complete",
            request: req,
            response,
        }];
}
